<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2.1 Mass spectrometry data handling and (pre-)processing | The MetaRbolomics book</title>
  <meta name="description" content="2.1 Mass spectrometry data handling and (pre-)processing | The MetaRbolomics book" />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="2.1 Mass spectrometry data handling and (pre-)processing | The MetaRbolomics book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2.1 Mass spectrometry data handling and (pre-)processing | The MetaRbolomics book" />
  
  
  

<meta name="author" content="Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="0200-R_packages_for_metabolomics.html"/>
<link rel="next" href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.11/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-rowgroup-1.10.19/css/rowGroup.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-rowgroup-1.10.19/js/dataTables.rowGroup.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/vis-4.20.1/vis.css" rel="stylesheet" />
<script src="libs/vis-4.20.1/vis.min.js"></script>
<script src="libs/visNetwork-binding-2.0.9/visNetwork.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The MetaRbolomics book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to"><i class="fa fa-check"></i>How to …</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#this-book-in-other-formats"><i class="fa fa-check"></i>This book in other formats</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="0100-introduction.html"><a href="0100-introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="0101-introduction-Metabolomics_data_processing_and_analysis.html"><a href="0101-introduction-Metabolomics_data_processing_and_analysis.html"><i class="fa fa-check"></i><b>1.1</b> Metabolomics data processing and analysis</a></li>
<li class="chapter" data-level="1.2" data-path="0102-introduction-The_R_package_landscape.html"><a href="0102-introduction-The_R_package_landscape.html"><i class="fa fa-check"></i><b>1.2</b> The R package landscape</a></li>
<li class="chapter" data-level="1.3" data-path="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><a href="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><i class="fa fa-check"></i><b>1.3</b> Dependences and connectivity of metabolomics packages</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="0200-R_packages_for_metabolomics.html"><a href="0200-R_packages_for_metabolomics.html"><i class="fa fa-check"></i><b>2</b> R-packages for metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.1</b> Mass spectrometry data handling and (pre-)processing</a><ul>
<li class="chapter" data-level="2.1.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#profile-mode-and-centroided-data"><i class="fa fa-check"></i><b>2.1.1</b> Profile mode and centroided data</a></li>
<li class="chapter" data-level="2.1.2" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#direct-infusion-mass-spectrometry-data"><i class="fa fa-check"></i><b>2.1.2</b> Direct infusion mass spectrometry data</a></li>
<li class="chapter" data-level="2.1.3" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#hyphenated-ms-and-non-targeted-data"><i class="fa fa-check"></i><b>2.1.3</b> Hyphenated MS and non-targeted data</a></li>
<li class="chapter" data-level="2.1.4" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#targeted-data-and-alternative-representations-of-data"><i class="fa fa-check"></i><b>2.1.4</b> Targeted data and alternative representations of data</a></li>
<li class="chapter" data-level="2.1.5" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#additional-dimensionality"><i class="fa fa-check"></i><b>2.1.5</b> Additional dimensionality</a></li>
<li class="chapter" data-level="2.1.6" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#structuring-data-and-metadata"><i class="fa fa-check"></i><b>2.1.6</b> Structuring data and metadata</a></li>
<li class="chapter" data-level="2.1.7" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#ion-species-grouping-and-annotation"><i class="fa fa-check"></i><b>2.1.7</b> Ion species grouping and annotation</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><i class="fa fa-check"></i><b>2.2</b> Metabolite identification with MS/MS data</a><ul>
<li class="chapter" data-level="2.2.1" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#msms-data-handling-spectral-matching-and-clustering"><i class="fa fa-check"></i><b>2.2.1</b> MS/MS data handling, spectral matching and clustering</a></li>
<li class="chapter" data-level="2.2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#reading-of-spectral-databases"><i class="fa fa-check"></i><b>2.2.2</b> Reading of spectral databases</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><a href="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.3</b> NMR data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.4" data-path="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><a href="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.4</b> UV data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.5" data-path="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><a href="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><i class="fa fa-check"></i><b>2.5</b> Statistical analysis of metabolomics data</a></li>
<li class="chapter" data-level="2.6" data-path="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><a href="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><i class="fa fa-check"></i><b>2.6</b> Handling of molecule structures and chemical structure databases</a></li>
<li class="chapter" data-level="2.7" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><i class="fa fa-check"></i><b>2.7</b> Network analysis and biochemical pathways</a><ul>
<li class="chapter" data-level="2.7.1" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#network-infrastructure-and-analysis"><i class="fa fa-check"></i><b>2.7.1</b> Network infrastructure and analysis</a></li>
<li class="chapter" data-level="2.7.2" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#metabolite-annotation"><i class="fa fa-check"></i><b>2.7.2</b> Metabolite annotation</a></li>
<li class="chapter" data-level="2.7.3" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#generation-of-metabolic-networks"><i class="fa fa-check"></i><b>2.7.3</b> Generation of metabolic networks</a></li>
<li class="chapter" data-level="2.7.4" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-analysis"><i class="fa fa-check"></i><b>2.7.4</b> Pathway analysis</a></li>
<li class="chapter" data-level="2.7.5" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-resources-and-interfaces"><i class="fa fa-check"></i><b>2.7.5</b> Pathway resources and interfaces</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><a href="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><i class="fa fa-check"></i><b>2.8</b> Multifunctional workflows</a></li>
<li class="chapter" data-level="2.9" data-path="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><a href="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><i class="fa fa-check"></i><b>2.9</b> User interfaces and workflow management systems</a></li>
<li class="chapter" data-level="2.10" data-path="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><a href="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><i class="fa fa-check"></i><b>2.10</b> Metabolomics data sets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="0300-conclusion.html"><a href="0300-conclusion.html"><i class="fa fa-check"></i><b>3</b> Conclusions</a></li>
<li class="chapter" data-level="" data-path="0400-references.html"><a href="0400-references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="" data-path="9900-Appendices.html"><a href="9900-Appendices.html"><i class="fa fa-check"></i>Appendices</a><ul>
<li class="chapter" data-level="" data-path="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><a href="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><i class="fa fa-check"></i>Appendix 1: The MSP File Format and package support</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><i class="fa fa-check"></i>Appendix 2: metaRbolomics dependencies network</a><ul>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#libraries-and-settings"><i class="fa fa-check"></i>Libraries and settings</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#read-package-names-from-our-table"><i class="fa fa-check"></i>Read package names from our table</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#get-reverse-dependencies"><i class="fa fa-check"></i>Get reverse dependencies</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#building-dependency-network"><i class="fa fa-check"></i>Building dependency network</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#save-network-plot"><i class="fa fa-check"></i>Save network plot</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#notes"><i class="fa fa-check"></i>Notes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The MetaRbolomics book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div style="page-break-before: always;" />

<div id="mass-spectrometry-data-handling-and-pre-processing" class="section level2">
<h2><span class="header-section-number">2.1</span> Mass spectrometry data handling and (pre-)processing</h2>
<p>For all mass spectrometers, the fundamental data generated is a mass spectrum, i.e. mass-signal intensity pairs. MS-based metabolomics data is typically acquired either as a single mass spectrum or a collection of mass spectra over time, with the time axis (retention time) defined by chromatographic (or other time domain) separation. One of the first steps in metabolomics data processing is usually the reduction of the typically large raw data produced by the instrument to a much smaller set of so called <em>features</em>, which are then subjected to downstream data analysis and interpretation. Features normally represent integrated peaks for a given mass that have been aligned across samples. Establishing these features is called <em>pre-processing</em>. The feature detection approaches and packages applicable depend on the type and characteristics of the input data. This section describes the basic data structure for some of the common analytical approaches and shows appropriate tools in R for pre-processing such data, see Table 1 for an overview of the corresponding packages.</p>
<div id="profile-mode-and-centroided-data" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Profile mode and centroided data</h3>
<p>The mass spectra can be recorded in profile (also called continuum) mode, but are often ‘centroided’. Centroiding is, in effect, a process of peak detection for a profile mode mass spectrum (hence in the <em>m/z</em> dimension, not in a chromatographic dimension) - a gaussian region of a continuum spectrum with a sufficiently high signal to noise ratio is integrated to give a centroided mass (a “stick” in the mass spectrum as opposed to a continuous signal) and integrated area under the curve. This results in data of reduced size - what was many <em>m/z</em>-intensity pairs has been reduced to a single <em>m/z</em>-intensity pair. Practically, this reduces the file size considerably and many data processing tools (e.g. <em>centWave</em> in xcms) require MS data that has been centroided. The centroiding can be done either during acquisition on the fly by the instrument software, or as an initial processing step. Post-acquisition centroiding can be performed during conversion of the vendor data format to open formats; typically using msconvert from ProteoWizard <span class="citation">[<a href="0400-references.html#ref-chambers_2012">27</a>,<a href="0400-references.html#ref-kessner_2008">28</a>]</span> which in some cases provides access to vendor centroiding algorithms or can alternatively use its own built-in centroinding method. Dedicated vendor tools can also be used, and the R packages MSnbase also provides centroiding capabilities.</p>
</div>
<div id="direct-infusion-mass-spectrometry-data" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Direct infusion mass spectrometry data</h3>
<p>Currently, one of the highest throughput analytical approaches is direct infusion MS, where the sample is directly injected into the mass spectrometer without any chromatographic separation. This approach can be used with high mass resolution or ultra-high resolution mass spectrometers to discriminate isobaric analytes <span class="citation">[<a href="0400-references.html#ref-fuhrer_2011">29</a>]</span>. Summing or averaging these spectra generates a single mass spectrum, which is representative of that sample. Peak picking can be done using MassSpecWavelet that applies a continuous wavelet transform-based peak detection. xcms provides a wrapper for this function in the <em>findPeaks.MSW</em> function. In the Flow Injection Analysis analytical approach (FIA), the sample is transiently injected into the carrier stream flowing directly into the MS instrument. In the absence of chromatographic separation, matrix effects are a challenge for the quantification, especially in complex matrices. FIA coupled to High-Resolution Mass Spectrometry data can be processed with the proFIA workflow which provides efficient and robust peak detection and quantification.</p>
</div>
<div id="hyphenated-ms-and-non-targeted-data" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Hyphenated MS and non-targeted data</h3>
<p>Chromatographic separation before MS enables better measurement of complex samples and the ability to separate isobaric compounds. Here, the mass spectra are acquired over time as the sample components separate on the chromatography column. The mass spectrum at any given time has the same data structure as any mass spectrum - units of mass to charge ratio and time. As can be inferred from the above descriptions, chromatographically coupled mass spectrometry data is three-dimensional, with dimensions of retention time, <em>m/z</em>, and intensity.</p>
<p>For the pre-processing of LC-MS and GC-MS data, xcms is widely used. A recent paper reviewed some of the “xcms family” packages <span class="citation">[<a href="0400-references.html#ref-mahieu_2016a">30</a>]</span> though many more packages exist that build on xcms by providing tools for specialised analyses while others provide improvements of some of the xcms processing steps such as improved peak picking (xMSanalyzer, warpgroup, cosmiq). xcms itself provides a number of different algorithms for peak picking such as <em>matchedFilter</em> <span class="citation">[<a href="0400-references.html#ref-smith_2006">31</a>]</span><em>, centWave</em> <span class="citation">[<a href="0400-references.html#ref-tautenhahn_2008">32</a>]</span> and <em>massifquant</em> <span class="citation">[<a href="0400-references.html#ref-conley_2014">33</a>]</span>. apLCMS, yamss, KPIC2 and enviPick also provide peak picking for LC-MS data independently of xcms. In cases where the alignment of the peak data of different samples is considered (e.g. in cohort studies), xcms and apLCMS include methods to group the peaks by their <em>m/z</em> and retention times within tolerance levels. The groups are split into sub-groups using density functions and the consensus <em>m/z</em> and retention time is assigned to each bin.</p>
</div>
<div id="targeted-data-and-alternative-representations-of-data" class="section level3">
<h3><span class="header-section-number">2.1.4</span> Targeted data and alternative representations of data</h3>
<p>In addition to the most standard “spectra over time” representation of chromatographically separated MS data, there are a number of alternative ways to represent the data or simplify the data. The signal intensity for a given mass (or mass range) over chromatographic time can be represented as two equal length vectors, with retention time and intensity as units for the values of those vectors. Examples of these vector pairs include the extracted ion chromatogram (EIC, sometimes also referred to as selected or eXtracted ion chromatogram SIC, XIC), where these chromatograms represent the intensity of a given mass over (retention) time. The data thus contains no spectra but a number of SICs. Frequently this is accomplished by summarizing the raw data in a two dimensional matrix consisting of <em>m/z</em> and time dimensions, with each cell holding the signal intensity for that <em>m/z</em> and retention time range (or bin). Low mass resolution mass spectrometers often represent the data natively as a SIC and targeted data are also usually represented this way. Recent versions of xcms are also able to process such data, and additional xcms-based functionalities for analysis of targeted data can be found in the packages TargetSearch and SWATHtoMRM, while analysis of isotope labeled data can be found in the packages X13CMS, geoRge, and IsotopicLabelling. SIMAT also provides processing for targeted data and does not rely on xcms.</p>
</div>
<div id="additional-dimensionality" class="section level3">
<h3><span class="header-section-number">2.1.5</span> Additional dimensionality</h3>
<p>The vast majority of data collected for metabolomics comprises of three dimensions: retention time, <em>m/z</em>, and intensity. However, there are more complicated analytical approaches that add additional dimensionality to the data. Two dimensional chromatography offer two separations in the chromatographic (retention time) domain. The eluent from one column is captured by retention time range and transferred to a second column where a fast orthogonal separation occurs. When coupled to a mass spectrometer, this generates four-dimensional data (<em>m/z</em>, first retention time, second retention time, intensity).</p>
<p>Ion mobility separation (IMS) is a gas phase separation method offering resolution of ions based on molecular shape. This separation occurs on timescales of tens of microseconds, which generates a nested data structure in which there are dozens to hundreds of mass spectra collected across the IMS separation time scale. One can envision this as an ion mobility ‘chromatogram’ - however, this chromatogram is nested within the actual chromatographic separation, thus LC-IMS-MS data is also four dimensional.</p>
<p>Most MS instruments offer the capability to perform selection (or filtering) of ions for fragmentation. The precursor selection can be performed through a quadrupole or ion trap, and fragmentation is often induced by collisions with an inert collision gas. Because this adds a level of mass spectrometry, it is called tandem MS, MS2 or MS/MS. Ion trap instruments can further select fragment ions and acquire MSn spectra.</p>
<p>There are several data independent MS/MS approaches, whereby MS/MS precursor selection is done, typically, on a scanning basis. These approaches perform precursor selection in a manner which does not depend on any feedback from the instrument control software or the MS level data. In practice, this precursor window can be either <em>m/z</em> or ion mobility based. The processing tools within the R universe (discussed below) are so far underdeveloped for these approaches. With the increased popularity of multidimensional separation, the need for algorithms that can fully utilize the increased separation power is also increasing.</p>
<p>Currently, osd provide peak picking for unit resolution GC×GC-MS. While the msPeak package provides peak picking for GC×GC-MS data, the peak picking is done on the total ion chromatogram, thus not taking advantage of the mass selectivity provided by the MS detector. It does not appear that any package for R exists that provides peak picking for GC×GC-MS, LC×LC-MS or LC-IMS-MS, similar to (or even better than) commercial tools (e.g. ChromaTOF, GC Image, ChromSquare). Also, at least in the case of GC×GC-MS, unit mass resolution still seems to be the most common use-case, even though high-resolution MS could further improve signal deconvolution and ultimately, analyte identification. Such capabilities are crucial for moving these new powerful analytical approaches into mainstream metabolomics analysis.</p>
</div>
<div id="structuring-data-and-metadata" class="section level3">
<h3><span class="header-section-number">2.1.6</span> Structuring data and metadata</h3>
<p>The result from the pre-processing is usually a matrix of abundances, rows being features (or features grouped into compounds/molecules) and columns being the samples. Within the statistical community, it is common nowadays to manipulate data matrices with rows as observations and columns as features, this difference stems from the early days, when spreadsheet programs could only handle a limited number of columns smaller than the number of <em>e.g.</em> genes. Such matrices can be easily encapsulated into an <em>ExpressionSet</em> class from Bioconductor’s Biobase package <span class="citation">[<a href="0400-references.html#ref-huber_2015">34</a>]</span>, the more recent <em>SummarizedExperiment</em> defined in the SummarizedExperiment <span class="citation">[<a href="0400-references.html#ref-martinmorgan_2017">35</a>]</span> package or the <em>mSet</em> class from the metabolomics focussed Metabase <span class="citation">[<a href="0400-references.html#ref-zhu_website_2019">36</a>]</span> package. The main advantage of such objects is their inherent support to align quantitative data along with related metadata (i.e. feature definitions/annotations as row - and sample annotations as column metadata). As an example, a <em>SummarizedExperiment</em> can be generated from xcms pre-processing results by adding the output from the <em>featureValues</em> function on the xcms result object as quantitative assay and the outputs of the <em>featureDefinitions</em> and <em>pData</em> functions as row and column annotations, respectively. Many Bioconductor packages for omics data analysis have native support for such objects (e.g., pcaMethods, STATegRa, ropls, biosigner, omicade4).</p>
<p>For the downstream export of mass spectrometry data from metabolomics or lipidomics experiments, the package rmzTab-M provides support for exporting quantitative and identification results backed by analytical and mass spectrometric evidence into the mzTab-M metabolomics file format <span class="citation">[<a href="0400-references.html#ref-hoffmann_2019">37</a>]</span>.</p>
<div id="htmlwidget-ecfea33e4dec350181c6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ecfea33e4dec350181c6">{"x":{"filter":"none","extensions":["RowGroup"],"caption":"<caption>Table 1: R packages for mass spectrometry data handling and (pre-)processing.<\/caption>","fillContainer":false,"data":[["MS data handling","MS data handling","MS data handling","MS data handling","MS data handling","MS data handling","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Peak picking, grouping and alignment (LC-MS focussed or general)","Isotope labeling using MS","Isotope labeling using MS","Isotope labeling using MS","Isotope labeling using MS","Isotope labeling using MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","Targeted MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","GC-MS and GC×GC-MS","Flow injection / direct infusion analysis","Flow injection / direct infusion analysis","Flow injection / direct infusion analysis","Other","Other","Other","Other","Other","Other","Other","Other"],["Parser for common file formats: mzXML, mzData, mzML and netCDF. Usually not used directly by the end user, but provides functions to read raw data for other packages.","Infrastructure to manipulate, process and visualise MS and proteomics data, ranging from raw to quantitative and annotated data.","Export and import of processed metabolomics MS results to and from the mzTab-M for metabolomics data format.","Converts MRM-MS (.mzML) files to LC-MS style .mzML.","Infrastructure for import, handling, representation and analysis of chromatographic MS data.","Infrastructure for import, handling, representation and analysis of MS spectra.","Pre-processing and visualization for (LC/GC-)MS data. Includes visualization and simple statistics.","Automatic optimization of XCMS parameters based on isotopes.","Parameter tuning algorithm forXCMS, MZmine2, and other metabolomics data processingsoftware.","Pre-processing and visualization for (LC/GC-)MS data. Includes visualization and simple statistics.","Peak picking with XCMS and apLCMS, low intensity peak detection via replicate analyses. Multi-parameter feature extraction and data merging, sample quality and feature consistency evaluation. Annotation with METLIN and KEGG.","Pre-processing and alignment of LC-MS data without assuming a parametric peak shape model allowing maximum flexibility. It utilizes the knowledge of known metabolites, as well as robust machine learning.","Peak detection using chromatogram subregion detection, consensus integration bound determination and Accurate missing value integration. Outputs in XCMS-compatible format.","Peak picking for (LC/GC-)MS data, improving the detection of low abundance signals via a master map of m/z/RT space before peak detection. Results are XCMS-compatible.","m/z detection (i.e. peak-picking) for accurate mass data, collecting all data points above an intensity threshold, grouping them by m/z values and estimating representative m/z values for the clusters; extracting EICs.","(GC/LC)-MS data analysis for environmental science, including raw data processing, analysis of molecular isotope ratios, matrix effects, and short-chain chlorinated paraffins.","Sequential partitioning, clustering and peak detection of centroided LC-MS mass spectrometry data (.mzXML), with Interactive result and raw data plot.","PeakpickingwithXCMS. Groups chemically related features beforealignmentacross samples. Additional processing after alignment includes feature validation, re-integration and annotation based on custom database.","KPIC2 extracts pure ion chromatograms (PIC) via K-means clustering of ions in region of interest, performs grouping and alignment, grouping of isotopic and adduct features, peak filling and Random Forest classification.","Analysis of untargeted LC/MS data from stable isotope-labeling experiments. Also uses XCMS for feature detection.","Correction of MS and MS/MS data from stable isotope labeling (any tracer isotope) experiments for natural isotope abundance and tracer impurity. Separate GUI available in IsoCorrectoRGUI.","Extension of XCMS that provides support for isotopic labeling. Detection of metabolites that have been enriched with isotopic labeling.","Analysis of isotopic patterns in isotopically-labeled MS data. Estimates the isotopic abundance of the stable isotope (either 2H or 13C) within specified compounds.","Finding the dual (or multiple) isotope labeled analytes using dual labeling of metabolites for metabolome analysis (DLEMMA) approach, described in Liron [42].","Peak picking using peak apex intensities for selected masses. Reference library matching, RT/RI conversion plus metabolite identification using multiple correlated masses. Includes GUI.","Pre-processing for targeted (SIM) GC-MS data. Guided selection of appropriate fragments for the targets of interest by using an optimization algorithm based on user provided library.","Deconvolution of MS2 spectra obtained with wide isolation windows.","Deconvolution of SWATH-MS experiments to MRM transitions.","Automatic analysis of large scale MRM experiments.","Tailors peak detection for targeted metabolites through iterative user interface. It automatically integrates peak areas for all isotopologues and outputs extracted ion chromatograms (EICs).","Targeted peak picking and annotation. Includes Shiny GUI.","Toolkit for working with Selective Reaction Monitoring (SRM) MS data and other variants of targeted LC-MS data.","Deconvolution of SWATH-MS data.","Targeted peak picking and annotation. All functions through Shiny GUI.","Unsupervised data mining on GC-MS. Clustering of mass spectra to detect compound spectra. The output can be searched in NIST and ARISTO [50].","Pre-processing for GC/MS, MassBank search, NIST format export.","Pre-processing using AMDIS [53, 54] for untargeted GC-MS analysis. Feature grouping across samples, improved quantification, removal of false positives, normalisation via internal standard or biomass; basic statistics.","Deconvolution of GC-MS and GC×GC-MS unit resolution data using orthogonal signal deconvolution (OSD), independent component regression (ICR) and multivariate curve resolution (MCR-ALS).","Corrects overloaded signals directly in raw data (from GC-APCI-MS) automatically by using a Gaussian or isotopic-ratio approach.","Alignment of GC data. Also GC-FID or any single channel data since it works directly on peak lists.","GC-MS data processing and compound annotation pipeline. Includes the building, validating, and query of in-house databases.","Peak picking for GC×GC-MS using bayes factor and mixture probability models.","Peak alignment for GC×GC-MS data with homogeneous peaks based on mixture similarity measures.","Peak alignment for GC×GC-MS data with homogeneous and/or heterogenous peaks based on mixture similarity measures.","Chemometrics analysis GC×GC-MS: baseline correction, smoothing, COW peak alignment, multiway PCA is incorporated.","Retention time and mass spectra similarity threshold-free alignments, seamlessly integrates retention time standards for universally reproducible alignments, performs common ion filtering, and provides compatibility with multiple peak quantification methods.","Pre-processing of data from Flow Injection Analysis (FIA) coupled to High-Resolution Mass Spectrometry (HRMS).","Flow In-jection Electrospray Mass Spectrometry Processing: data processing, classification modelling and variable selection in metabolite fingerprinting","Processing Mass Spectrometry spectrum by using wavelet based algorithm. Can be used for direct infusion experiments.","Filtering of features originating from artifactual interference. Based on the analysis of an extract of E. coli grown in 13C-enriched media.","Wrappers for XCMS and CAMERA. Also includes matching to a spectral library and a GUI.","Processing of peaktables from AMDIS, XCMS or ChromaTOF. Functions for plotting also provided.","Parametric Time Warping (RT correction) for both DAD and LC-MS.","R wrapper for X!Tandem software for protein identification.","Building, validation, and statistical analysis of extended assay libraries for SWATH proteomics data.","Split a data set into a set of likely true metabolites and likely measurement artifacts by comparing missing rates of pooled plasma samples and biological samples.","Quality of LC-MS and direct infusion MS data. Generates a report that contains a comprehensive set of quality control metrics and charts."],["<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.mzR>mzR<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.MSnbase>MSnbase<\/a>","<a target=_blank href=https://lifs-tools.github.io/rmzTab-m/index.html>rmzTab-M<\/a>","<a target=_blank href=https://github.com/wilsontom/MRMConverteR>MRMConverteR<\/a>","<a target=_blank href=https://github.com/RforMassSpectrometry/Chromatograms>Chromatograms<\/a>","<a target=_blank href=https://github.com/RforMassSpectrometry/Spectra>Spectra<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.xcms>xcms<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/IPO.html>IPO<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.Autotuner>Autotuner<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/yamss.html>yamss<\/a>","<a target=_blank href=https://sourceforge.net/projects/xmsanalyzer/>xMSanalyzer<\/a>","<a target=_blank href=http://web1.sph.emory.edu/apLCMS/>apLCMS<\/a>","<a target=_blank href=https://github.com/nathaniel-mahieu/warpgroup>warpgroup<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.cosmiq>cosmiq<\/a>","<a target=_blank href=http://amdorap.sourceforge.net/>AMDORAP<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/enviGCMS/index.html>enviGCMS<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/enviPick/index.html>enviPick<\/a>","<a target=_blank href=https://github.com/lauzikaite/massFlowR>massFlowR<\/a>","<a target=_blank href=https://github.com/hcji/KPIC2>KPIC2<\/a>","<a target=_blank href=https://github.com/jcapelladesto/geoRge>geoRge<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/IsoCorrectoR.html>IsoCorrectoR<\/a>","<a target=_blank href=http://pattilab.wustl.edu/software/x13cms/x13cms.php>X13CMS<\/a>","<a target=_blank href=https://github.com/RuggeroFerrazza/IsotopicLabelling>IsotopicLabelling<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/Miso>Miso<\/a>","<a target=_blank href=http://bioconductor.org/packages/2.5/bioc/html/TargetSearch.html>TargetSearch<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.SIMAT>SIMAT<\/a>","<a target=_blank href=http://pattilab.wustl.edu/software/decoms2/decoms2.php>decoMS2<\/a>","<a target=_blank href=http://www.metabolomics-shanghai.org/softwaredetail.php?id=128>SWATHtoMRM<\/a>","<a target=_blank href=http://www.metabolomics-shanghai.org/softwaredetail.php?id=34>MRMAnalyzer<\/a>","<a target=_blank href=https://github.com/stevehoang/assayr>AssayR<\/a>","<a target=_blank href=https://github.com/phenomecentre/peakPantheR>peakPantheR<\/a>","<a target=_blank href=https://github.com/wilsontom/sRm>sRm<\/a>","<a target=_blank href=https://github.com/ZhuMSLab/DecoMetDIA>DecoMetDIA<\/a>","<a target=_blank href=https://github.com/hcji/TarMet>TarMet<\/a>","<a target=_blank href=https://cran.r-project.org/package=MSeasy>MSeasy<\/a>","<a target=_blank href=https://cran.r-project.org/package=erah>erah<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.Metab>Metab<\/a>","<a target=_blank href=http://cran.r-project.org/package=osd>osd<\/a>","<a target=_blank href=https://cran.r-project.org/package=CorrectOverloadedPeaks>CorrectOverloadedPeaks<\/a>","<a target=_blank href=https://cran.r-project.org/package=GCalignR>GCalignR<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/metaMS.html>metaMS<\/a>","<a target=_blank href=http://mrr.sourceforge.net/download.html>msPeak<\/a>","<a target=_blank href=http://mrr.sourceforge.net/download.html>mSPA<\/a>","<a target=_blank href=http://mrr.sourceforge.net/download.html>SWPA<\/a>","<a target=_blank href=https://cran.r-project.org/package=RGCxGC>RGCxGC<\/a>","<a target=_blank href=https://github.com/rramaker/R2DGC>R2DGC<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.proFIA>proFIA<\/a>","<a target=_blank href=https://github.com/aberHRML/FIEmspro>FIEmspro<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/MassSpecWavelet.html>MassSpecWavelet<\/a>","<a target=_blank href=https://github.com/pattilab/credential>credential<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.metaMS>metaMS<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.flagme>flagme<\/a>","<a target=_blank href=https://cran.r-project.org/package=ptw>ptw<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/rTANDEM.html>rTANDEM<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/SwathXtend.html>SwathXtend<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetProc>MetProc<\/a>","<a target=_blank href=https://github.com/computational-metabolomics/qcrms>qcrms<\/a>"],["BioC","BioC","GitHub","GitHub","GitHub","GitHub","BioC","BioC","BioC","BioC","SF","SF","GitHub","BioC","SF","CRAN","CRAN","GitHub","GitHub","GitHub","BioC",null,"GitHub","CRAN","BioC","BioC",null,null,null,"GitHub","GitHub","GitHub","GitHub","GitHub","CRAN","CRAN","BioC","CRAN","CRAN","CRAN","BioC","SF","SF","SF","CRAN","GitHub","BioC","GitHub","BioC","GitHub","BioC","BioC","CRAN","BioC","BioC","CRAN","GitHub"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Section<\/th>\n      <th>Functionalities<\/th>\n      <th>Package<\/th>\n      <th>Repo<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"rowGroup":{"dataSrc":0},"ordering":false,"columnDefs":[{"visible":false,"targets":0}],"pageLength":-1,"dom":"ft","order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[-1,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ion-species-grouping-and-annotation" class="section level3">
<h3><span class="header-section-number">2.1.7</span> Ion species grouping and annotation</h3>
<p>In MS-based metabolomics, the characterisation and identification of metabolites involves several steps and approaches. After peak (feature) table generation, several tools can be used for grouping features that are postulated to originate from the same molecule. These include the widely-used CAMERA for MS1 data, as well as RAMClustR (particularly for DIA data), MetTailor, nontarget, CliqueMS and peakANOVA. Packages that support interpretation of the relationship between the ion species, including adducts, isotopes and in-source fragmentation, are InterpretMSSpectrum, CAMERA, nontarget and mzMatch <span class="citation">[<a href="0400-references.html#ref-scheltema_2011">38</a>]</span>. See Table 2 for a summary of these packages.</p>
<p>Detailed reconstructed isotope patterns can be used to determine the molecular formula of potential candidates. In the case of molecular formula and isotope analysis, the <em>m/z</em> and intensities for a given (set of) features can be used to calculate a ranked list of possible molecular formulas, based on the accurate mass and relative isotope abundances. The Rdisop, GenFormR and enviPat packages are able to simulate and decompose isotopic patterns into molecular formula candidates. Some post processing can calculate e.g. the double bond equivalents (DBE) and similar characteristics to reduce the number of false positive assignments. Another additional source of information to improve molecular formula estimation is to include MS/MS spectra, as used in MFAssignR, InterpretMSSpectrum or GenFormR.</p>
<p>A typical next step is the annotation of <em>m/z</em> with putative metabolites using accurate mass lookup, or if the molecular formula was calculated, lookup of the formula in metabolite databases. It has to be noted that annotation with accurate mass search by no means is equivalent to identification. Under the assumption that all the metabolites measured in a sample have some biochemical relation, a global annotation strategy as used in ProbMetab can help as well. Here, the individual ranked lists of formulae are re-evaluated to also maximize the number of pairs with (potential) biochemical substrate-product pairs. The masstrixR package contains several utility functions for accurate mass lookup. It enables matching of measured <em>m/z</em> values against a given database or library and can additionally perform matching based on retention times (RT) and/or collisional cross sections (CCS) if available.</p>
<div id="htmlwidget-7dafb211b5b2739c4bf7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7dafb211b5b2739c4bf7">{"x":{"filter":"none","extensions":["RowGroup"],"caption":"<caption>Table 2: R packages for ion species grouping, annotation, molecular formula generation and accurate mass lookup.<\/caption>","fillContainer":false,"data":[["Molecular formula and isotope analysis","Molecular formula and isotope analysis","Molecular formula and isotope analysis","Molecular formula and isotope analysis","Molecular formula and isotope analysis","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","MS feature grouping","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","Ion/adduct/fragment annotation","MS feature grouping"],["Simulation of and decomposition of Isotopic Patterns.","Calculation of isotope fine patterns. Also adduct calculations and molecular formula parsing. Web version available at www.envipat.eawag.ch.","Molecular formula assignment, mass recalibration, signal-to-noise evaluation, and unambiguous formula selections are provided.","Uses GenForm for molecular formula generation on mass accuracy, isotope and/or MS/MS fragments, as well as performing MS/MS subformula annotation.","Checking element isotopes, calculating (isotope labelled) exact monoisotopic mass,m/zvalues, mass accuracy, and inspecting possible contaminant mass peaks, examining possible adducts in ESI and MALDI.","Grouping of correlated features into pseudo compound spectra using correlation across samples and similarity of peak shape. Annotation of isotopes and adducts. Works as an add-on to XCMS.","Grouping of features based on similarity between coelution profiles.","Cluster based feature grouping for non-targeted GC or LC-MS data.","Uses dynamic block summarisation to group features belong to the same compound. Correction for peak misalignments and isotopic pattern validation.","Isotope & adduct peak grouping, homologous series detection.","Bayesian approach for grouping peaks originating from the same compound.","Combination of data from positive and negative ionization mode finding common molecular entities.","Grouping of correlated features into pseudo compound spectra using correlation across sample. Annotation of isotopes and adducts. Can work directly with the XCMS output.","Navigation of high-resolution MS/MS data in a GUI based on mass spectral similarity.","Deconvolution of MS/MS spectra obtained with wide isolation windows.","Bayesian probabilistic annotation.","Isotope & adduct peak grouping, unsupervised homologous series detection.","Automatic interpretation of fragments and adducts in MS spectra. Molecular formula prediction based on fragmentation.","Automated annotation using MS2 data or databases and retention time. Calculation of spectral and chemical networks.","Screening, annotation, and putative identification of mass spectral features in lipidomics. Default databases contain ~25,000 compounds.","Automated annotation of fragments from MS and MS2 and putative identification against simulated library fragments of ~500,000 lipid species across ~60 lipid types.","Annotation of lipid type and acyl groups on independent acquisition-mass spectrometry lipidomics based on fragmentation and intensity rules.","Accurate mass and/or retention time and/or collisional cross section matching.","Downloads KEGG compounds orthology data and wraps the KEGGREST package to extractgene data.","Paired mass distance analysis to find independent peaks inm/z-retention time profiles based on retention time hierarchical cluster analysis and frequency analysis of paired mass distances within retention time groups. Structure directed analysis to find potential relationship among those independent peaks. Shiny GUI included.","A Bayesian annotation method for LC/MS data integrating biochemical relations, isotope patterns and adduct formation.","Preprocessing (xcms), replicate merging, noise, blank and missingness filtering, feature grouping, annotation of known compounds, isotopic labeling analysis, annotation from KEGG or HMDB, common biotransformations and probabilistic putative metabolite annotation using MetAssign.","Putative annotation of unknowns in MS1 data.","Feature grouping and annotation workflow based on MS-DIAL peak character estimation alghoritm and MS-Finder hydrogen rearengment rules"],["<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/Rdisop.html>Rdisop<\/a>","<a target=_blank href=https://cran.r-project.org/package=enviPat>enviPat<\/a>","<a target=_blank href=https://github.com/ChARM-Group/MFAssignR>MFAssignR<\/a>","<a target=_blank href=https://github.com/schymane/GenFormR>GenFormR<\/a>","<a target=_blank href=https://cran.r-project.org/package=MSbox>MSbox<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.CAMERA>CAMERA<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/cliqueMS/>CliqueMS<\/a>","<a target=_blank href=https://github.com/cbroeckl/RAMClustR>RAMClustR<\/a>","<a target=_blank href=https://sourceforge.net/projects/mettailor/>MetTailor<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/nontarget/>nontarget<\/a>","<a target=_blank href=http://research.cs.aalto.fi/pml/software/peakANOVA/>peakANOVA<\/a>","<a target=_blank href=https://cran.r-project.org/package=MScombine>MScombine<\/a>","<a target=_blank href=http://www.urr.cat/AStream/AStream.html>Astream<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/MetCirc.html>MetCirc<\/a>","<a target=_blank href=http://pattilab.wustl.edu/software/decoms2/decoms2.php>decoMS2<\/a>","<a target=_blank href=https://github.com/rsilvabioinfo/ProbMetab>ProbMetab<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/nontarget/>nontarget<\/a>","<a target=_blank href=https://cran.r-project.org/package=InterpretMSSpectrum>InterpretMSSpectrum<\/a>","<a target=_blank href=https://github.com/WMBEdmands/compMS2Miner>compMS2Miner<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.LOBSTAHS>LOBSTAHS<\/a>","<a target=_blank href=https://github.com/GarrettLab-UF/LipidMatch>LipidMatch<\/a>","<a target=_blank href=https://cran.r-project.org/package=LipidMS>LipidMS<\/a>","<a target=_blank href=https://github.com/michaelwitting/masstrixR>masstrixR<\/a>","<a target=_blank href=https://cran.r-project.org/package=omu>omu<\/a>","<a target=_blank href=https://cran.r-project.org/package=pmd>pmd<\/a>","<a target=_blank href=https://github.com/francescodc87/IPA>IPA<\/a>","<a target=_blank href=https://github.com/andzajan/mzmatch.R>mzMatch<\/a>","<a target=_blank href=https://sourceforge.net/projects/xmsannotator/>xMSAnnotator<\/a>","<a target=_blank href=https://github.com/eMetaboHUB/MS-CleanR>MS-CleanR<\/a>"],["BioC","CRAN","GitHub","GitHub","CRAN","BioC","CRAN","CRAN","SF","CRAN",null,"CRAN",null,"BioC",null,"GitHub","CRAN","CRAN","GitHub","BioC","GitHub","CRAN","GitHub","CRAN","CRAN","GitHub","GitHub","SF","GitHub"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Section<\/th>\n      <th>Functionalities<\/th>\n      <th>Package<\/th>\n      <th>Repo<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"rowGroup":{"dataSrc":0},"ordering":false,"columnDefs":[{"visible":false,"targets":0}],"pageLength":-1,"dom":"ft","order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[-1,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>

            </section>

          </div>
        </div>
      </div>
<a href="0200-R_packages_for_metabolomics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rformassspectrometry/metaRbolomics-book/edit/master/rmd/02-R_packages_for_metabolomics/0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["The_MetaRbolomics_book.pdf", "The_MetaRbolomics_book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
