<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2.5 Statistical analysis of metabolomics data | The MetaRbolomics book</title>
  <meta name="description" content="2.5 Statistical analysis of metabolomics data | The MetaRbolomics book" />
  <meta name="generator" content="bookdown 0.16 and GitBook 2.6.7" />

  <meta property="og:title" content="2.5 Statistical analysis of metabolomics data | The MetaRbolomics book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2.5 Statistical analysis of metabolomics data | The MetaRbolomics book" />
  
  
  

<meta name="author" content="Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"/>
<link rel="next" href="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.11/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-rowgroup-1.10.19/css/rowGroup.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-rowgroup-1.10.19/js/dataTables.rowGroup.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/vis-4.20.1/vis.css" rel="stylesheet" />
<script src="libs/vis-4.20.1/vis.min.js"></script>
<script src="libs/visNetwork-binding-2.0.9/visNetwork.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The MetaRbolomics book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to"><i class="fa fa-check"></i>How to …</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#this-book-in-other-formats"><i class="fa fa-check"></i>This book in other formats</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="0100-introduction.html"><a href="0100-introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="0101-introduction-Metabolomics_data_processing_and_analysis.html"><a href="0101-introduction-Metabolomics_data_processing_and_analysis.html"><i class="fa fa-check"></i><b>1.1</b> Metabolomics data processing and analysis</a></li>
<li class="chapter" data-level="1.2" data-path="0102-introduction-The_R_package_landscape.html"><a href="0102-introduction-The_R_package_landscape.html"><i class="fa fa-check"></i><b>1.2</b> The R package landscape</a></li>
<li class="chapter" data-level="1.3" data-path="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><a href="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><i class="fa fa-check"></i><b>1.3</b> Dependences and connectivity of metabolomics packages</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="0200-R_packages_for_metabolomics.html"><a href="0200-R_packages_for_metabolomics.html"><i class="fa fa-check"></i><b>2</b> R-packages for metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.1</b> Mass spectrometry data handling and (pre-)processing</a><ul>
<li class="chapter" data-level="2.1.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#profile-mode-and-centroided-data"><i class="fa fa-check"></i><b>2.1.1</b> Profile mode and centroided data</a></li>
<li class="chapter" data-level="2.1.2" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#direct-infusion-mass-spectrometry-data"><i class="fa fa-check"></i><b>2.1.2</b> Direct infusion mass spectrometry data</a></li>
<li class="chapter" data-level="2.1.3" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#hyphenated-ms-and-non-targeted-data"><i class="fa fa-check"></i><b>2.1.3</b> Hyphenated MS and non-targeted data</a></li>
<li class="chapter" data-level="2.1.4" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#targeted-data-and-alternative-representations-of-data"><i class="fa fa-check"></i><b>2.1.4</b> Targeted data and alternative representations of data</a></li>
<li class="chapter" data-level="2.1.5" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#additional-dimensionality"><i class="fa fa-check"></i><b>2.1.5</b> Additional dimensionality</a></li>
<li class="chapter" data-level="2.1.6" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#structuring-data-and-metadata"><i class="fa fa-check"></i><b>2.1.6</b> Structuring data and metadata</a></li>
<li class="chapter" data-level="2.1.7" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#ion-species-grouping-and-annotation"><i class="fa fa-check"></i><b>2.1.7</b> Ion species grouping and annotation</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><i class="fa fa-check"></i><b>2.2</b> Metabolite identification with MS/MS data</a><ul>
<li class="chapter" data-level="2.2.1" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#msms-data-handling-spectral-matching-and-clustering"><i class="fa fa-check"></i><b>2.2.1</b> MS/MS data handling, spectral matching and clustering</a></li>
<li class="chapter" data-level="2.2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#reading-of-spectral-databases"><i class="fa fa-check"></i><b>2.2.2</b> Reading of spectral databases</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><a href="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.3</b> NMR data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.4" data-path="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><a href="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.4</b> UV data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.5" data-path="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><a href="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><i class="fa fa-check"></i><b>2.5</b> Statistical analysis of metabolomics data</a></li>
<li class="chapter" data-level="2.6" data-path="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><a href="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><i class="fa fa-check"></i><b>2.6</b> Handling of molecule structures and chemical structure databases</a></li>
<li class="chapter" data-level="2.7" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><i class="fa fa-check"></i><b>2.7</b> Network analysis and biochemical pathways</a><ul>
<li class="chapter" data-level="2.7.1" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#network-infrastructure-and-analysis"><i class="fa fa-check"></i><b>2.7.1</b> Network infrastructure and analysis</a></li>
<li class="chapter" data-level="2.7.2" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#metabolite-annotation"><i class="fa fa-check"></i><b>2.7.2</b> Metabolite annotation</a></li>
<li class="chapter" data-level="2.7.3" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#generation-of-metabolic-networks"><i class="fa fa-check"></i><b>2.7.3</b> Generation of metabolic networks</a></li>
<li class="chapter" data-level="2.7.4" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-analysis"><i class="fa fa-check"></i><b>2.7.4</b> Pathway analysis</a></li>
<li class="chapter" data-level="2.7.5" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-resources-and-interfaces"><i class="fa fa-check"></i><b>2.7.5</b> Pathway resources and interfaces</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><a href="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><i class="fa fa-check"></i><b>2.8</b> Multifunctional workflows</a></li>
<li class="chapter" data-level="2.9" data-path="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><a href="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><i class="fa fa-check"></i><b>2.9</b> User interfaces and workflow management systems</a></li>
<li class="chapter" data-level="2.10" data-path="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><a href="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><i class="fa fa-check"></i><b>2.10</b> Metabolomics data sets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="0300-conclusion.html"><a href="0300-conclusion.html"><i class="fa fa-check"></i><b>3</b> Conclusions</a></li>
<li class="chapter" data-level="" data-path="0400-references.html"><a href="0400-references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="" data-path="9900-Appendices.html"><a href="9900-Appendices.html"><i class="fa fa-check"></i>Appendices</a><ul>
<li class="chapter" data-level="" data-path="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><a href="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><i class="fa fa-check"></i>Appendix 1: The MSP File Format and package support</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><i class="fa fa-check"></i>Appendix 2: metaRbolomics dependencies network</a><ul>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#libraries-and-settings"><i class="fa fa-check"></i>Libraries and settings</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#read-package-names-from-our-table"><i class="fa fa-check"></i>Read package names from our table</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#get-reverse-dependencies"><i class="fa fa-check"></i>Get reverse dependencies</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#building-dependency-network"><i class="fa fa-check"></i>Building dependency network</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#save-network-plot"><i class="fa fa-check"></i>Save network plot</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#notes"><i class="fa fa-check"></i>Notes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The MetaRbolomics book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div style="page-break-before: always;" />

</div>
<div id="statistical-analysis-of-metabolomics-data" class="section level2">
<h2><span class="header-section-number">2.5</span> Statistical analysis of metabolomics data</h2>
<p>Following the feature detection and grouping steps outlined in the sections above, different paths to statistical analysis are available in R and Bioconductor. Once the “sample versus variable” feature matrix of molecule intensities or abundances has been generated, comprehensive statistical analyzes can be performed by using the vast range of packages provided by the R statistical software and the Bioconductor project (see Table 6), see for instance StatisticalMethod biocViews <span class="citation">[<a href="0400-references.html#ref-bioconductor_website_ndb">64</a>]</span> and the ExperimentalDesign <span class="citation">[<a href="0400-references.html#ref-groemping_website_2019">65</a>]</span>, Cluster <span class="citation">[<a href="0400-references.html#ref-leisch_website_2019">66</a>]</span>, Multivariate <span class="citation">[<a href="0400-references.html#ref-hewson_website_2018">67</a>]</span>, MachineLearning <span class="citation">[<a href="0400-references.html#ref-hothorn_website_2019">68</a>]</span> CRAN Task Views <span class="citation">[<a href="0400-references.html#ref-thecomprehensiverarchivenetwork_website_nda">69</a>]</span>. As mentioned in the introduction we will only cover common statistical approaches used in metabolomics. Areas such as time-series analysis, clustering methods, machine learning and visualisation of high-dimensional data were dealt with in various books and literature reviews <span class="citation">[<a href="0400-references.html#ref-bishop_2006">70</a>–<a href="0400-references.html#ref-zaslavsky_2016">78</a>]</span>.</p>
<p>With regard to statistical analyses in untargeted metabolomics, two strategies can be differentiated that necessitate the use of different methods. The first strategy “metabolite profiling” is performed by most untargeted metabolomics studies. Here, a bottom-up approach is taken where sets or classes of pre-defined metabolites are studied usually in different phenotypes of the same biological species and differences in metabolites are usually related to more coarse functional or biological levels (e.g. to phenotype or to control vs. treatment in biomedical studies) <span class="citation">[<a href="0400-references.html#ref-hall_2011">79</a>]</span>. Exploratory data analysis, univariate methods, hierarchical clustering (HCA), Principal Component Analysis (PCA) and Multi-Dimensional Scaling (MDS) like methods are very common in metabolite profiling approaches. Feature/variable selection is performed to find only the most significant metabolite candidates that explain the underlying research question, usually using univariate methods to target only specific metabolites that are interesting to the research question of the study <span class="citation">[<a href="0400-references.html#ref-cai_2017">80</a>–<a href="0400-references.html#ref-wen_2017">83</a>]</span>.</p>
<p>The second strategy “metabolite fingerprinting” is commonly used in biomedicine, environmental metabolomics and eco-metabolomics to find metabolite patterns across metabolite profiles. Here, metabolites are characterised without necessarily identifying them and characterisation usually occurs from spatiotemporally coarser scales to intrinsic scales within biological species <span class="citation">[<a href="0400-references.html#ref-peters_2018">84</a>]</span>. Multivariate statistical methods are used that require reduction of high-dimensional data and, thus, ordination methods are commonly applied like (Orthogonal) Partial Least Squares regression (sometimes also coupled to Discriminant Analysis) ((O)PLS(-DA)), (Linear) Discriminant Analysis ((L)DA), and (Canonical) Correspondence Analysis ((C)CA) that allow to relate sets of explanatory variables containing species traits or environmental properties (such as soil type, plant height, smoker/non-smoker, gender, etc.) to the metabolite feature matrix <span class="citation">[<a href="0400-references.html#ref-shaw_2003">77</a>,<a href="0400-references.html#ref-gromski_2014">85</a>,<a href="0400-references.html#ref-legendre_2012">86</a>]</span>. Other machine learning methods like Random Forests (RF), Support Vector Machines (SVM) and Neural Networks (NN or ANN) are also applicable <span class="citation">[<a href="0400-references.html#ref-clarke_2009">87</a>]</span>. Lately, untargeted metabolomics data is related to other ‘omics using network analysis or Procrustes analysis to visualise (dis)similarities between two or more ‘omics data sets <span class="citation">[<a href="0400-references.html#ref-feng_2018">88</a>–<a href="0400-references.html#ref-wei_2018">91</a>]</span>.</p>
<p>Extracting a restricted list of features which still provide a high prediction performance (i.e., a molecular signature) is critical for biomarker validation and clinical diagnostic. Several strategies have been described for feature selection <span class="citation">[<a href="0400-references.html#ref-degenhardt_2019">92</a>,<a href="0400-references.html#ref-saeys_2007">93</a>]</span> (e.g., wrapper approaches such as Recursive Feature Elimination, Genetic Algorithms, or sparse models such as Lasso, Elastic Net, or sparse PLS). Such techniques are implemented in R packages, which also provide detailed comparisons on real datasets in terms of the stability and the size of the selected signature, the prediction performance of the final model, and the computation time <span class="citation">[<a href="0400-references.html#ref-determanjr_2014">94</a>–<a href="0400-references.html#ref-wehrens_2012">97</a>]</span>.</p>
<p>A great number of packages is available to perform statistics on metabolomics datasets. Some of them focus on performing a number of specific tasks, such as sample size estimation, batch normalization, exploratory data analysis, univariate hypothesis testing, multivariate modeling and Omics data integration. Others, listed in the section ‘Multi workflow steps’ in Table 9, adopt a more comprehensive approach, providing statistics toolbox that cover different methods and functionalities.</p>
<p>muma is a package designed to be compatible with MS and NMR generated data. The package mainly focuses on performing statistics. It does not contain functions for data extraction and the user has to provide values arranged in a <em>data.frame</em> format. The pre-processing is limited to missing value imputation, noise filtering, variable scaling and normalization. The package also provides tools for outlier detection, univariate and multivariate analysis. Notably, the package offers a script for Statistical TOtal Correlation SpetroscopY (STOCSY) on NMR data.</p>
<p>MOFA proposes tools for the integration of data coming from different omics disciplines (multi-omics). Using factor analysis it allows to calculate hidden factors that capture the biological sample variation across multi-omics datasets, thus allowing marker discovery. MOFA also provide various tools for the visualization of results. IntLIM also supports integration of other omics datasets with metabolomics data by leveraging linear modeling to identify gene-metabolite pairs whose relationship differs from one phenotype to another (e.g. positive correlation in one phenotype, negative or no correlation in another). IntLIM includes a user-friendly web interface to perform data quality control of input data, identification of phenotype-dependent gene-metabolite pairs, and interactive visualization of results. This tool is particularly useful for integrating transcriptomic and metabolomic or other omics data by generating novel hypothesis in a data-driven manner.</p>
<p>MetaboDiff is presented as an entry level, user friendly package for differential metabolomics analysis. The information contained in the input data (metabolomics measurements and metadata) are stored in S4 objects which are used for the downstream processing. The pre-processing consists of missing value imputation, outlier removal and data normalization, while the data analysis part offers a variety of statistical methods including tools to explore how metabolites relate to each other in sub-pathways.</p>
<p>MetaboAnalystR is a toolbox built over several R packages and contains more than 500 functions organised in eleven modules. The package was created to overcome the limitations of the homonymous web application, such as the possibility of creating flexible customized workflows (including xcms interoperability) and the capacity of dealing with large data sets. MetaboAnalystR functionalities cover a wide range of tools: exploratory statistical analysis, biomarker analysis, power analysis, biomarker meta-analysis, functional enrichment analysis, pathway and joint pathway analysis. Through an implementation of the <em>mummichog</em> algorithm <span class="citation">[<a href="0400-references.html#ref-li_2013">98</a>]</span>, MetaboAnalystR also allows to infer pathways for from user-generated <em>m/z</em> peak-lists. Using the MetaboAnalyst knowledgebase, MetaboAnalystR provides access to metabolite set libraries, compound libraries and pathway libraries.</p>
<div id="htmlwidget-8686c5d4cc1a48934fb9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8686c5d4cc1a48934fb9">{"x":{"filter":"none","extensions":["RowGroup"],"caption":"<caption>Table 6: R packages for statistical analysis of metabolomics data.<\/caption>","fillContainer":false,"data":[["Sample size","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Normalization","Exploratory Data Analysis","Exploratory Data Analysis","Exploratory Data Analysis","Univariate hypothesis testing","Univariate hypothesis testing","Univariate hypothesis testing","Univariate hypothesis testing","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Multivariate modeling and feature selection","Omics Data integration","Omics Data integration","Omics Data integration","Omics Data integration","Omics Data integration","Omics Data integration","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Missing value imputation","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Multiple workflow steps","Omics Data integration","Omics Data integration"],["Estimate sample sizes for metabolomics experiments, (NMR and targeted approaches supported).","Cross-contribution robust multiple standard normalization. Normalization using internal standards.","Within and between batch correction of LC-MS metabolomics data using either QC samples or all samples.","Normalisation for low concentration metabolites. Mixed model with simultaneous estimation of a correlation matrix.","A collection of data distribution normalization methods.","Functions for drift removal and data normalisation based on: component correction, median fold change, ComBat or common PCA (CPCA).","Normalisation using a singular value decomposition.","Normalization based on RUV-random [164].","SVR-based normalization and integration for large-scale metabolomics data.","Drift correction using QC samples or all study samples.","Signal and Batch Correction for Mass Spectrometry","Multiple fitting models to correct intra- and inter-batch effects.","Collection of functions designed to implement, assess, and choose a suitable normalization method for a given metabolomics study.","A large number of methods available for PCA.","Chemometric analysis of NMR, IR or Raman spectroscopy data. It includes functions for spectral visualisation, peak alignment, HCA, PCA and model-based clustering.","Joint analysis of MS and MS2 data, where hierarchical cluster analysis is applied to MS2 data to annotate metabolite families and principal component analysis is applied to MS data to discover regulated metabolite families.","Many methods for corrections for multiple testing.","Estimate tail area-based false discovery rates (FDR) as well as local false discovery rates (fdr) for a variety of null models (p-values, z-scores, correlation coefficients, t-scores).","GUI for statistical analysis using linear mixed models to normalize data and ANOVA to test for treatment effects.","Derives stable estimates of the metabolome-wide significance level within a univariate approach based on a permutation procedure which effectively controls the maximum overall type I error rate at the α level.","Various multivariate methods to analyze metabolomics datasets. Main methods include PCA, Partial Least Squares regression (PLS), and extensions to the PLS like PLS Discriminant Analysis PLS-DA and the orthogonal variants OPLS(-DA).","Package for performing Partial Least Squares regression (PLS).","PPCA, PPCCA, MPPCA.","General framework for building regression and classification models.","ASCA, figure of merit (FoM), PCA, Goeman’s global test for metabolomic pathways (Q-stat), Penalized Jacobian method (for calculating network connections), Time-lagged correlation method and zero slopes method. It also includes centering and scaling functions.","RF for the construction, optimization and validation of classification models with the aim of identifying biomarkers. Also normalization, scaling, PCA, MDS.","PLS-DA, RF, SVM, GBM, GLMNET, PAM.","Recursive feature elimination approach that selects features which significantly contribute to the performance of PLS-DA, Random Forest or SVM classifiers.","Find Biomarkers in two class discrimination problems with variable selection methods provided for several classification metods (lasso/elastic net, PC-LDA, PLS-DA, and several t-tests).","Unsupervised feature extraction specifically designed for analysing noisy and high-dimensional datasets.","Non-parametric method for identifying differentially expressed features based on the estimated percentage of false predictions.","Fits multi-way component models via alternating least squares algorithms with optional constraints: orthogonal, non-negative, unimodal, monotonic, periodic, smooth, or structure. Fit models include InDScal, PARAFAC, PARAFAC2, SCA, Tucker.","Decompose a tensor of any order, as a generalisation of SVD also supporting non-identity metrics and penalisations. 2-way SVD is also available. Also includes PCAn (Tucker-n) and PARAFAC/CANDECOMP.","Fits multi-way component models via alternating least squares algorithms with optional constraints. Fit models include Individual Differences Scaling, Multiway Covariates Regression, PARAFAC (1 and 2), SCA, and Tucker Factor Analysis.","Performs variable selection in a multivariate linear model by estimating the covariance matrix of the residuals then use it to remove the dependence that may exist among the responses and eventually performs variable selection by using the Lasso criterion.","Performs the O2PLS data integration method for two datasets yielding joint and data-specific parts for each dataset.","Contains ordination methods such as ReDundancy Analysis (RDA), (Canonical or Detrended) Correspondence Analysis (CCA, DCA for binary explanatory variables), (Non-metric) Multi-Dimensional Scaling ((N)MDS) and other univariate and multivariate methods. Originally developed for vegetation ecologists, many functions are also applicable to metabolomics.","Linear and non-linear Discriminant Analysis methods (e.g. LDA), stepwise selection and classification methods useful for feature selection.","Variable selection methods for PLS, including significance multivariate correlation (SMC), selectivity ratio (SR), variable importance in projections (VIP), loading weights (LW), and regression coefficients (RC). It contains also some other modelling methods.","Predictive multivariate modelling using PLS and Random Forest Data. Repeated double cross unbiased validation and variable selection.","Biomarker validation for predicting survival. Cross validation methods to validate and select biomarkers when the outcome of interest is survival.","Pre-treatment, classification, feature selection and correlation analyses of metabolomics data.","Components search, optimal model components number search, optimal model validity test by permutation tests, observed values evaluation of optimal model parameters and predicted categories, bootstrap values evaluation of optimal model parameters and predicted cross-validated categories.","Robust identification of time intervals are significantly different between groups.","Multiple co-inertia analysis of omics datasets (MCIA) is a multivariate approach for visualization and integration of multi-omics datasets. The MCIA method is not dependent on feature annotation therefore can extract important features even when there are not present across all datasets.","STATegRa combines information in multiple omics datasets to evaluate the reproducibility among samples and across experimental condition using component analysis (omicsNPC implements the NonParametric Combination) and clustering.","Statistical framework supporting many different types of multivariate analyses (e.g. PCA, PLS, CCA, PLS-DA, etc.).","STatistics in R Using Class Templates - Classes for building statistical workflows using methods, models and validation objects. Provides mechanism for STATO integration.","Multi-omics base clases integrable with commonly used R Bioconductor objects for omics data; container that holds omics results.","Identifies analyte-analyte (e.g. gene-metabolite) pairs whose relationship differs by phenotype (e.g. positive correlation in one phenotype, negative or no correlation in another). The software is also accessible as a user-friendly interface at intlim.bmi.osumc.edu.","Mixture-model for accounting for data missingness'.","Kernel-Based Metabolite Differential Analysis provides a kernel-based score test to cluster metabolites between treatment groups, in order to handle missing values.","Visualization and imputation of missing values. VIM provides methods for the evaluation and visualization of the type and patterns of missing data. The included imputation approaches are kNN, Hot-Deck, iterative robust model-based imputation (IRMI), fast matching/imputation based on categorical variables and regression imputation.","Graphical user interface for VIM.","kNN based imputation for microarray data.","Bootstrap based algorithm and diagnostics for fast and robust multiple imputation for cross sectional, time series or combined cross sectional and time series data.","Algorithms and diagnostics for the univariate imputation of time series data.","Methods for the Imputation of incomplete continuous or categorical datasets. missMDA allows missing data imputation using in categorial, continuous or mixed-type datasets using PCA, CA, a multiple correspondence analysis (MCA) model, a multiple factor analysis (MFA) model or factorial analysis for mixed data (FAMD).","Random forest based missing data imputation for mixed-type, nonparametric data. An out-of-bag (OOB) error estimate is used for model optimization.","Multivariate imputation by chained equations using fully conditional specifications (FCS) for categorical, continuous and binary datasets. It includes various diagnostic plots for the evaluation of the imputation quality.","Missing data imputation using an approximate Bayesian framework. Diagnostic algorithms are included to analyze the models, the assumptions of the imputation algorithm and the multiply imputed datasets.","Iterative Gibbs sampler based left-censored missing value imputation.","Missing value imputation, filtering, normalisation and averaging of technical replications.","HCA, Fold change analysis, heat maps, linear models (ordinary and empirical Bayes), PCA and volcano plots. Also log transformation, missing value replacement and methods for normalisation. Cross-contribution compensating multiple internal standard normalisation (ccmn) and remove unwanted variation (ruv2).","Data processing, normalization, statistical analysis, metabolite set enrichment analysis, metabolic pathway analysis, and biomarker analysis.","Pipeline for metabolomics data pre-processing, with particular focus on data representation using univariate and multivariate statistics. Built on already published functions.","Framework for multiomics experiments. Identifies sources of variability in the experiment and performs additional analysis (identification of subgroups, data imputation, outlier detection).","Performs entry-level differential analysis on metabolomics data.","STRUCT wrappers for filtering, normalisation, missing value imputation, glog transform, HCA, PCA, PLSDA, PLSR, t-test, fold-change, ANOVA, Mixed Effects, post-hoc tests","Data transformation, filtering of feature and/or samples and data normalization. Quality control processing, statistical analysis and visualization of MS data.","Quality control, signal drift and batch correction, transformation, univariate hypothesis testing.","Missing value filtering and imputation, zero value filtering, data normalization, data integration, data quality assessment, univariate statistical analysis, multivariate statistical analysis such as PCA and PLS-D and potential marker selection","Univariate analysis (linear model), PCA, clustered heatmap, and partial correlation network analysis. Based on classes from the Metabase package(Zhu 2019).","Outlier detection, PCA, drift correction,visualization, missing value imputation,classification.","Pre-processing, differential compound identification and grouping, traditional PK parameters calculation, multivariate statistical analysis, correlations, cluster analyses and resulting visualization.","Integration of omics data using multivariate methods such as PLS. Performs community detection and network analysis to allow visualization of positive or negative associations between different datasets generated using samples from the same individuals. Also available as ashinyapp (https://kuppal.shinyapps.io/xmwas).","Joint metabolic model-based analysis of metabolomics measurements and taxonomic composition from microbial communities."],["<a target=_blank href=https://cran.r-project.org/package=MetSizeR>MetSizeR<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/crmn/index.html>crmn<\/a>","<a target=_blank href=https://gitlab.com/CarlBrunius/batchCorr>batchCorr<\/a>","<a target=_blank href=https://sourceforge.net/projects/metabnorm/>Metabnorm<\/a>","<a target=_blank href=http://quantitativeproteomics.org/normalyzer>Normalizer<\/a>","<a target=_blank href=http://b2slab.upc.edu/software-and-downloads/intensity-drift-correction/>intCor<\/a>","<a target=_blank href=https://sourceforge.net/projects/eigenms/>EigenMS<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/MetNorm/index.html>MetNorm<\/a>","<a target=_blank href=http://www.metabolomics-shanghai.org/softwaredetail.php?id=39>MetNormalizer<\/a>","<a target=_blank href=https://github.com/rwehrens/BatchCorrMetabolomics>BatchCorrMetabolomics<\/a>","<a target=_blank href=https://github.com/computational-metabolomics/sbcms>SBCMS<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetaboQC>MetaboQC<\/a>","<a target=_blank href=https://cran.r-project.org/package=NormalizeMets>NormalizeMets<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/pcaMethods.html>pcaMethods<\/a>","<a target=_blank href=https://cran.r-project.org/package=ChemoSpec>ChemoSpec<\/a>","<a target=_blank href=https://github.com/ipb-halle/MetFamily>MetFamily<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/multtest.html>multtest<\/a>","<a target=_blank href=http://strimmerlab.org/software/fdrtool/>fdrtool<\/a>","<a target=_blank href=http://metabr.r-forge.r-project.org/>MetabR<\/a>","<a target=_blank href=https://github.com/AlinaPeluso/MWSL>MWSL<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/ropls.html>ropls<\/a>","<a target=_blank href=https://cran.r-project.org/package=pls>pls<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetabolAnalyze>MetabolAnalyze<\/a>","<a target=_blank href=https://cran.r-project.org/package=caret>caret<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetStaT>MetStaT<\/a>","<a target=_blank href=https://cran.r-project.org/package=RFmarkerDetector>RFmarkerDetector<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/OmicsMarkeR.html>OmicsMarkeR<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/biosigner.html>biosigner<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/BioMark/index.html>BioMark<\/a>","<a target=_blank href=https://cran.r-project.org/package=KODAMA>KODAMA<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.RankProd>RankProd<\/a>","<a target=_blank href=https://cran.r-project.org/package=multiway>multiway<\/a>","<a target=_blank href=https://cran.r-project.org/package=PTAk>PTAk<\/a>","<a target=_blank href=https://cran.r-project.org/package=ThreeWay>ThreeWay<\/a>","<a target=_blank href=https://cran.r-project.org/package=MultiVarSel>MultiVarSel<\/a>","<a target=_blank href=https://cran.r-project.org/package=OmicsPLS>OmicsPLS<\/a>","<a target=_blank href=https://cran.r-project.org/package=vegan>vegan<\/a>","<a target=_blank href=https://cran.r-project.org/package=klaR>klaR<\/a>","<a target=_blank href=https://cran.r-project.org/package=plsVarSel>plsVarSel<\/a>","<a target=_blank href=https://gitlab.com/CarlBrunius/MUVR>MUVR<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetabolicSurv>MetabolicSurv<\/a>","<a target=_blank href=https://github.com/jasenfinch/metabolyseR>metabolyseR<\/a>","<a target=_blank href=https://cran.r-project.org/package=packMBPLSDA>packMBPLSDA<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.OmicsLonDA>OmicsLonDA<\/a>","<a target=_blank href=http://bioconductor.org/packages/release/bioc/html/omicade4.html>omicade4<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.STATegRa>STATegRa<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/mixOmics/index.html>mixOmics<\/a>","<a target=_blank href=http://www.github.com/computational-metabolomics/struct>STRUCT<\/a>","<a target=_blank href=https://bioconductor.org/packages/release/bioc/html/MultiDataSet.html>MultiDataSet<\/a>","<a target=_blank href=https://github.com/mathelab/IntLIM>IntLIM<\/a>","<a target=_blank href=https://doi.org/doi:10.18129/B9.bioc.metabomxtr>metabomxtr<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/KMDA/>KMDA<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/VIM/index.html>VIM<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/VIMGUI/VIMGUI.pdf>VIMGUI<\/a>","<a target=_blank href=https://www.bioconductor.org/packages//2.10/bioc/html/impute.html>impute<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/Amelia/Amelia.pdf>Amelia<\/a>","<a target=_blank href=https://cran.r-project.org/package=imputeTS>imputeTS<\/a>","<a target=_blank href=https://cran.r-project.org/package=missMDA>missMDA<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/missForest/missForest.pdf>missForest<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/mice/mice.pdf>mice<\/a>","<a target=_blank href=https://cran.r-project.org/package=mi>mi<\/a>","<a target=_blank href=https://github.com/WandeRum/GSimp>GSimp<\/a>","<a target=_blank href=https://sourceforge.net/projects/msprep/>MSPrep<\/a>","<a target=_blank href=https://cran.r-project.org/package=metabolomics>metabolomics<\/a>","<a target=_blank href=https://github.com/xia-lab/MetaboAnalystR>MetaboAnalystR<\/a>","<a target=_blank href=https://cran.r-project.org/web/packages/muma/index.html>muma<\/a>","<a target=_blank href=https://www.bioconductor.org/packages/release/bioc/html/MOFA.html>MOFA<\/a>","<a target=_blank href=https://github.com/andreasmock/MetaboDiff>MetaboDiff<\/a>","<a target=_blank href=https://github.com/computational-metabolomics/structToolbox>STRUCTToolbox<\/a>","<a target=_blank href=https://github.com/pmartR/pmartR>pmartR<\/a>","<a target=_blank href=https://github.com/ethevenot/phenomis>phenomis<\/a>","<a target=_blank href=https://github.com/MetabLAB/MetCleaning>MetCleaning<\/a>","<a target=_blank href=https://github.com/zhuchcn/ShinyMetabase>ShinyMetabase<\/a>","<a target=_blank href=https://cran.r-project.org/package=MetabolomicsBasics>MetabolomicsBasics<\/a>","<a target=_blank href=https://cran.r-project.org/package=polyPK>polyPK<\/a>","<a target=_blank href=https://github.com/kuppal2/xMWAS>xMWAS<\/a>","<a target=_blank href=https://github.com/borenstein-lab/MIMOSA/tree/master/mimosa>MIMOSA<\/a>"],["CRAN","CRAN","GitLab","SF",null,null,"SF","CRAN","GitHub","GitHub","GitHub","CRAN","CRAN","BioC","BioC","GitHub","BioC","CRAN","RF","GitHub","BioC","CRAN","CRAN","CRAN","CRAN","CRAN","BioC","BioC","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","GitLab","CRAN","GitHub","CRAN","BioC","BioC","BioC","CRAN","GitHub","BioC","GitHub","BioC","CRAN","CRAN","CRAN","BioC","CRAN","CRAN","CRAN","CRAN","CRAN","CRAN","GitHub","SF","CRAN","GitHub","GitHub","BioC","GitHub","GitHub","GitHub","GitHub","GitHub","GitHub","CRAN","CRAN","GitHub","GitHub"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Section<\/th>\n      <th>Functionalities<\/th>\n      <th>Package<\/th>\n      <th>Repo<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"rowGroup":{"dataSrc":0},"ordering":false,"columnDefs":[{"visible":false,"targets":0}],"pageLength":-1,"dom":"ft","order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[-1,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>

            </section>

          </div>
        </div>
      </div>
<a href="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rformassspectrometry/metaRbolomics-book/edit/master/rmd/02-R_packages_for_metabolomics/0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["The_MetaRbolomics_book.pdf", "The_MetaRbolomics_book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
