<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Appendix 2: metaRbolomics dependencies network | The MetaRbolomics book</title>
  <meta name="description" content="Appendix 2: metaRbolomics dependencies network | The MetaRbolomics book" />
  <meta name="generator" content="bookdown 0.13.2 and GitBook 2.6.7" />

  <meta property="og:title" content="Appendix 2: metaRbolomics dependencies network | The MetaRbolomics book" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Appendix 2: metaRbolomics dependencies network | The MetaRbolomics book" />
  
  
  

<meta name="author" content="Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"/>

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.9/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/dt-ext-rowgroup-1.10.19/css/rowGroup.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-rowgroup-1.10.19/js/dataTables.rowGroup.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/vis-4.20.1/vis.css" rel="stylesheet" />
<script src="libs/vis-4.20.1/vis.min.js"></script>
<script src="libs/visNetwork-binding-2.0.8/visNetwork.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The MetaRbolomics book</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to"><i class="fa fa-check"></i>How to …</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#this-book-in-other-formats"><i class="fa fa-check"></i>This book in other formats</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="0100-introduction.html"><a href="0100-introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="0101-introduction-Metabolomics_data_processing_and_analysis.html"><a href="0101-introduction-Metabolomics_data_processing_and_analysis.html"><i class="fa fa-check"></i><b>1.1</b> Metabolomics data processing and analysis</a></li>
<li class="chapter" data-level="1.2" data-path="0102-introduction-The_R_package_landscape.html"><a href="0102-introduction-The_R_package_landscape.html"><i class="fa fa-check"></i><b>1.2</b> The R package landscape</a></li>
<li class="chapter" data-level="1.3" data-path="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><a href="0103-introduction-Dependences_and_connectivity_of_metabolomics_packages.html"><i class="fa fa-check"></i><b>1.3</b> Dependences and connectivity of metabolomics packages</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="0200-R_packages_for_metabolomics.html"><a href="0200-R_packages_for_metabolomics.html"><i class="fa fa-check"></i><b>2</b> R-packages for metabolomics</a><ul>
<li class="chapter" data-level="2.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.1</b> Mass spectrometry data handling and (pre-)processing</a><ul>
<li class="chapter" data-level="2.1.1" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#profile-mode-and-centroided-data"><i class="fa fa-check"></i><b>2.1.1</b> Profile mode and centroided data</a></li>
<li class="chapter" data-level="2.1.2" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#direct-infusion-mass-spectrometry-data"><i class="fa fa-check"></i><b>2.1.2</b> Direct infusion mass spectrometry data</a></li>
<li class="chapter" data-level="2.1.3" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#hyphenated-ms-and-non-targeted-data"><i class="fa fa-check"></i><b>2.1.3</b> Hyphenated MS and non-targeted data</a></li>
<li class="chapter" data-level="2.1.4" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#targeted-data-and-alternative-representations-of-data"><i class="fa fa-check"></i><b>2.1.4</b> Targeted data and alternative representations of data</a></li>
<li class="chapter" data-level="2.1.5" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#additional-dimensionality"><i class="fa fa-check"></i><b>2.1.5</b> Additional dimensionality</a></li>
<li class="chapter" data-level="2.1.6" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#structuring-data-and-metadata"><i class="fa fa-check"></i><b>2.1.6</b> Structuring data and metadata</a></li>
<li class="chapter" data-level="2.1.7" data-path="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html"><a href="0201-R_packages_for_metabolomics-Mass_spectrometry_data_handling_and_pre_processing.html#ion-species-grouping-and-annotation"><i class="fa fa-check"></i><b>2.1.7</b> Ion species grouping and annotation</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><i class="fa fa-check"></i><b>2.2</b> Metabolite identification with MS/MS data</a><ul>
<li class="chapter" data-level="2.2.1" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#msms-data-handling-spectral-matching-and-clustering"><i class="fa fa-check"></i><b>2.2.1</b> MS/MS data handling, spectral matching and clustering</a></li>
<li class="chapter" data-level="2.2.2" data-path="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html"><a href="0202-R_packages_for_metabolomics-Metabolite_identification_with_MS_MS_data.html#reading-of-spectral-databases"><i class="fa fa-check"></i><b>2.2.2</b> Reading of spectral databases</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><a href="0203-R_packages_for_metabolomics-NMR_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.3</b> NMR data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.4" data-path="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><a href="0204-R_packages_for_metabolomics-UV_data_handling_and_pre_processing.html"><i class="fa fa-check"></i><b>2.4</b> UV data handling and (pre-)processing</a></li>
<li class="chapter" data-level="2.5" data-path="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><a href="0205-R_packages_for_metabolomics-Statistical_analysis_of_metabolomics_data.html"><i class="fa fa-check"></i><b>2.5</b> Statistical analysis of metabolomics data</a></li>
<li class="chapter" data-level="2.6" data-path="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><a href="0206-R_packages_for_metabolomics-Handling_of_molecule_structures_and_chemical_structure_databases.html"><i class="fa fa-check"></i><b>2.6</b> Handling of molecule structures and chemical structure databases</a></li>
<li class="chapter" data-level="2.7" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><i class="fa fa-check"></i><b>2.7</b> Network analysis and biochemical pathways</a><ul>
<li class="chapter" data-level="2.7.1" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#network-infrastructure-and-analysis"><i class="fa fa-check"></i><b>2.7.1</b> Network infrastructure and analysis</a></li>
<li class="chapter" data-level="2.7.2" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#metabolite-annotation"><i class="fa fa-check"></i><b>2.7.2</b> Metabolite annotation</a></li>
<li class="chapter" data-level="2.7.3" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#generation-of-metabolic-networks"><i class="fa fa-check"></i><b>2.7.3</b> Generation of metabolic networks</a></li>
<li class="chapter" data-level="2.7.4" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-analysis"><i class="fa fa-check"></i><b>2.7.4</b> Pathway analysis</a></li>
<li class="chapter" data-level="2.7.5" data-path="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html"><a href="0207-R_packages_for_metabolomics-Network_analysis_and_biochemical_pathways.html#pathway-resources-and-interfaces"><i class="fa fa-check"></i><b>2.7.5</b> Pathway resources and interfaces</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><a href="0208-R_packages_for_metabolomics-Multifunctional_workflows.html"><i class="fa fa-check"></i><b>2.8</b> Multifunctional workflows</a></li>
<li class="chapter" data-level="2.9" data-path="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><a href="0209-R_packages_for_metabolomics-User_interfaces_and_workflow_management_systems.html"><i class="fa fa-check"></i><b>2.9</b> User interfaces and workflow management systems</a></li>
<li class="chapter" data-level="2.10" data-path="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><a href="0210-R_packages_for_metabolomics-Metabolomics_data_sets.html"><i class="fa fa-check"></i><b>2.10</b> Metabolomics data sets</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="0300-conclusion.html"><a href="0300-conclusion.html"><i class="fa fa-check"></i><b>3</b> Conclusions</a></li>
<li class="chapter" data-level="" data-path="0400-references.html"><a href="0400-references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="" data-path="9900-Appendices.html"><a href="9900-Appendices.html"><i class="fa fa-check"></i>Appendices</a><ul>
<li class="chapter" data-level="" data-path="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><a href="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html"><i class="fa fa-check"></i>Appendix 1: The MSP File Format and package support</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><i class="fa fa-check"></i>Appendix 2: metaRbolomics dependencies network</a><ul>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#libraries-and-settings"><i class="fa fa-check"></i>Libraries and settings</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#read-package-names-from-our-table"><i class="fa fa-check"></i>Read package names from our table</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#get-reverse-dependencies"><i class="fa fa-check"></i>Get reverse dependencies</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#building-dependency-network"><i class="fa fa-check"></i>Building dependency network</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#save-network-plot"><i class="fa fa-check"></i>Save network plot</a></li>
<li class="chapter" data-level="" data-path="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html"><a href="9902-Appendices-Appendix_2_metaRbolomics_dependencies.html#notes"><i class="fa fa-check"></i>Notes</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The MetaRbolomics book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div style="page-break-before: always;" />

</div>
<div id="appendix-2-metarbolomics-dependencies-network" class="section level2 unnumbered">
<h2>Appendix 2: metaRbolomics dependencies network</h2>
<div id="libraries-and-settings" class="section level3 unnumbered">
<h3>Libraries and settings</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">options</span>(<span class="st">&quot;repos&quot;</span> =<span class="st"> </span><span class="kw">list</span>(<span class="dt">CRAN=</span><span class="st">&quot;http://cran.rstudio.com/&quot;</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(devtools)   <span class="co"># for revdp()</span></a></code></pre></div>
<pre><code>## Loading required package: usethis</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw">library</span>(igraph)     <span class="co"># for graph_from_edgelist/( and simplify() )</span></a></code></pre></div>
<pre><code>## 
## Attaching package: &#39;igraph&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:purrr&#39;:
## 
##     compose, simplify</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     as_data_frame, groups, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     decompose, spectrum</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     union</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">library</span>(visNetwork) <span class="co"># for visNetwork() and friends</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw">library</span>(networkD3)  <span class="co"># for saveNetwork()</span></a></code></pre></div>
<pre><code>## 
## Attaching package: &#39;networkD3&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:DT&#39;:
## 
##     JS</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">library</span>(chromote)   <span class="co"># for default_chromote_object()</span></a></code></pre></div>
<pre><code>## Registered S3 method overwritten by &#39;fastmap&#39;:
##   method            from 
##   print.key_missing shiny</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw">library</span>(webshot2)   <span class="co"># for webshot()</span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="kw">library</span>(png)        <span class="co"># For displaying an image</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="kw">library</span>(purrr)</a>
<a class="sourceLine" id="cb14-5" data-line-number="5"></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="kw">source</span>(<span class="st">&quot;scripts/revDepNetHelper.R&quot;</span>)</a>
<a class="sourceLine" id="cb14-7" data-line-number="7"></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="kw">set_default_chromote_object</span>(Chromote<span class="op">$</span><span class="kw">new</span>(<span class="dt">browser =</span> Chrome<span class="op">$</span><span class="kw">new</span>(<span class="dt">args =</span> <span class="st">&quot;--no-sandbox&quot;</span>)))</a></code></pre></div>
</div>
<div id="read-package-names-from-our-table" class="section level3 unnumbered">
<h3>Read package names from our table</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">reviewTables &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;public/data/AllMetaRbolomicsTables.csv&quot;</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">reviewPkgs &lt;-<span class="st"> </span>reviewTables[,<span class="st">&quot;Package&quot;</span>]</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"></a>
<a class="sourceLine" id="cb15-4" data-line-number="4">pkgs &lt;-<span class="st"> </span>reviewPkgs</a></code></pre></div>
</div>
<div id="get-reverse-dependencies" class="section level3 unnumbered">
<h3>Get reverse dependencies</h3>
<div id="for-cran-and-bioc-packages" class="section level4 unnumbered">
<h4>For CRAN and BioC packages</h4>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">el &lt;-<span class="st"> </span><span class="kw">sapply</span>(pkgs, <span class="cf">function</span>(pkg) {</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">  rd &lt;-<span class="st"> </span><span class="kw">revdep</span>(pkg, <span class="dt">dependencies =</span> <span class="kw">c</span>(<span class="st">&quot;Depends&quot;</span>, <span class="st">&quot;Imports&quot;</span>, <span class="st">&quot;LinkingTo&quot;</span>), </a>
<a class="sourceLine" id="cb16-3" data-line-number="3">               <span class="dt">recursive =</span> <span class="ot">FALSE</span>, <span class="dt">ignore =</span> <span class="ot">NULL</span>, <span class="dt">bioconductor =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">  <span class="kw">as.matrix</span>(<span class="kw">cbind</span>(<span class="dt">Package=</span><span class="kw">rep</span>(pkg, <span class="dt">length.out=</span><span class="kw">length</span>(rd)), <span class="dt">ReverseDep=</span>rd))</a>
<a class="sourceLine" id="cb16-5" data-line-number="5">})</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">el &lt;-<span class="st"> </span><span class="kw">do.call</span>(rbind, el)</a></code></pre></div>
</div>
<div id="for-github-and-gitlab" class="section level4 unnumbered">
<h4>For GitHub and GitLab</h4>
<p>The above <code>devtools::revdep</code> cannot read from GitHub/GitLab repositories.
We have a helper function that downloads and parses the DESCRIPTION file from GitHub/GitLab.
Since we cannot get reverse dependencies directly for GitHub/GitLab packages, those packages they are only used as additional reverse depedencies for the CRAN/BioC packages.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">gitdeps_reverse &lt;-<span class="st"> </span>reviewTables <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"><span class="st">                      </span><span class="kw">mutate</span>(<span class="dt">dep_tree =</span> <span class="kw">map</span>(Code_link, get_git_deps)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="st">                      </span><span class="kw">pull</span>(dep_tree) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-4" data-line-number="4"><span class="st">                      </span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="st">                      </span><span class="kw">filter</span>(Dep <span class="op">%in%</span><span class="st"> </span>el[,<span class="st">&quot;Package&quot;</span>]) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="st">                      </span><span class="kw">rename</span>(<span class="dt">Package =</span> Dep, <span class="dt">ReverseDep =</span> Package) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="st">                      </span><span class="kw">as.matrix</span>()</a></code></pre></div>
<pre><code>## Loading required package: desc</code></pre>
<pre><code>## 
## Attaching package: &#39;desc&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     desc</code></pre>
<pre><code>## Warning in readLines(file): incomplete final line found on &#39;/tmp/
## RtmpAfVlkk/file3bbf6d1cd44a&#39;</code></pre>
<pre><code>## Warning in readLines(file): incomplete final line found on &#39;/tmp/
## RtmpAfVlkk/file3bbfff5f76d&#39;</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1">el &lt;-<span class="st"> </span><span class="kw">rbind</span>(el, gitdeps_reverse)</a></code></pre></div>
</div>
</div>
<div id="building-dependency-network" class="section level3 unnumbered">
<h3>Building dependency network</h3>
<p>In total, we were analysing 297 packages.
For each package, this returns the set of packages in CRAN or BioC that depend on, import from or link to the package (i.e., its direct reverse dependencies) using the <code>devtools::revdep()</code> function. A few packages with the highest number of reverse dependencies have been excluded, as they would dominate the visualisation.
It was not possible to detect reverse dependencies from other hosting places such as GitHub or GitLab.</p>
<p>From the total, 69 packages had at least one
such reverse dependency.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">## Remove packages with most reverse dependencies </a>
<a class="sourceLine" id="cb24-2" data-line-number="2">## which would dominate the network</a>
<a class="sourceLine" id="cb24-3" data-line-number="3"></a>
<a class="sourceLine" id="cb24-4" data-line-number="4">el &lt;-<span class="st"> </span>el[<span class="op">!</span><span class="st"> </span>el[,<span class="st">&quot;Package&quot;</span>] <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Rcpp&quot;</span>, <span class="st">&quot;igraph&quot;</span>, <span class="st">&quot;vegan&quot;</span>, <span class="st">&quot;caret&quot;</span>, <span class="st">&quot;rJava&quot;</span>, <span class="st">&quot;reticulate&quot;</span>), ]</a>
<a class="sourceLine" id="cb24-5" data-line-number="5"></a>
<a class="sourceLine" id="cb24-6" data-line-number="6">## Create graph, and simplify redundancy</a>
<a class="sourceLine" id="cb24-7" data-line-number="7">g &lt;-<span class="st"> </span><span class="kw">graph_from_edgelist</span>(el, <span class="dt">directed =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb24-8" data-line-number="8">g &lt;-<span class="st"> </span>igraph<span class="op">::</span><span class="kw">simplify</span>(g, <span class="dt">remove.multiple =</span> <span class="ot">TRUE</span>, <span class="dt">remove.loops =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb24-9" data-line-number="9"></a>
<a class="sourceLine" id="cb24-10" data-line-number="10"><span class="co"># get data and plot :</span></a>
<a class="sourceLine" id="cb24-11" data-line-number="11">data &lt;-<span class="st"> </span><span class="kw">toVisNetworkData</span>(g)</a>
<a class="sourceLine" id="cb24-12" data-line-number="12"></a>
<a class="sourceLine" id="cb24-13" data-line-number="13">data<span class="op">$</span>nodes &lt;-<span class="st"> </span><span class="kw">cbind</span>(data<span class="op">$</span>nodes, </a>
<a class="sourceLine" id="cb24-14" data-line-number="14">                    <span class="dt">font.size=</span><span class="dv">30</span>, </a>
<a class="sourceLine" id="cb24-15" data-line-number="15">                    <span class="dt">color.background =</span> <span class="kw">ifelse</span>(data<span class="op">$</span>nodes[,<span class="st">&quot;id&quot;</span>] <span class="op">%in%</span><span class="st"> </span>pkgs , </a>
<a class="sourceLine" id="cb24-16" data-line-number="16">                           <span class="kw">rgb</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">200</span>, <span class="dv">128</span>, <span class="dt">max =</span> <span class="dv">255</span>), </a>
<a class="sourceLine" id="cb24-17" data-line-number="17">                           <span class="kw">rgb</span>(<span class="dv">0</span>, <span class="dv">200</span>, <span class="dv">0</span>, <span class="dv">128</span>, <span class="dt">max =</span> <span class="dv">255</span>)))</a>
<a class="sourceLine" id="cb24-18" data-line-number="18"></a>
<a class="sourceLine" id="cb24-19" data-line-number="19">vn &lt;-<span class="st"> </span><span class="kw">visNetwork</span>(<span class="dt">nodes =</span> data<span class="op">$</span>nodes, </a>
<a class="sourceLine" id="cb24-20" data-line-number="20">                 <span class="dt">edges =</span> data<span class="op">$</span>edges,</a>
<a class="sourceLine" id="cb24-21" data-line-number="21">                 <span class="dt">width=</span><span class="dv">1000</span>, <span class="dt">height=</span><span class="dv">1000</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb24-22" data-line-number="22"><span class="st">  </span><span class="kw">visPhysics</span>(<span class="dt">timestep =</span> <span class="fl">0.3</span>,</a>
<a class="sourceLine" id="cb24-23" data-line-number="23">             <span class="dt">barnesHut =</span> <span class="kw">list</span>(<span class="dt">centralGravity=</span><span class="fl">0.35</span>,</a>
<a class="sourceLine" id="cb24-24" data-line-number="24">                              <span class="dt">springLength =</span> <span class="dv">95</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb24-25" data-line-number="25"><span class="st">  </span><span class="kw">visOptions</span>(<span class="dt">highlightNearest =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb24-26" data-line-number="26"></a>
<a class="sourceLine" id="cb24-27" data-line-number="27">vn</a></code></pre></div>
<div id="htmlwidget-df4ce0292d46d4fde99d" style="width:1000px;height:1000px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-df4ce0292d46d4fde99d">{"x":{"nodes":{"id":["mzR","adductomicsR","Autotuner","CluMSID","MSGFgui","MSnbase","MSnID","msPurity","Pbase","ProteomicsAnnotationHubData","RMassBank","SIMAT","topdownr","xcms","yamss","DAPAR","DEP","MetCirc","msmsEDA","msmsTests","MSstatsQC","PrInCE","pRoloc","pRolocGUI","proteoQC","qPLEXanalyzer","synapter","CAMERA","cosmiq","flagme","IPO","LOBSTAHS","MAIT","Metab","metaMS","ncGTW","proFIA","Risa","enviPick","MetaboList","IsoCorrectoR","IsoCorrectoRGUI","erah","baitmet","MassSpecWavelet","Rnmr1D","speaq","ptw","alsace","chipPCR","MODIS","PepsNMR","RGCxGC","RpeakChrom","rTANDEM","PGA","sapFinder","shinyTANDEM","Rdisop","HiResTEC","InterpretMSSpectrum","enviPat","nontarget","RAMClustR","OrgMassSpecR","ASICS","MWASTools","MetaboSignal","baseline","NADfinder","pems.utils","powdR","spectacles","spectralAnalysis","takos","crmn","NormalizeMets","pcaMethods","ADAPTS","CompGO","consensusDE","DeconRNASeq","DiffCorr","imputeLCMD","LOST","MEET","MetabolomicsBasics","missCompare","multiDimBio","OUTRIDER","PanVizGenerator","polyPK","polyRAD","santaR","scde","SomaticSignatures","multtest","a4Base","ABarray","aCGH","adSplit","ALDEx2","anota","anota2seq","apmsWAPP","BicARE","ChIPpeakAnno","cp4p","EMA","GExMap","hddplot","INCATome","iPAC","IsoGeneGUI","KCsmart","KnowSeq","LMGene","mAPKL","metabomxtr","MetaIntegrator","mutoss","nethet","nlcv","OCplus","PCS","phyloseq","PREDA","pRF","rain","REDseq","RTopper","SAGx","siggenes","singleCellTK","structSSI","TcGSA","webbioc","fdrtool","beam","cape","carSurv","CosW","crossmeta","DBGSA","DChIPRep","DMCHMM","DNLC","ERP","FitHiC","GeneCycle","GeneNet","GSRI","HDMT","HybridMTest","IHW","LANDD","les","meshr","netbenchmark","NetworkToolbox","nlnet","oposSOM","OVESEG","qgraph","rags2ridges","ragt2ridges","sda","SecKW","SimSeq","SinIW","st","TanB","TBEST","ropls","biosigner","lipidr","rqt","pls","apercu","autopls","bestglm","BioMark","BioMedR","chemmodlab","chemometrics","ChemometricsWithR","chillR","cocoreg","CSFA","DaMiRseq","dbstats","dmm","EasyABC","enpls","fdm2id","funtooNorm","groc","IntClust","LINselect","lmdme","lspls","mephas","MetabolicSurv","MetStaT","mixlm","MLInterfaces","monomvn","muma","obliqueRF","OHPL","plsRcox","plsVarSel","RcmdrPlugin.NMBU","regressoR","rminer","robCompositions","rrcovHD","RVAideMemoire","RVtests","spls","SQB","sRAP","statTarget","svapls","SVAPLSseq","tempoR","tsensembler","MetabolAnalyze","MetSizeR","RankProd","HTSanalyzeR","RNAither","sigQC","synlet","tRanslatome","multiway","staRdom","PTAk","ttTensor","ThreeWay","FPDclustering","GSparO","PCovR","rrcov3way","klaR","agricolae","BioSeqClass","CLUSTShiny","DeducerPlugInScaling","diceR","fingerPro","minque","MLDAShiny","TestDimorph","tiger","omicade4","omicRexposome","mixOmics","bootsPLS","compartmap","DepecheR","mixKernel","PCA4you","R2GUESS","sgPLS","YuGene","MultiDataSet","MEAL","VIM","analytics","destiny","sdcMicro","simPop","smartdata","VIMGUI","impute","AMARETTO","armada","CancerSubtypes","CGHcall","ChAMP","cola","doppelgangR","EGAD","ENmix","ePCR","FAMT","fastLiquidAssociation","FEM","genefu","genomation","iC10","maGUI","MetaPath","MethylMix","miRLAB","moduleColor","netboost","Pigengene","PMA","REMP","Rnits","samr","snpReady","swamp","TIN","WGCNA","xseq","Amelia","autovarCore","bmem","TestDataImputation","Zelig","zeligverse","imputeTS","EventDetectR","gimme","hpiR","imputeTestbench","psychNET","TrendSLR","missMDA","INSPIRE","missForest","bartMachine","KarsTS","obliqueRSF","mice","accelmissing","BaM","bootImpute","CALIBERrfimpute","dlookr","dynr","finalfit","genpathmox","HardyWeinberg","hmi","hot.deck","ImputeRobust","JWileymisc","logistf","MatchIt.mice","miceadds","micemd","miceMNAR","mipred","MRPC","PathSelectMP","Qtools","RBtest","RegularizedSCA","Replication","rexposome","roughrf","sociome","weights","weightTAPSPACK","mi","migui","sem","tidygraph","clustree","clustringr","fastnet","ggdag","ggraph","HCABrowser","particles","scholar","stminsights","statnet","ergMargins","RedeR","dc3net","PANR","RTN","transcriptogramer","diffcoexp","CEMiTool","CVE","DCD","DGCA","eclust","epihet","GOGANPA","MCbiclust","MetNet","miRSM","MODA","Patterns","proBatch","seq2pathway","tmod","disco","sybil","BacArena","CoRegFlux","exp2flux","g2f","GlobalFit","metaboGSE","RSeed","sybilccFBA","sybilcycleFreeFlux","sybilDynFBA","SPIA","EnrichmentBrowser","rBiopaxParser","AnnotationHubData","pwOmics","graphite","ICDS","mogsa","netgsa","PoTRA","ReactomePA","StarBioTrek","ToPASeq","NCIgraph","DEGraph","pathview","BioNetStat","debrowser","EGSEA","GDCRNATools","MAGeCKFlute","pathfindR","RNASeqR","SBGNview","TCGAbiolinksGUI","KEGGgraph","clipper","iCARH","KEGGlincs","NFP","ROntoTools","rsbml","BiGGR","PSICQUIC","RefNet","KEGGREST","ADAM","attract","CNEr","FELLA","gage","KEGGprofile","omu","PAPi","PathwaySplice","RnaSeqSampleSize","YAPSA","faahKO","rcdk","enviGCMS","pmd","rcellminer","Rcpi","RxnSim","SimuChemPC","ChemmineR","bioassayR","eiR","fmcsR","MetID","Rchemcpp","uCAREChemSuiteCLI","fingerprint","rpubchem","webchem","MRMConverteR","massFlowR","KPIC2","geoRge","IsotopicLabelling","peakPantheR","sRm","DecoMetDIA","TarMet","FIEmspro","credential","qcrms","ProbMetab","compMS2Miner","IPA","MetShot","MatchWeiz","MSnio","ReSOLUTION","Retip","SOAP-NMR","MetaboMate","rDolphin","MetFamily","metabolyseR","IntLIM","MetaboAnalystR","MetaboDiff","structToolbox","phenomis","MetCleaning","MetDNA","mzmatch.R","simExTargId","MetMSLine","specmine","patRoon","RMassScreening","METABOseek","MetaboShiny","amp","RChemMass","MetaDBparse","CompoundDb","xMWAS"],"label":["mzR","adductomicsR","Autotuner","CluMSID","MSGFgui","MSnbase","MSnID","msPurity","Pbase","ProteomicsAnnotationHubData","RMassBank","SIMAT","topdownr","xcms","yamss","DAPAR","DEP","MetCirc","msmsEDA","msmsTests","MSstatsQC","PrInCE","pRoloc","pRolocGUI","proteoQC","qPLEXanalyzer","synapter","CAMERA","cosmiq","flagme","IPO","LOBSTAHS","MAIT","Metab","metaMS","ncGTW","proFIA","Risa","enviPick","MetaboList","IsoCorrectoR","IsoCorrectoRGUI","erah","baitmet","MassSpecWavelet","Rnmr1D","speaq","ptw","alsace","chipPCR","MODIS","PepsNMR","RGCxGC","RpeakChrom","rTANDEM","PGA","sapFinder","shinyTANDEM","Rdisop","HiResTEC","InterpretMSSpectrum","enviPat","nontarget","RAMClustR","OrgMassSpecR","ASICS","MWASTools","MetaboSignal","baseline","NADfinder","pems.utils","powdR","spectacles","spectralAnalysis","takos","crmn","NormalizeMets","pcaMethods","ADAPTS","CompGO","consensusDE","DeconRNASeq","DiffCorr","imputeLCMD","LOST","MEET","MetabolomicsBasics","missCompare","multiDimBio","OUTRIDER","PanVizGenerator","polyPK","polyRAD","santaR","scde","SomaticSignatures","multtest","a4Base","ABarray","aCGH","adSplit","ALDEx2","anota","anota2seq","apmsWAPP","BicARE","ChIPpeakAnno","cp4p","EMA","GExMap","hddplot","INCATome","iPAC","IsoGeneGUI","KCsmart","KnowSeq","LMGene","mAPKL","metabomxtr","MetaIntegrator","mutoss","nethet","nlcv","OCplus","PCS","phyloseq","PREDA","pRF","rain","REDseq","RTopper","SAGx","siggenes","singleCellTK","structSSI","TcGSA","webbioc","fdrtool","beam","cape","carSurv","CosW","crossmeta","DBGSA","DChIPRep","DMCHMM","DNLC","ERP","FitHiC","GeneCycle","GeneNet","GSRI","HDMT","HybridMTest","IHW","LANDD","les","meshr","netbenchmark","NetworkToolbox","nlnet","oposSOM","OVESEG","qgraph","rags2ridges","ragt2ridges","sda","SecKW","SimSeq","SinIW","st","TanB","TBEST","ropls","biosigner","lipidr","rqt","pls","apercu","autopls","bestglm","BioMark","BioMedR","chemmodlab","chemometrics","ChemometricsWithR","chillR","cocoreg","CSFA","DaMiRseq","dbstats","dmm","EasyABC","enpls","fdm2id","funtooNorm","groc","IntClust","LINselect","lmdme","lspls","mephas","MetabolicSurv","MetStaT","mixlm","MLInterfaces","monomvn","muma","obliqueRF","OHPL","plsRcox","plsVarSel","RcmdrPlugin.NMBU","regressoR","rminer","robCompositions","rrcovHD","RVAideMemoire","RVtests","spls","SQB","sRAP","statTarget","svapls","SVAPLSseq","tempoR","tsensembler","MetabolAnalyze","MetSizeR","RankProd","HTSanalyzeR","RNAither","sigQC","synlet","tRanslatome","multiway","staRdom","PTAk","ttTensor","ThreeWay","FPDclustering","GSparO","PCovR","rrcov3way","klaR","agricolae","BioSeqClass","CLUSTShiny","DeducerPlugInScaling","diceR","fingerPro","minque","MLDAShiny","TestDimorph","tiger","omicade4","omicRexposome","mixOmics","bootsPLS","compartmap","DepecheR","mixKernel","PCA4you","R2GUESS","sgPLS","YuGene","MultiDataSet","MEAL","VIM","analytics","destiny","sdcMicro","simPop","smartdata","VIMGUI","impute","AMARETTO","armada","CancerSubtypes","CGHcall","ChAMP","cola","doppelgangR","EGAD","ENmix","ePCR","FAMT","fastLiquidAssociation","FEM","genefu","genomation","iC10","maGUI","MetaPath","MethylMix","miRLAB","moduleColor","netboost","Pigengene","PMA","REMP","Rnits","samr","snpReady","swamp","TIN","WGCNA","xseq","Amelia","autovarCore","bmem","TestDataImputation","Zelig","zeligverse","imputeTS","EventDetectR","gimme","hpiR","imputeTestbench","psychNET","TrendSLR","missMDA","INSPIRE","missForest","bartMachine","KarsTS","obliqueRSF","mice","accelmissing","BaM","bootImpute","CALIBERrfimpute","dlookr","dynr","finalfit","genpathmox","HardyWeinberg","hmi","hot.deck","ImputeRobust","JWileymisc","logistf","MatchIt.mice","miceadds","micemd","miceMNAR","mipred","MRPC","PathSelectMP","Qtools","RBtest","RegularizedSCA","Replication","rexposome","roughrf","sociome","weights","weightTAPSPACK","mi","migui","sem","tidygraph","clustree","clustringr","fastnet","ggdag","ggraph","HCABrowser","particles","scholar","stminsights","statnet","ergMargins","RedeR","dc3net","PANR","RTN","transcriptogramer","diffcoexp","CEMiTool","CVE","DCD","DGCA","eclust","epihet","GOGANPA","MCbiclust","MetNet","miRSM","MODA","Patterns","proBatch","seq2pathway","tmod","disco","sybil","BacArena","CoRegFlux","exp2flux","g2f","GlobalFit","metaboGSE","RSeed","sybilccFBA","sybilcycleFreeFlux","sybilDynFBA","SPIA","EnrichmentBrowser","rBiopaxParser","AnnotationHubData","pwOmics","graphite","ICDS","mogsa","netgsa","PoTRA","ReactomePA","StarBioTrek","ToPASeq","NCIgraph","DEGraph","pathview","BioNetStat","debrowser","EGSEA","GDCRNATools","MAGeCKFlute","pathfindR","RNASeqR","SBGNview","TCGAbiolinksGUI","KEGGgraph","clipper","iCARH","KEGGlincs","NFP","ROntoTools","rsbml","BiGGR","PSICQUIC","RefNet","KEGGREST","ADAM","attract","CNEr","FELLA","gage","KEGGprofile","omu","PAPi","PathwaySplice","RnaSeqSampleSize","YAPSA","faahKO","rcdk","enviGCMS","pmd","rcellminer","Rcpi","RxnSim","SimuChemPC","ChemmineR","bioassayR","eiR","fmcsR","MetID","Rchemcpp","uCAREChemSuiteCLI","fingerprint","rpubchem","webchem","MRMConverteR","massFlowR","KPIC2","geoRge","IsotopicLabelling","peakPantheR","sRm","DecoMetDIA","TarMet","FIEmspro","credential","qcrms","ProbMetab","compMS2Miner","IPA","MetShot","MatchWeiz","MSnio","ReSOLUTION","Retip","SOAP-NMR","MetaboMate","rDolphin","MetFamily","metabolyseR","IntLIM","MetaboAnalystR","MetaboDiff","structToolbox","phenomis","MetCleaning","MetDNA","mzmatch.R","simExTargId","MetMSLine","specmine","patRoon","RMassScreening","METABOseek","MetaboShiny","amp","RChemMass","MetaDBparse","CompoundDb","xMWAS"],"font.size":[30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30,30],"color.background":["#0000C880","#00C80080","#0000C880","#0000C880","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#0000C880","#0000C880","#0000C880","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#00C80080","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#00C80080","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880","#0000C880"]},"edges":{"from":["mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","mzR","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","MSnbase","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","xcms","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","CAMERA","enviPick","enviPick","enviPick","IsoCorrectoR","erah","MassSpecWavelet","MassSpecWavelet","MassSpecWavelet","MassSpecWavelet","MassSpecWavelet","speaq","speaq","speaq","speaq","ptw","ptw","ptw","ptw","ptw","ptw","ptw","ptw","ptw","ptw","PepsNMR","rTANDEM","rTANDEM","rTANDEM","rTANDEM","Rdisop","Rdisop","Rdisop","Rdisop","Rdisop","Rdisop","InterpretMSSpectrum","InterpretMSSpectrum","InterpretMSSpectrum","enviPat","enviPat","enviPat","enviPat","enviPat","enviPat","enviPat","enviPat","enviPat","OrgMassSpecR","OrgMassSpecR","MWASTools","baseline","baseline","baseline","baseline","baseline","baseline","baseline","baseline","crmn","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","pcaMethods","DiffCorr","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","multtest","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","fdrtool","ropls","ropls","ropls","ropls","ropls","ropls","ropls","ropls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","pls","plsVarSel","MetabolAnalyze","RankProd","RankProd","RankProd","RankProd","RankProd","RankProd","multiway","multiway","PTAk","ThreeWay","ThreeWay","ThreeWay","ThreeWay","klaR","klaR","klaR","klaR","klaR","klaR","klaR","klaR","klaR","klaR","omicade4","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","mixOmics","MultiDataSet","MultiDataSet","MultiDataSet","MultiDataSet","MultiDataSet","MultiDataSet","VIM","VIM","VIM","VIM","VIM","VIM","VIM","VIM","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","impute","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","WGCNA","Amelia","Amelia","Amelia","Amelia","Amelia","Amelia","Amelia","imputeTS","imputeTS","imputeTS","imputeTS","imputeTS","imputeTS","missMDA","missMDA","missMDA","missForest","missForest","missForest","missForest","missForest","missForest","missForest","missForest","missForest","missForest","missForest","missForest","missForest","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mice","mi","mi","mi","tidygraph","tidygraph","tidygraph","tidygraph","tidygraph","tidygraph","tidygraph","tidygraph","tidygraph","statnet","RedeR","RedeR","RedeR","RedeR","tmod","sybil","sybil","sybil","sybil","sybil","sybil","sybil","sybil","sybil","sybil","SPIA","rBiopaxParser","rBiopaxParser","graphite","graphite","graphite","graphite","graphite","graphite","graphite","graphite","NCIgraph","pathview","pathview","pathview","pathview","pathview","pathview","pathview","pathview","pathview","pathview","pathview","pathview","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","KEGGgraph","rsbml","PSICQUIC","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","KEGGREST","faahKO","faahKO","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","rcdk","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","ChemmineR","fingerprint","fingerprint","fingerprint","fingerprint","fingerprint","webchem","webchem"],"to":["adductomicsR","Autotuner","CluMSID","MSGFgui","MSnbase","MSnID","msPurity","Pbase","ProteomicsAnnotationHubData","RMassBank","SIMAT","topdownr","xcms","yamss","MRMConverteR","KPIC2","peakPantheR","sRm","TarMet","qcrms","compMS2Miner","ReSOLUTION","MetFamily","mzmatch.R","patRoon","METABOseek","Autotuner","CluMSID","MSnID","Pbase","ProteomicsAnnotationHubData","RMassBank","topdownr","xcms","DAPAR","DEP","MetCirc","msmsEDA","msmsTests","MSstatsQC","PrInCE","pRoloc","pRolocGUI","proteoQC","qPLEXanalyzer","synapter","massFlowR","peakPantheR","qcrms","MSnio","MetaboAnalystR","patRoon","METABOseek","CompoundDb","CAMERA","cosmiq","flagme","IPO","LOBSTAHS","MAIT","Metab","metaMS","ncGTW","proFIA","Risa","massFlowR","geoRge","IsotopicLabelling","sRm","DecoMetDIA","credential","qcrms","MetShot","MetFamily","MetaboAnalystR","mzmatch.R","simExTargId","specmine","patRoon","METABOseek","flagme","IPO","LOBSTAHS","MAIT","metaMS","DecoMetDIA","credential","ProbMetab","MetShot","MetaboAnalystR","simExTargId","patRoon","METABOseek","MetaboList","patRoon","RMassScreening","IsoCorrectoRGUI","baitmet","xcms","cosmiq","Rnmr1D","speaq","TarMet","Rnmr1D","MetaboMate","rDolphin","specmine","Rnmr1D","alsace","chipPCR","MODIS","PepsNMR","RGCxGC","RpeakChrom","SOAP-NMR","MetaboMate","mzmatch.R","ASICS","proteoQC","PGA","sapFinder","shinyTANDEM","HiResTEC","InterpretMSSpectrum","MatchWeiz","MetDNA","METABOseek","MetaboShiny","HiResTEC","RAMClustR","MetaboShiny","InterpretMSSpectrum","nontarget","TarMet","IPA","ReSOLUTION","patRoon","MetaboShiny","RChemMass","MetaDBparse","adductomicsR","ReSOLUTION","MetaboSignal","NADfinder","pems.utils","powdR","spectacles","spectralAnalysis","takos","rDolphin","specmine","NormalizeMets","MSnbase","DAPAR","RAMClustR","crmn","ADAPTS","CompGO","consensusDE","DeconRNASeq","DiffCorr","imputeLCMD","LOST","MEET","MetabolomicsBasics","missCompare","multiDimBio","OUTRIDER","PanVizGenerator","polyPK","polyRAD","santaR","scde","SomaticSignatures","MetaboMate","MetFamily","MetaboAnalystR","MetCleaning","simExTargId","MetMSLine","MetaboShiny","amp","diffcoexp","xcms","synapter","DiffCorr","a4Base","ABarray","aCGH","adSplit","ALDEx2","anota","anota2seq","apmsWAPP","BicARE","ChIPpeakAnno","cp4p","EMA","GExMap","hddplot","INCATome","iPAC","IsoGeneGUI","KCsmart","KnowSeq","LMGene","mAPKL","metabomxtr","MetaIntegrator","mutoss","nethet","nlcv","OCplus","PCS","phyloseq","PREDA","pRF","rain","REDseq","RTopper","SAGx","siggenes","singleCellTK","structSSI","TcGSA","webbioc","ProbMetab","DEP","DiffCorr","beam","cape","carSurv","CosW","crossmeta","DBGSA","DChIPRep","DMCHMM","DNLC","ERP","FitHiC","GeneCycle","GeneNet","GSRI","HDMT","HybridMTest","IHW","LANDD","les","meshr","netbenchmark","NetworkToolbox","nlnet","oposSOM","OVESEG","qgraph","rags2ridges","ragt2ridges","sda","SecKW","SimSeq","SinIW","st","TanB","TBEST","MatchWeiz","proFIA","ASICS","MetabolomicsBasics","polyPK","biosigner","lipidr","rqt","phenomis","MAIT","rqt","apercu","autopls","bestglm","BioMark","BioMedR","chemmodlab","chemometrics","ChemometricsWithR","chillR","cocoreg","CSFA","DaMiRseq","dbstats","dmm","EasyABC","enpls","fdm2id","funtooNorm","groc","IntClust","LINselect","lmdme","lspls","mephas","MetabolicSurv","MetStaT","mixlm","MLInterfaces","monomvn","muma","obliqueRF","OHPL","plsRcox","plsVarSel","RcmdrPlugin.NMBU","regressoR","rminer","robCompositions","rrcovHD","RVAideMemoire","RVtests","spls","SQB","sRAP","statTarget","svapls","SVAPLSseq","tempoR","tsensembler","MetaboAnalystR","structToolbox","MetCleaning","specmine","MetaboShiny","DaMiRseq","MetSizeR","INCATome","HTSanalyzeR","RNAither","sigQC","synlet","tRanslatome","cocoreg","staRdom","ttTensor","FPDclustering","GSparO","PCovR","rrcov3way","agricolae","BioSeqClass","CLUSTShiny","DeducerPlugInScaling","diceR","fingerPro","minque","MLDAShiny","TestDimorph","tiger","omicRexposome","MetabolomicsBasics","polyPK","plsRcox","RVAideMemoire","bootsPLS","compartmap","DepecheR","mixKernel","PCA4you","R2GUESS","sgPLS","YuGene","MetFamily","xMWAS","ropls","biosigner","omicRexposome","MEAL","IntLIM","phenomis","missCompare","robCompositions","analytics","destiny","sdcMicro","simPop","smartdata","VIMGUI","MSnbase","DAPAR","Rnmr1D","speaq","NormalizeMets","imputeLCMD","polyPK","statTarget","AMARETTO","armada","CancerSubtypes","CGHcall","ChAMP","cola","doppelgangR","EGAD","ENmix","ePCR","FAMT","fastLiquidAssociation","FEM","genefu","genomation","iC10","maGUI","MetaPath","MethylMix","miRLAB","moduleColor","netboost","Pigengene","PMA","REMP","Rnits","samr","snpReady","swamp","TIN","WGCNA","xseq","FIEmspro","MetaboAnalystR","MetaboDiff","MetCleaning","MetDNA","specmine","MetaboShiny","ADAPTS","fastLiquidAssociation","maGUI","netboost","Pigengene","TIN","MRPC","diffcoexp","CEMiTool","CVE","DCD","DGCA","eclust","epihet","GOGANPA","MCbiclust","MetNet","miRSM","MODA","Patterns","proBatch","seq2pathway","MetaboDiff","xMWAS","missCompare","smartdata","autovarCore","bmem","TestDataImputation","Zelig","zeligverse","EventDetectR","gimme","hpiR","imputeTestbench","psychNET","TrendSLR","missCompare","smartdata","INSPIRE","proFIA","speaq","ADAPTS","missCompare","smartdata","bartMachine","KarsTS","obliqueRSF","metabolyseR","MetCleaning","MetDNA","MetaboShiny","amp","missCompare","smartdata","TestDataImputation","missMDA","accelmissing","BaM","bootImpute","CALIBERrfimpute","dlookr","dynr","finalfit","genpathmox","HardyWeinberg","hmi","hot.deck","ImputeRobust","JWileymisc","logistf","MatchIt.mice","miceadds","micemd","miceMNAR","mipred","MRPC","PathSelectMP","Qtools","RBtest","RegularizedSCA","Replication","rexposome","roughrf","sociome","weights","weightTAPSPACK","MetaboShiny","missCompare","migui","sem","clustree","clustringr","fastnet","ggdag","ggraph","HCABrowser","particles","scholar","stminsights","ergMargins","dc3net","PANR","RTN","transcriptogramer","disco","BacArena","CoRegFlux","exp2flux","g2f","GlobalFit","metaboGSE","RSeed","sybilccFBA","sybilcycleFreeFlux","sybilDynFBA","EnrichmentBrowser","AnnotationHubData","pwOmics","EnrichmentBrowser","ICDS","mogsa","netgsa","PoTRA","ReactomePA","StarBioTrek","ToPASeq","DEGraph","CompGO","KnowSeq","EnrichmentBrowser","BioNetStat","debrowser","EGSEA","GDCRNATools","MAGeCKFlute","pathfindR","RNASeqR","SBGNview","TCGAbiolinksGUI","MWASTools","MetaboSignal","maGUI","SPIA","EnrichmentBrowser","NCIgraph","DEGraph","pathview","clipper","iCARH","KEGGlincs","NFP","ROntoTools","MetaboAnalystR","specmine","MetaboShiny","BiGGR","RefNet","MWASTools","MetaboSignal","maGUI","g2f","EnrichmentBrowser","pathview","KEGGlincs","ROntoTools","ADAM","attract","CNEr","FELLA","gage","KEGGprofile","omu","PAPi","PathwaySplice","RnaSeqSampleSize","YAPSA","specmine","MetaboShiny","MetaDBparse","cosmiq","massFlowR","RMassBank","BioMedR","enviGCMS","pmd","rcellminer","Rcpi","RxnSim","SimuChemPC","TarMet","ReSOLUTION","Retip","patRoon","MetaboShiny","RChemMass","MetaDBparse","BioMedR","Rcpi","bioassayR","eiR","fmcsR","MetID","Rchemcpp","uCAREChemSuiteCLI","MetaboShiny","MetaDBparse","CompoundDb","rcdk","rcellminer","RxnSim","rpubchem","patRoon","RAMClustR","MetaDBparse"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"physics":{"timestep":0.3,"barnesHut":{"centralGravity":0.35,"springLength":95}}},"groups":null,"width":1000,"height":1000,"idselection":{"enabled":false,"style":"width: 150px; height: 26px","useLabels":true,"main":"Select by id"},"byselection":{"enabled":false,"style":"width: 150px; height: 26px","multiple":false,"hideColor":"rgba(200,200,200,0.5)"},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","highlight":{"enabled":true,"hoverNearest":false,"degree":1,"algorithm":"all","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":false,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"}},"evals":[],"jsHooks":[]}</script>
<p>Figure S1: Dependency network of R packages. Shown in blue are packages mentioned in the review. Edges connect to packages that depend on another package, as long as that is in CRAN or BioC. Green nodes correspond to packages in CRAN or BioC not covered in the review. Not shown are 1) infrastructure packages e.g. rJava, Rcpp 2) packages from the review without reverse dependencies and 3) data packages. Some packages from the review are not in current versions of CRAN or BioC. An interactive version of this figure is available from <a href="appendix-2-metarbolomics-dependencies-network.html">https://stanstrup.gitlab.io/metaRbolomics-book/appendix-2-metarbolomics-dependencies-network.html</a>.</p>
</div>
<div id="save-network-plot" class="section level3 unnumbered">
<h3>Save network plot</h3>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="kw">saveNetwork</span>(vn, <span class="st">&quot;vn.html&quot;</span>)</a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="kw">webshot</span>(<span class="st">&quot;vn.html&quot;</span>, <span class="st">&quot;revDepNet-60.png&quot;</span>, <span class="dt">delay =</span> <span class="dv">60</span>)</a></code></pre></div>
<p><img src="The_MetaRbolomics_book_files/figure-html/fileVis-1.png" width="672" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1">vnZoom &lt;-<span class="st"> </span><span class="kw">visNetwork</span>(<span class="dt">nodes =</span> data<span class="op">$</span>nodes, </a>
<a class="sourceLine" id="cb26-2" data-line-number="2">                 <span class="dt">edges =</span> data<span class="op">$</span>edges,</a>
<a class="sourceLine" id="cb26-3" data-line-number="3">                 <span class="dt">width=</span><span class="dv">1000</span>, <span class="dt">height=</span><span class="dv">1000</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb26-4" data-line-number="4"><span class="st">    </span><span class="kw">visIgraphLayout</span>()<span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb26-5" data-line-number="5"><span class="st">    </span><span class="kw">visEvents</span>(<span class="dt">type=</span><span class="st">&quot;once&quot;</span>, <span class="dt">startStabilizing =</span> <span class="st">&#39;function() {</span></a>
<a class="sourceLine" id="cb26-6" data-line-number="6"><span class="st">              this.fit({nodes:[&quot;ptw&quot;, &quot;Rnmr1D&quot;, &quot;RpeakChrom&quot;, &quot;alsace&quot;, </span></a>
<a class="sourceLine" id="cb26-7" data-line-number="7"><span class="st">                               &quot;PepsNMR&quot;, &quot;ASICS&quot;, &quot;MODIS&quot;, &quot;RGCxGC&quot;]})</span></a>
<a class="sourceLine" id="cb26-8" data-line-number="8"><span class="st">            }&#39;</span>)</a>
<a class="sourceLine" id="cb26-9" data-line-number="9"><span class="kw">saveNetwork</span>(vnZoom, <span class="st">&quot;vnZoom.html&quot;</span>)</a>
<a class="sourceLine" id="cb26-10" data-line-number="10"><span class="kw">webshot</span>(<span class="st">&quot;vnZoom.html&quot;</span>, <span class="st">&quot;revDepNet-zoom.png&quot;</span>, <span class="dt">delay =</span> <span class="dv">5</span>)</a></code></pre></div>
<p><img src="The_MetaRbolomics_book_files/figure-html/fileVisZoom-1.png" width="672" /></p>
<p>You ca access the files at:</p>
<ul>
<li><a href="vn.html" class="uri">vn.html</a></li>
<li><a href="revDepNet-60.png" class="uri">revDepNet-60.png</a></li>
<li><a href="vnZoom.html" class="uri">vnZoom.html</a></li>
<li><a href="revDepNet-zoom.png" class="uri">revDepNet-zoom.png</a></li>
</ul>
</div>
<div id="notes" class="section level3 unnumbered">
<h3>Notes</h3>
<p>The source code for this page is on GitHub at <a href="https://gitlab.com/stanstrup/metaRbolomics-book">gitlab.com/stanstrup/metaRbolomics-book</a></p>
<p>The HTML output is shown at
<a href="appendix-2-metarbolomics-dependencies-network.html">https://stanstrup.gitlab.io/metaRbolomics-book/appendix-2-metarbolomics-dependencies-network.html</a></p>
<p>and <a href="vn.html">https://stanstrup.gitlab.io/metaRbolomics-book/vn.html</a> (Caveat: long rendering time, blank page without any visible progress)</p>
<p>This page was created with the following packages:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.6.1 (2017-01-27)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.6 LTS
## 
## Matrix products: default
## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so
## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] desc_1.2.0          png_0.1-7           webshot2_0.0.0.9000
##  [4] chromote_0.0.0.9001 networkD3_0.4       visNetwork_2.0.8   
##  [7] igraph_1.2.4.1      devtools_2.2.0      usethis_1.5.1      
## [10] tikzDevice_0.12.3   purrr_0.3.2         kableExtra_1.1.0   
## [13] DT_0.9              dplyr_0.8.3         googlesheets_0.3.0 
## [16] readr_1.3.1         knitr_1.25         
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.1          pkgload_1.0.2       jsonlite_1.6       
##  [4] viridisLite_0.3.0   shiny_1.3.2         assertthat_0.2.1   
##  [7] BiocManager_1.30.4  cellranger_1.1.0    yaml_2.2.0         
## [10] remotes_2.1.0       sessioninfo_1.1.1   pillar_1.4.2       
## [13] backports_1.1.4     glue_1.3.1          digest_0.6.21      
## [16] promises_1.0.1.9002 rvest_0.3.4         colorspace_1.4-1   
## [19] websocket_1.1.0     htmltools_0.3.6     httpuv_1.5.2       
## [22] pkgconfig_2.0.2     bookdown_0.13.2     xtable_1.8-4       
## [25] scales_1.0.0        webshot_0.5.1       processx_3.4.1     
## [28] later_0.8.0.9004    tibble_2.1.3        ellipsis_0.2.0.1   
## [31] withr_2.1.2         cli_1.1.0           magrittr_1.5       
## [34] crayon_1.3.4        mime_0.7            memoise_1.1.0      
## [37] evaluate_0.14       ps_1.3.0            fs_1.3.1           
## [40] xml2_1.2.2          pkgbuild_1.0.5      tools_3.6.1        
## [43] prettyunits_1.0.2   hms_0.5.1           stringr_1.4.0      
## [46] munsell_0.5.0       callr_3.3.1         compiler_3.6.1     
## [49] rlang_0.4.0         grid_3.6.1          rstudioapi_0.10    
## [52] htmlwidgets_1.3     filehash_2.4-2      crosstalk_1.0.0    
## [55] rmarkdown_1.15      testthat_2.2.1      codetools_0.2-16   
## [58] curl_4.0            R6_2.4.0            fastmap_1.0.0      
## [61] zeallot_0.1.0       rprojroot_1.3-2     stringi_1.4.3      
## [64] Rcpp_1.0.2          vctrs_0.2.0         tidyselect_0.2.5   
## [67] xfun_0.9</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="9901-Appendices-Appendix_1_The_MSP_File_Format_and_package_support.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rformassspectrometry/metaRbolomics-book/edit/master/rmd/99-Appendices/9902-Appendices-Appendix_2_metaRbolomics_dependencies.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["The_MetaRbolomics_book.pdf", "The_MetaRbolomics_book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
