\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdftitle={The MetaRbolomics book},
            pdfauthor={Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs}
% Fix footnotes in tables (requires footnote package)
\IfFileExists{footnote.sty}{\usepackage{footnote}\makesavenoteenv{longtable}}{}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother

\usepackage{array}
\usepackage[table]{xcolor}
\usepackage{caption}

\title{The MetaRbolomics book}
\author{Jan Stanstrup, Corey D. Broeckling, Rick Helmus, Nils Hoffmann, Ewy Mathé, Thomas Naake, Luca Nicolotti, Kristian Peters, Johannes Rainer, Reza M. Salek, Tobias Schulze, Emma L. Schymanski, Michael A. Stravs, Etienne A Thévenot, Hendrik Treutler, Ralf J. M. Weber, Egon Willighagen, Michael Witting, Steffen Neumann}
\date{}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\newpage

\hypertarget{preface}{%
\section*{Preface}\label{preface}}
\addcontentsline{toc}{section}{Preface}

The term metaRbolomics has been coined for a workshop at the 2016
annual conference of the international Metabolomics society in Dublin,
Ireland. On May 11, 2016, a small team of authors started to compile a
review of ``a few R packages'' and how they can be used to do
metabolomics data analysis. After some hiatus, Jan Stanstrup revived
the idea and this turned into a 60+ page review (accepted in MDPI
Metabolites) with more than 200 packages and more than 300
references. But the journey did not end there. In march 2018 Egon
Willighagen suggested that this effort should turn into a book, and
more importantly, should become a community effort, with more
introduction and also include code examples. It now found a home at
\href{https://rformassspectrometry.github.io/RforMassSpectrometry/}{RforMassSpectrometry},
a community website dedicated to R software for the analysis and
interpretation of high throughput mass spectrometry assays, including
proteomics and metabolomics experiments.

\hypertarget{how-to}{%
\subsection*{How to \ldots{}}\label{how-to}}
\addcontentsline{toc}{subsection}{How to \ldots{}}

\textbf{\ldots{} download data}
The list of packages found in the tables in this book can be downloaded from
\url{public/data/AllMetaRbolomicsTables.csv}.

\textbf{\ldots{} add packages to the tables}
Go to \href{https://docs.google.com/spreadsheets/d/1QTe_FhiiFEJJ7hn25V973OV3Y9up8CsiLXgZXgTg_UU/edit?usp=sharing}{the googlesheet}
and add the package. Please be careful with adding it to the right section.
If it belongs in more than one table add it mulitple times as appropiate.

The package will not appear instantly in the book but only after a change is made to
the book itself. You can also \href{https://github.com/rformassspectrometry/metaRbolomics-book/issues}{open an issue}
and request the reload such that the package shows up.

\textbf{\ldots{} contribute to the text}
There are several options. In order of convenience for the maintainer you can:

\begin{itemize}
\tightlist
\item
  make a pull request on the \href{https://github.com/rformassspectrometry/metaRbolomics-book}{GitHub repository}. You will find the text in the \texttt{rmd} folder.
\item
  \href{https://github.com/rformassspectrometry/metaRbolomics-book/issues}{open an issue} with the text you want to contribute. Clearly indicate where the text belongs.
\item
  Send your contribution by email to jst( a t )nexs.ku.dk.
\end{itemize}

Remember to add yourself to author contributions.

\hypertarget{this-book-in-other-formats}{%
\subsection*{This book in other formats}\label{this-book-in-other-formats}}
\addcontentsline{toc}{subsection}{This book in other formats}

The book is also available as static \href{https://rformassspectrometry.github.io/metaRbolomics-book/The_MetaRbolomics_book.pdf}{PDF}
file, and as \href{https://rformassspectrometry.github.io/metaRbolomics-book/The_MetaRbolomics_book.epub}{ePUB}.

\newpage

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

Metabolomics aims to measure, identify and (semi-)quantify a large number of metabolites in a biological system. The methods of choice are generally Nuclear Magnetic Resonance (NMR) spectroscopy or Mass Spectrometry (MS). The latter can be used directly (e.g.~direct infusion MS), but is normally coupled to a separation system such as Gas Chromatography (GC-MS), Liquid Chromatography (LC-MS) or Capillary Electrophoresis (CE-MS). In order to increase the separation power multidimensional separation systems are becoming common, such as comprehensive two-dimensional GC or LC (GC×GC, LC×LC) or LC combined with ion mobility spectrometry (LC-IMS) before MS detection. Other detection techniques include Raman spectroscopy, UV/VIS (ultraviolet/visible absorbance spectrophotometric detection- typically with a Diode Array Detector (DAD)) and fluorescence. NMR also benefits from separation techniques, such as LC-MS-NMR or LC-SPE-NMR. Additionally, there are a wide variety of pulse programs commonly used in 1D and even bigger set of 2D pulse programs used in metabolomics and for metabolite identification, for a comprehensive review on this see {[}\protect\hyperlink{ref-emwas_2019}{1}{]}. A general introduction to metabolomics can be found in textbooks like {[}\protect\hyperlink{ref-lammerhofer_2013}{2}--\protect\hyperlink{ref-wehrens_2019}{4}{]} or online courses like {[}\protect\hyperlink{ref-InternationalMetabolomicsSociety_2019}{5},\protect\hyperlink{ref-salek_website_nd}{6}{]}.

All of these analytical platforms and methodologies generate large amounts of high dimensional and complex experimental raw data when used in a metabolomics context. The amount of data, the need for reproducible research, and the complexities of the biological problem under investigation necessitates a high degree of automation and standard workflows in the data analysis. Beside vendor software, which is usually not open, open source projects offer the possibility to work in community-driven teams, perform reproducible data analysis and to work with different types of raw data. Many tools and methods have been developed to facilitate the processing and analysis of metabolomics data; most seek to solve a specific challenge in the multi-step data processing and analysis workflow.

This review provides an overview of the metabolomics-related tools that are made available as packages (and a limited number of non-trivial, non-packaged scripts) for the statistics environment and programming language R {[}\protect\hyperlink{ref-rcoredevelopmentteam_website_2018}{7}{]}. We have included packages even if they are not anymore part of current CRAN or Bioconductor, i.e.~as archived versions only. We have not included packages described in the literature if no longer available for download at all. We did include packages that are currently available, but not yet published in the scientific literature. The package descriptions have been grouped in sections according to the typical steps in the metabolomics data analysis pipeline for different analytical technologies, following the typical workflow steps from MS, NMR and UV data analysis, metabolite annotation, statistical analysis, molecular structure, network and pathway analysis and finally covering packages embracing large parts of the workflow.

\newpage

\hypertarget{metabolomics-data-processing-and-analysis}{%
\subsection{Metabolomics data processing and analysis}\label{metabolomics-data-processing-and-analysis}}

\includegraphics{figures/Figure 1.png}
\emph{Figure 1: Overview of typical tasks in metabolomics workflows, ranging from metabolite profiling (left, green) via metabolite annotation (center, purple) to data analysis using statistics and metabolite networks (right, red).}

The remainder of this section gives a broad overview and explains the typical steps, which are summarized in Figure 1, while common approaches and the available R packages are described in more detail in Section 2.

The first step for any metabolomics study is conversion from vendor formats into open data formats and pre-processing of the obtained raw data. The latter entails converting chromatographic (usually hyphenated to MS) or spectroscopic data into a data matrix suitable for data analysis. For LC-MS data this typically involves feature detection (or peak-picking) in individual samples followed by matching of features between samples. For spectroscopic data, this typically means alignment of spectra and potentially binning of the spectra into `buckets'. The final matrix will have samples in one dimension and so-called features (unique chromatographic features or spectral bins) in the other dimension. In NMR based metabolomics, several steps are carried out to process raw time domain data to a spectrum to improve quality such as phasing and baseline correction of the spectrum. Next is alignment of peaks across spectra and samples, followed by segmenting data into bins or a peak fitting step depending on the method used.

Once the analytical data has been preprocessed, it is generally subjected to different statistical approaches to find features that are ``interesting'' in the context of the experimental design, e.g.~differentiating diseased patients from healthy controls.
In untargeted metabolomics, the selected features contain only the characteristics (e.g.~m/z, retention time, chemical shift, intensity) obtained from the measurement, but not (yet) the metabolite identification or chemical structure as such. Different approaches exist for this metabolite annotation step, ranging from (usually insufficient) database lookup of exact mass (MS) or chemical shift (NMR) alone, to the use of fragmentation patterns obtained in tandem MS experiments, which can be searched against spectral databases or analysed with in silico algorithms, to spectral searching or de novo structure elucidation using combinations of NMR experiments (often 1D and 2D).

Large parts of the metabolomics software landscape in general have been covered in reviews, recent ones include the large list of software packages {[}\protect\hyperlink{ref-spicer_website_2018}{8}{]} first described by Spicer et al. {[}\protect\hyperlink{ref-spicer_2017}{9}{]}, and a series of annual reviews covering the list maintained by Misra and others {[}\protect\hyperlink{ref-misra_2016}{10}--\protect\hyperlink{ref-misra_website_2017}{13}{]}, a review by Kannan et al. {[}\protect\hyperlink{ref-kannan_2016}{14}{]} and the review focussing on approaches for compound identification of LC-MS/MS data by Blaženović et al. {[}\protect\hyperlink{ref-blaenovi_2018}{15}{]}. These reviews did include software regardless of the programming environment or language used for the implementation. In section 2.9 we briefly mention how those can be accessed from within R.

This review will focus on the ecosystem of R packages for metabolomics. It provides an overview of packages to carry out one or multiple of the above mentioned steps. Some aspects are not covered in depth or not at all. For example, MS based imaging in metabolomics is an area that has unique challenges and merits its own review, and it is also beyond the scope of this review to discuss all statistical methods that could be applied in metabolomics.

\newpage

\hypertarget{the-r-package-landscape}{%
\subsection{The R package landscape}\label{the-r-package-landscape}}

The core of the R language was started in 1997 and provided the basic functionality of a programming language, with some functions targeting statistics. The real power driving the popularity of R today is the huge number of contributed packages providing algorithms and data types for a myriad of application realms. Many packages have an Open Source license. This is not a phenomenon exclusive to R, but is rather a positive cultural aspect of bioinformatics software being mostly published under Open Source license terms, regardless of the implementation language. An R interpreter can be embedded in several other languages to execute R code snippets, and R code can also be executed via different workflow systems (e.g.~KNIME or Galaxy, see section 2.9), which is beneficial for analysis workflows, interoperability and reuse.

These packages are typically hosted on platforms that serve as an umbrella project and are a ``home'' for the developer and user communities. The Comprehensive R Archive Network (CRAN) repository contains over 14,500 packages for many application areas, including some for bioinformatics and metabolomics. The ``CRAN Task Views'', which are manually curated resources describing available packages, books etc, help users navigate CRAN and find packages for a particular task. For metabolomics, the most relevant Task View is ``\emph{Chemometrics and Computational Physics}'' {[}\protect\hyperlink{ref-mullen_website_2019}{16}{]} edited by Katharine Mullen, which includes sections on Spectroscopy, Mass Spectrometry and other tasks relevant for metabolomics applications. The Bioconductor project (BioC for short) was started by a team around Robert Gentleman in 2001 {[}\protect\hyperlink{ref-gentleman_2004}{17}{]}, and has become a vibrant community of around 1,000 contributors, working on 1,741 software, 371 data and 948 annotation packages (BioC release 3.9). In addition to a rich development infrastructure (website, developer infrastructure, version control, build farm, etc) there are regular workshops for developers and users. To enable reproducible research, BioC runs bi-annual software releases tied to a particular R release, thus ensuring and guaranteeing interoperability of packages within the same BioC release and allowing to install BioC packages from a certain release to reproduce or repeat old data analyses. On both CRAN and BioC, each package has a landing page pointing to sources, build information, binary packages and documentation. On BioC, packages are sorted (by their respective authors) into ``BiocViews'', where most packages are targeting genomics and gene expression analysis, and the most relevant ones for metabolomics are Cheminformatics (containing 11 packages), Lipidomics (11), SystemsBiology (66) and, of course, Metabolomics (56). Bioconductor workflows (organised as separate BioC View {[}\protect\hyperlink{ref-bioconductor_website_nda}{18}{]}) provide well documented examples of typical analyses. For community support, BioC maintains mailing lists, a web-based support site, slack communication channels and more. Both CRAN and BioC have a well-defined process for accepting new packages, and the respective developer guidelines (see guidelines for CRAN {[}\protect\hyperlink{ref-thecomprehensiverarchivenetwork_website_nd}{19}{]} and for BioC {[}\protect\hyperlink{ref-bioconductor_website_nd}{20}{]}) cover the package life-cycle from submission, updates and maintenance, to deprecation/orphaning of packages. In the case of BioC, new submissions undergo a peer review process, which also provides feedback on technical aspects and integration with the BioC landscape.

A smaller number of packages are also hosted on sites like \href{https://rforge.net/}{rforge.net}, \href{http://r-forge.wu-wien.ac.at/}{r-forge.wu-wien.ac.at} {[}\protect\hyperlink{ref-theul_2009}{21}{]}, or \href{https://sourceforge.net/}{sourceforge.net} (SF). The non-profit initiative rOpenSci {[}\protect\hyperlink{ref-boettiger_2015}{22}{]} maintains an ecosystem around reproducible research, including staff and community-contributed R packages with additional peer review. Currently, there are no specific metabolomics related packages.

The GitHub (and also GitLab, Bitbucket) hosting services are not specific to R development, but have gained a lot of popularity due to their excellent support for participation and contribution to software projects. The maintenance of BioC packages on one of the git-based sites has become easier since the BioC team migrated to git as its version control system. A downside of these generic repository hosting sites is that there is no central point of entry, and finding packages for specific tasks is difficult compared with dedicated platforms and relies on search engines and publications. Also, while these hosting services make it easier to provide packages that do not meet BioC and CRAN requirements (e.g.~rinchi due to limitations in the InChI algorithm itself), it also allows users to postpone (or circumvent entirely) the review process that helps ensure the quality of BioC contributions. In addition to generic search engines like \href{http://google.com}{Google}\href{http://google.com}{.com} or \href{http://bing.com}{Bing}\href{http://bing.com}{.com}, the \href{https://rdrr.io/}{rdrr.io} is a comprehensive index of R packages and documentation from CRAN, Bioconductor, GitHub and R-Forge. Initially, its main purpose was to find R packages by name, perform full-text search in package documentation, functions and R source code. Recently, it also serves as hub to actually run R code without local installation, see Section 2.9.

\newpage

\hypertarget{dependences-and-connectivity-of-metabolomics-packages}{%
\subsection{Dependences and connectivity of metabolomics packages}\label{dependences-and-connectivity-of-metabolomics-packages}}

\includegraphics{figures/Figure 2.png}
\emph{Figure 2: Dependency network of R packages. Shown in blue are packages mentioned in the review. Edges connect to packages that depend on another package, as long as they are in CRAN or BioC. Green nodes correspond to packages in CRAN or BioC not covered in the review. The inset shows the neighbourhood of the ptw package. Not shown are 1) infrastructure packages, e.g.~rJava, Rcpp 2) packages from the review without reverse dependencies and 3) data packages. Some packages from the review are not in current versions of CRAN or BioC. An interactive version of this figure is also available online (\href{https://rformassspectrometry.github.io/metaRbolomics-book/}{rformassspectrometry.github.io/metaRbolomics-book}, Appendix 2) and as supplemental file 2.}

Code reuse and object inheritance can be a sign for a well-connected and interacting community. At the useR!2015 and JSM2015 conferences, A. de Vries and J. Rickert (both Microsoft, London, UK) showed the analysis of the CRAN and BioC dependency network structure {[}\protect\hyperlink{ref-devries_website_2015b}{23}--\protect\hyperlink{ref-devries_website_2015}{25}{]}. Compared to CRAN, BioC packages had a higher connectivity: ``\emph{It seems that the Bioconductor policy encourages package authors to reuse existing material and write packages that work better together}''. We repeated such an analysis {[}\protect\hyperlink{ref-neumann_website_2019}{26}{]} with the packages mentioned in this review and created a network of reverse dependencies (i.e., the set of packages that depend on these metabolomics related packages in BioC or CRAN). The resulting network is shown in Figure 2.

\newpage

\hypertarget{r-packages-for-metabolomics}{%
\section{R-packages for metabolomics}\label{r-packages-for-metabolomics}}

This section reviews packages, relates some of those with similar functionality, and mentions how some of the packages can be used together. The sections in this review are ordered according to specific analytical approaches and the individual required steps.

\newpage

\hypertarget{mass-spectrometry-data-handling-and-pre-processing}{%
\subsection{Mass spectrometry data handling and (pre-)processing}\label{mass-spectrometry-data-handling-and-pre-processing}}

For all mass spectrometers, the fundamental data generated is a mass spectrum, i.e.~mass-signal intensity pairs. MS-based metabolomics data is typically acquired either as a single mass spectrum or a collection of mass spectra over time, with the time axis (retention time) defined by chromatographic (or other time domain) separation. One of the first steps in metabolomics data processing is usually the reduction of the typically large raw data produced by the instrument to a much smaller set of so called \emph{features}, which are then subjected to downstream data analysis and interpretation. Features normally represent integrated peaks for a given mass that have been aligned across samples. Establishing these features is called \emph{pre-processing}. The feature detection approaches and packages applicable depend on the type and characteristics of the input data. This section describes the basic data structure for some of the common analytical approaches and shows appropriate tools in R for pre-processing such data, see Table 1 for an overview of the corresponding packages.

\hypertarget{profile-mode-and-centroided-data}{%
\subsubsection{Profile mode and centroided data}\label{profile-mode-and-centroided-data}}

The mass spectra can be recorded in profile (also called continuum) mode, but are often `centroided'. Centroiding is, in effect, a process of peak detection for a profile mode mass spectrum (hence in the \emph{m/z} dimension, not in a chromatographic dimension) - a gaussian region of a continuum spectrum with a sufficiently high signal to noise ratio is integrated to give a centroided mass (a ``stick'' in the mass spectrum as opposed to a continuous signal) and integrated area under the curve. This results in data of reduced size - what was many \emph{m/z}-intensity pairs has been reduced to a single \emph{m/z}-intensity pair. Practically, this reduces the file size considerably and many data processing tools (e.g. \emph{centWave} in xcms) require MS data that has been centroided. The centroiding can be done either during acquisition on the fly by the instrument software, or as an initial processing step. Post-acquisition centroiding can be performed during conversion of the vendor data format to open formats; typically using msconvert from ProteoWizard {[}\protect\hyperlink{ref-chambers_2012}{27},\protect\hyperlink{ref-kessner_2008}{28}{]} which in some cases provides access to vendor centroiding algorithms or can alternatively use its own built-in centroinding method. Dedicated vendor tools can also be used, and the R packages MSnbase also provides centroiding capabilities.

\hypertarget{direct-infusion-mass-spectrometry-data}{%
\subsubsection{Direct infusion mass spectrometry data}\label{direct-infusion-mass-spectrometry-data}}

Currently, one of the highest throughput analytical approaches is direct infusion MS, where the sample is directly injected into the mass spectrometer without any chromatographic separation. This approach can be used with high mass resolution or ultra-high resolution mass spectrometers to discriminate isobaric analytes {[}\protect\hyperlink{ref-fuhrer_2011}{29}{]}. Summing or averaging these spectra generates a single mass spectrum, which is representative of that sample. Peak picking can be done using MassSpecWavelet that applies a continuous wavelet transform-based peak detection. xcms provides a wrapper for this function in the \emph{findPeaks.MSW} function. In the Flow Injection Analysis analytical approach (FIA), the sample is transiently injected into the carrier stream flowing directly into the MS instrument. In the absence of chromatographic separation, matrix effects are a challenge for the quantification, especially in complex matrices. FIA coupled to High-Resolution Mass Spectrometry data can be processed with the proFIA workflow which provides efficient and robust peak detection and quantification.

\hypertarget{hyphenated-ms-and-non-targeted-data}{%
\subsubsection{Hyphenated MS and non-targeted data}\label{hyphenated-ms-and-non-targeted-data}}

Chromatographic separation before MS enables better measurement of complex samples and the ability to separate isobaric compounds. Here, the mass spectra are acquired over time as the sample components separate on the chromatography column. The mass spectrum at any given time has the same data structure as any mass spectrum - units of mass to charge ratio and time. As can be inferred from the above descriptions, chromatographically coupled mass spectrometry data is three-dimensional, with dimensions of retention time, \emph{m/z}, and intensity.

For the pre-processing of LC-MS and GC-MS data, xcms is widely used. A recent paper reviewed some of the ``xcms family'' packages {[}\protect\hyperlink{ref-mahieu_2016a}{30}{]} though many more packages exist that build on xcms by providing tools for specialised analyses while others provide improvements of some of the xcms processing steps such as improved peak picking (xMSanalyzer, warpgroup, cosmiq). xcms itself provides a number of different algorithms for peak picking such as \emph{matchedFilter} {[}\protect\hyperlink{ref-smith_2006}{31}{]}\emph{, centWave} {[}\protect\hyperlink{ref-tautenhahn_2008}{32}{]} and \emph{massifquant} {[}\protect\hyperlink{ref-conley_2014}{33}{]}. apLCMS, yamss, KPIC2 and enviPick also provide peak picking for LC-MS data independently of xcms. In cases where the alignment of the peak data of different samples is considered (e.g.~in cohort studies), xcms and apLCMS include methods to group the peaks by their \emph{m/z} and retention times within tolerance levels. The groups are split into sub-groups using density functions and the consensus \emph{m/z} and retention time is assigned to each bin.

\hypertarget{targeted-data-and-alternative-representations-of-data}{%
\subsubsection{Targeted data and alternative representations of data}\label{targeted-data-and-alternative-representations-of-data}}

In addition to the most standard ``spectra over time'' representation of chromatographically separated MS data, there are a number of alternative ways to represent the data or simplify the data. The signal intensity for a given mass (or mass range) over chromatographic time can be represented as two equal length vectors, with retention time and intensity as units for the values of those vectors. Examples of these vector pairs include the extracted ion chromatogram (EIC, sometimes also referred to as selected or eXtracted ion chromatogram SIC, XIC), where these chromatograms represent the intensity of a given mass over (retention) time. The data thus contains no spectra but a number of SICs. Frequently this is accomplished by summarizing the raw data in a two dimensional matrix consisting of \emph{m/z} and time dimensions, with each cell holding the signal intensity for that \emph{m/z} and retention time range (or bin). Low mass resolution mass spectrometers often represent the data natively as a SIC and targeted data are also usually represented this way. Recent versions of xcms are also able to process such data, and additional xcms-based functionalities for analysis of targeted data can be found in the packages TargetSearch and SWATHtoMRM, while analysis of isotope labeled data can be found in the packages X13CMS, geoRge, and IsotopicLabelling. SIMAT also provides processing for targeted data and does not rely on xcms.

\hypertarget{additional-dimensionality}{%
\subsubsection{Additional dimensionality}\label{additional-dimensionality}}

The vast majority of data collected for metabolomics comprises of three dimensions: retention time, \emph{m/z}, and intensity. However, there are more complicated analytical approaches that add additional dimensionality to the data. Two dimensional chromatography offer two separations in the chromatographic (retention time) domain. The eluent from one column is captured by retention time range and transferred to a second column where a fast orthogonal separation occurs. When coupled to a mass spectrometer, this generates four-dimensional data (\emph{m/z}, first retention time, second retention time, intensity).

Ion mobility separation (IMS) is a gas phase separation method offering resolution of ions based on molecular shape. This separation occurs on timescales of tens of microseconds, which generates a nested data structure in which there are dozens to hundreds of mass spectra collected across the IMS separation time scale. One can envision this as an ion mobility `chromatogram' - however, this chromatogram is nested within the actual chromatographic separation, thus LC-IMS-MS data is also four dimensional.

Most MS instruments offer the capability to perform selection (or filtering) of ions for fragmentation. The precursor selection can be performed through a quadrupole or ion trap, and fragmentation is often induced by collisions with an inert collision gas. Because this adds a level of mass spectrometry, it is called tandem MS, MS2 or MS/MS. Ion trap instruments can further select fragment ions and acquire MSn spectra.

There are several data independent MS/MS approaches, whereby MS/MS precursor selection is done, typically, on a scanning basis. These approaches perform precursor selection in a manner which does not depend on any feedback from the instrument control software or the MS level data. In practice, this precursor window can be either \emph{m/z} or ion mobility based. The processing tools within the R universe (discussed below) are so far underdeveloped for these approaches. With the increased popularity of multidimensional separation, the need for algorithms that can fully utilize the increased separation power is also increasing.

Currently, osd provide peak picking for unit resolution GC×GC-MS. While the msPeak package provides peak picking for GC×GC-MS data, the peak picking is done on the total ion chromatogram, thus not taking advantage of the mass selectivity provided by the MS detector. It does not appear that any package for R exists that provides peak picking for GC×GC-MS, LC×LC-MS or LC-IMS-MS, similar to (or even better than) commercial tools (e.g.~ChromaTOF, GC Image, ChromSquare). Also, at least in the case of GC×GC-MS, unit mass resolution still seems to be the most common use-case, even though high-resolution MS could further improve signal deconvolution and ultimately, analyte identification. Such capabilities are crucial for moving these new powerful analytical approaches into mainstream metabolomics analysis.

\hypertarget{structuring-data-and-metadata}{%
\subsubsection{Structuring data and metadata}\label{structuring-data-and-metadata}}

The result from the pre-processing is usually a matrix of abundances, rows being features (or features grouped into compounds/molecules) and columns being the samples. Within the statistical community, it is common nowadays to manipulate data matrices with rows as observations and columns as features, this difference stems from the early days, when spreadsheet programs could only handle a limited number of columns smaller than the number of \emph{e.g.} genes. Such matrices can be easily encapsulated into an \emph{ExpressionSet} class from Bioconductor's Biobase package {[}\protect\hyperlink{ref-huber_2015}{34}{]}, the more recent \emph{SummarizedExperiment} defined in the SummarizedExperiment {[}\protect\hyperlink{ref-martinmorgan_2017}{35}{]} package or the \emph{mSet} class from the metabolomics focussed Metabase {[}\protect\hyperlink{ref-zhu_website_2019}{36}{]} package. The main advantage of such objects is their inherent support to align quantitative data along with related metadata (i.e.~feature definitions/annotations as row - and sample annotations as column metadata). As an example, a \emph{SummarizedExperiment} can be generated from xcms pre-processing results by adding the output from the \emph{featureValues} function on the xcms result object as quantitative assay and the outputs of the \emph{featureDefinitions} and \emph{pData} functions as row and column annotations, respectively. Many Bioconductor packages for omics data analysis have native support for such objects (e.g., pcaMethods, STATegRa, ropls, biosigner, omicade4).

For the downstream export of mass spectrometry data from metabolomics or lipidomics experiments, the package rmzTab-M provides support for exporting quantitative and identification results backed by analytical and mass spectrometric evidence into the mzTab-M metabolomics file format {[}\protect\hyperlink{ref-hoffmann_2019}{37}{]}.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab1}R packages for mass spectrometry data handling and (pre-)processing.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab1}R packages for mass spectrometry data handling and (pre-)processing. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{MS data handling}}\\
Parser for common file formats: mzXML, mzData, mzML and netCDF. Usually not used directly by the end user, but provides functions to read raw data for other packages. & \href{https://doi.org/doi:10.18129/B9.bioc.mzR}{mzR} & BioC\\
Infrastructure to manipulate, process and visualise MS and proteomics data, ranging from raw to quantitative and annotated data. & \href{https://doi.org/doi:10.18129/B9.bioc.MSnbase}{MSnbase} & BioC\\
\rowcolor{gray!6}  Export and import of processed metabolomics MS results to and from the mzTab-M for metabolomics data format. & \href{https://lifs-tools.github.io/rmzTab-m/index.html}{rmzTab-M} & GitHub\\
Converts MRM-MS (.mzML) files to LC-MS style .mzML. & \href{https://github.com/wilsontom/MRMConverteR}{MRMConverteR} & GitHub\\
\rowcolor{gray!6}  Infrastructure for import, handling, representation and analysis of chromatographic MS data. & \href{https://github.com/RforMassSpectrometry/Chromatograms}{Chromatograms} & GitHub\\
Infrastructure for import, handling, representation and analysis of MS spectra. & \href{https://github.com/RforMassSpectrometry/Spectra}{Spectra} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Peak picking, grouping and alignment (LC-MS focussed or general)}}\\
Pre-processing and visualization for (LC/GC-)MS data. Includes visualization and simple statistics. & \href{https://doi.org/doi:10.18129/B9.bioc.xcms}{xcms} & BioC\\
Automatic optimization of XCMS parameters based on isotopes. & \href{http://bioconductor.org/packages/release/bioc/html/IPO.html}{IPO} & BioC\\
\rowcolor{gray!6}  Parameter tuning algorithm forXCMS, MZmine2, and other metabolomics data processingsoftware. & \href{https://doi.org/doi:10.18129/B9.bioc.Autotuner}{Autotuner} & BioC\\
Pre-processing and visualization for (LC/GC-)MS data. Includes visualization and simple statistics. & \href{http://bioconductor.org/packages/release/bioc/html/yamss.html}{yamss} & BioC\\
\rowcolor{gray!6}  Peak picking with XCMS and apLCMS, low intensity peak detection via replicate analyses. Multi-parameter feature extraction and data merging, sample quality and feature consistency evaluation. Annotation with METLIN and KEGG. & \href{https://sourceforge.net/projects/xmsanalyzer/}{xMSanalyzer} & SF\\
Pre-processing and alignment of LC-MS data without assuming a parametric peak shape model allowing maximum flexibility. It utilizes the knowledge of known metabolites, as well as robust machine learning. & \href{http://web1.sph.emory.edu/apLCMS/}{apLCMS} & SF\\
\rowcolor{gray!6}  Peak detection using chromatogram subregion detection, consensus integration bound determination and Accurate missing value integration. Outputs in XCMS-compatible format. & \href{https://github.com/nathaniel-mahieu/warpgroup}{warpgroup} & GitHub\\
Peak picking for (LC/GC-)MS data, improving the detection of low abundance signals via a master map of m/z/RT space before peak detection. Results are XCMS-compatible. & \href{https://doi.org/doi:10.18129/B9.bioc.cosmiq}{cosmiq} & BioC\\
\rowcolor{gray!6}  m/z detection (i.e. peak-picking) for accurate mass data, collecting all data points above an intensity threshold, grouping them by m/z values and estimating representative m/z values for the clusters; extracting EICs. & \href{http://amdorap.sourceforge.net/}{AMDORAP} & SF\\
(GC/LC)-MS data analysis for environmental science, including raw data processing, analysis of molecular isotope ratios, matrix effects, and short-chain chlorinated paraffins. & \href{https://cran.r-project.org/web/packages/enviGCMS/index.html}{enviGCMS} & CRAN\\
\rowcolor{gray!6}  Sequential partitioning, clustering and peak detection of centroided LC-MS mass spectrometry data (.mzXML), with Interactive result and raw data plot. & \href{https://cran.r-project.org/web/packages/enviPick/index.html}{enviPick} & CRAN\\
PeakpickingwithXCMS. Groups chemically related features beforealignmentacross samples. Additional processing after alignment includes feature validation, re-integration and annotation based on custom database. & \href{https://github.com/lauzikaite/massFlowR}{massFlowR} & GitHub\\
\rowcolor{gray!6}  KPIC2 extracts pure ion chromatograms (PIC) via K-means clustering of ions in region of interest, performs grouping and alignment, grouping of isotopic and adduct features, peak filling and Random Forest classification. & \href{https://github.com/hcji/KPIC2}{KPIC2} & GitHub\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Isotope labeling using MS}}\\
Analysis of untargeted LC/MS data from stable isotope-labeling experiments. Also uses XCMS for feature detection. & \href{https://github.com/jcapelladesto/geoRge}{geoRge} & GitHub\\
\rowcolor{gray!6}  Correction of MS and MS/MS data from stable isotope labeling (any tracer isotope) experiments for natural isotope abundance and tracer impurity. Separate GUI available in IsoCorrectoRGUI. & \href{https://bioconductor.org/packages/release/bioc/html/IsoCorrectoR.html}{IsoCorrectoR} & BioC\\
Extension of XCMS that provides support for isotopic labeling. Detection of metabolites that have been enriched with isotopic labeling. & \href{http://pattilab.wustl.edu/software/x13cms/x13cms.php}{X13CMS} & NA\\
\rowcolor{gray!6}  Analysis of isotopic patterns in isotopically-labeled MS data. Estimates the isotopic abundance of the stable isotope (either 2H or 13C) within specified compounds. & \href{https://github.com/RuggeroFerrazza/IsotopicLabelling}{IsotopicLabelling} & GitHub\\
Finding the dual (or multiple) isotope labeled analytes using dual labeling of metabolites for metabolome analysis (DLEMMA) approach, described in Liron [42]. & \href{https://cran.r-project.org/web/packages/Miso}{Miso} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Targeted MS}}\\
Peak picking using peak apex intensities for selected masses. Reference library matching, RT/RI conversion plus metabolite identification using multiple correlated masses. Includes GUI. & \href{http://bioconductor.org/packages/2.5/bioc/html/TargetSearch.html}{TargetSearch} & BioC\\
Pre-processing for targeted (SIM) GC-MS data. Guided selection of appropriate fragments for the targets of interest by using an optimization algorithm based on user provided library. & \href{https://doi.org/doi:10.18129/B9.bioc.SIMAT}{SIMAT} & BioC\\
\rowcolor{gray!6}  Deconvolution of MS2 spectra obtained with wide isolation windows. & \href{http://pattilab.wustl.edu/software/decoms2/decoms2.php}{decoMS2} & NA\\
Deconvolution of SWATH-MS experiments to MRM transitions. & \href{http://www.metabolomics-shanghai.org/softwaredetail.php?id=128}{SWATHtoMRM} & NA\\
\rowcolor{gray!6}  Automatic analysis of large scale MRM experiments. & \href{http://www.metabolomics-shanghai.org/softwaredetail.php?id=34}{MRMAnalyzer} & NA\\
Tailors peak detection for targeted metabolites through iterative user interface. It automatically integrates peak areas for all isotopologues and outputs extracted ion chromatograms (EICs). & \href{https://github.com/stevehoang/assayr}{AssayR} & GitHub\\
\rowcolor{gray!6}  Targeted peak picking and annotation. Includes Shiny GUI. & \href{https://github.com/phenomecentre/peakPantheR}{peakPantheR} & GitHub\\
Toolkit for working with Selective Reaction Monitoring (SRM) MS data and other variants of targeted LC-MS data. & \href{https://github.com/wilsontom/sRm}{sRm} & GitHub\\
\rowcolor{gray!6}  Deconvolution of SWATH-MS data. & \href{https://github.com/ZhuMSLab/DecoMetDIA}{DecoMetDIA} & GitHub\\
Targeted peak picking and annotation. All functions through Shiny GUI. & \href{https://github.com/hcji/TarMet}{TarMet} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{GC-MS and GC×GC-MS}}\\
Unsupervised data mining on GC-MS. Clustering of mass spectra to detect compound spectra. The output can be searched in NIST and ARISTO [50]. & \href{https://cran.r-project.org/package=MSeasy}{MSeasy} & CRAN\\
Pre-processing for GC/MS, MassBank search, NIST format export. & \href{https://cran.r-project.org/package=erah}{erah} & CRAN\\
\rowcolor{gray!6}  Pre-processing using AMDIS [53, 54] for untargeted GC-MS analysis. Feature grouping across samples, improved quantification, removal of false positives, normalisation via internal standard or biomass; basic statistics. & \href{https://doi.org/doi:10.18129/B9.bioc.Metab}{Metab} & BioC\\
Deconvolution of GC-MS and GC×GC-MS unit resolution data using orthogonal signal deconvolution (OSD), independent component regression (ICR) and multivariate curve resolution (MCR-ALS). & \href{http://cran.r-project.org/package=osd}{osd} & CRAN\\
\rowcolor{gray!6}  Corrects overloaded signals directly in raw data (from GC-APCI-MS) automatically by using a Gaussian or isotopic-ratio approach. & \href{https://cran.r-project.org/package=CorrectOverloadedPeaks}{CorrectOverloadedPeaks} & CRAN\\
Alignment of GC data. Also GC-FID or any single channel data since it works directly on peak lists. & \href{https://cran.r-project.org/package=GCalignR}{GCalignR} & CRAN\\
\rowcolor{gray!6}  GC-MS data processing and compound annotation pipeline. Includes the building, validating, and query of in-house databases. & \href{https://www.bioconductor.org/packages/release/bioc/html/metaMS.html}{metaMS} & BioC\\
Peak picking for GC×GC-MS using bayes factor and mixture probability models. & \href{http://mrr.sourceforge.net/download.html}{msPeak} & SF\\
\rowcolor{gray!6}  Peak alignment for GC×GC-MS data with homogeneous peaks based on mixture similarity measures. & \href{http://mrr.sourceforge.net/download.html}{mSPA} & SF\\
Peak alignment for GC×GC-MS data with homogeneous and/or heterogenous peaks based on mixture similarity measures. & \href{http://mrr.sourceforge.net/download.html}{SWPA} & SF\\
\rowcolor{gray!6}  Chemometrics analysis GC×GC-MS: baseline correction, smoothing, COW peak alignment, multiway PCA is incorporated. & \href{https://cran.r-project.org/package=RGCxGC}{RGCxGC} & CRAN\\
Retention time and mass spectra similarity threshold-free alignments, seamlessly integrates retention time standards for universally reproducible alignments, performs common ion filtering, and provides compatibility with multiple peak quantification methods. & \href{https://github.com/rramaker/R2DGC}{R2DGC} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Flow injection / direct infusion analysis}}\\
Pre-processing of data from Flow Injection Analysis (FIA) coupled to High-Resolution Mass Spectrometry (HRMS). & \href{https://doi.org/doi:10.18129/B9.bioc.proFIA}{proFIA} & BioC\\
Flow In-jection Electrospray Mass Spectrometry Processing: data processing, classification modelling and variable selection in metabolite fingerprinting & \href{https://github.com/aberHRML/FIEmspro}{FIEmspro} & GitHub\\
\rowcolor{gray!6}  Processing Mass Spectrometry spectrum by using wavelet based algorithm. Can be used for direct infusion experiments. & \href{http://bioconductor.org/packages/release/bioc/html/MassSpecWavelet.html}{MassSpecWavelet} & BioC\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Other}}\\
Filtering of features originating from artifactual interference. Based on the analysis of an extract of E. coli grown in 13C-enriched media. & \href{https://github.com/pattilab/credential}{credential} & GitHub\\
\rowcolor{gray!6}  Wrappers for XCMS and CAMERA. Also includes matching to a spectral library and a GUI. & \href{https://doi.org/doi:10.18129/B9.bioc.metaMS}{metaMS} & BioC\\
Processing of peaktables from AMDIS, XCMS or ChromaTOF. Functions for plotting also provided. & \href{https://doi.org/doi:10.18129/B9.bioc.flagme}{flagme} & BioC\\
\rowcolor{gray!6}  Parametric Time Warping (RT correction) for both DAD and LC-MS. & \href{https://cran.r-project.org/package=ptw}{ptw} & CRAN\\
R wrapper for X!Tandem software for protein identification. & \href{http://bioconductor.org/packages/release/bioc/html/rTANDEM.html}{rTANDEM} & BioC\\
\rowcolor{gray!6}  Building, validation, and statistical analysis of extended assay libraries for SWATH proteomics data. & \href{https://bioconductor.org/packages/release/bioc/html/SwathXtend.html}{SwathXtend} & BioC\\
Split a data set into a set of likely true metabolites and likely measurement artifacts by comparing missing rates of pooled plasma samples and biological samples. & \href{https://cran.r-project.org/package=MetProc}{MetProc} & CRAN\\
\rowcolor{gray!6}  Quality of LC-MS and direct infusion MS data. Generates a report that contains a comprehensive set of quality control metrics and charts. & \href{https://github.com/computational-metabolomics/qcrms}{qcrms} & GitHub\\*
\end{longtable}

\hypertarget{ion-species-grouping-and-annotation}{%
\subsubsection{Ion species grouping and annotation}\label{ion-species-grouping-and-annotation}}

In MS-based metabolomics, the characterisation and identification of metabolites involves several steps and approaches. After peak (feature) table generation, several tools can be used for grouping features that are postulated to originate from the same molecule. These include the widely-used CAMERA for MS1 data, as well as RAMClustR (particularly for DIA data), MetTailor, nontarget, CliqueMS and peakANOVA. Packages that support interpretation of the relationship between the ion species, including adducts, isotopes and in-source fragmentation, are InterpretMSSpectrum, CAMERA, nontarget and mzMatch {[}\protect\hyperlink{ref-scheltema_2011}{38}{]}. See Table 2 for a summary of these packages.

Detailed reconstructed isotope patterns can be used to determine the molecular formula of potential candidates. In the case of molecular formula and isotope analysis, the \emph{m/z} and intensities for a given (set of) features can be used to calculate a ranked list of possible molecular formulas, based on the accurate mass and relative isotope abundances. The Rdisop, GenFormR and enviPat packages are able to simulate and decompose isotopic patterns into molecular formula candidates. Some post processing can calculate e.g.~the double bond equivalents (DBE) and similar characteristics to reduce the number of false positive assignments. Another additional source of information to improve molecular formula estimation is to include MS/MS spectra, as used in MFAssignR, InterpretMSSpectrum or GenFormR.

A typical next step is the annotation of \emph{m/z} with putative metabolites using accurate mass lookup, or if the molecular formula was calculated, lookup of the formula in metabolite databases. It has to be noted that annotation with accurate mass search by no means is equivalent to identification. Under the assumption that all the metabolites measured in a sample have some biochemical relation, a global annotation strategy as used in ProbMetab can help as well. Here, the individual ranked lists of formulae are re-evaluated to also maximize the number of pairs with (potential) biochemical substrate-product pairs. The masstrixR package contains several utility functions for accurate mass lookup. It enables matching of measured \emph{m/z} values against a given database or library and can additionally perform matching based on retention times (RT) and/or collisional cross sections (CCS) if available.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab2}R packages for ion species grouping, annotation, molecular formula generation and accurate mass lookup.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab2}R packages for ion species grouping, annotation, molecular formula generation and accurate mass lookup. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Molecular formula and isotope analysis}}\\
Simulation of and decomposition of Isotopic Patterns. & \href{http://bioconductor.org/packages/release/bioc/html/Rdisop.html}{Rdisop} & BioC\\
Calculation of isotope fine patterns. Also adduct calculations and molecular formula parsing. Web version available at www.envipat.eawag.ch. & \href{https://cran.r-project.org/package=enviPat}{enviPat} & CRAN\\
\rowcolor{gray!6}  Molecular formula assignment, mass recalibration, signal-to-noise evaluation, and unambiguous formula selections are provided. & \href{https://github.com/ChARM-Group/MFAssignR}{MFAssignR} & GitHub\\
Uses GenForm for molecular formula generation on mass accuracy, isotope and/or MS/MS fragments, as well as performing MS/MS subformula annotation. & \href{https://github.com/schymane/GenFormR}{GenFormR} & GitHub\\
\rowcolor{gray!6}  Checking element isotopes, calculating (isotope labelled) exact monoisotopic mass,m/zvalues, mass accuracy, and inspecting possible contaminant mass peaks, examining possible adducts in ESI and MALDI. & \href{https://cran.r-project.org/package=MSbox}{MSbox} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{MS feature grouping}}\\
Grouping of correlated features into pseudo compound spectra using correlation across samples and similarity of peak shape. Annotation of isotopes and adducts. Works as an add-on to XCMS. & \href{https://doi.org/doi:10.18129/B9.bioc.CAMERA}{CAMERA} & BioC\\
\rowcolor{gray!6}  Grouping of features based on similarity between coelution profiles. & \href{https://cran.r-project.org/web/packages/cliqueMS/}{CliqueMS} & CRAN\\
Cluster based feature grouping for non-targeted GC or LC-MS data. & \href{https://github.com/cbroeckl/RAMClustR}{RAMClustR} & CRAN\\
\rowcolor{gray!6}  Uses dynamic block summarisation to group features belong to the same compound. Correction for peak misalignments and isotopic pattern validation. & \href{https://sourceforge.net/projects/mettailor/}{MetTailor} & SF\\
Isotope {\&} adduct peak grouping, homologous series detection. & \href{https://cran.r-project.org/web/packages/nontarget/}{nontarget} & CRAN\\
\rowcolor{gray!6}  Bayesian approach for grouping peaks originating from the same compound. & \href{http://research.cs.aalto.fi/pml/software/peakANOVA/}{peakANOVA} & NA\\
Combination of data from positive and negative ionization mode finding common molecular entities. & \href{https://cran.r-project.org/package=MScombine}{MScombine} & CRAN\\
\rowcolor{gray!6}  Grouping of correlated features into pseudo compound spectra using correlation across sample. Annotation of isotopes and adducts. Can work directly with the XCMS output. & \href{http://www.urr.cat/AStream/AStream.html}{Astream} & NA\\
Navigation of high-resolution MS/MS data in a GUI based on mass spectral similarity. & \href{https://bioconductor.org/packages/release/bioc/html/MetCirc.html}{MetCirc} & BioC\\
\rowcolor{gray!6}  Deconvolution of MS/MS spectra obtained with wide isolation windows. & \href{http://pattilab.wustl.edu/software/decoms2/decoms2.php}{decoMS2} & NA\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Ion/adduct/fragment annotation}}\\
Bayesian probabilistic annotation. & \href{https://github.com/rsilvabioinfo/ProbMetab}{ProbMetab} & GitHub\\
\rowcolor{gray!6}  Isotope {\&} adduct peak grouping, unsupervised homologous series detection. & \href{https://cran.r-project.org/web/packages/nontarget/}{nontarget} & CRAN\\
Automatic interpretation of fragments and adducts in MS spectra. Molecular formula prediction based on fragmentation. & \href{https://cran.r-project.org/package=InterpretMSSpectrum}{InterpretMSSpectrum} & CRAN\\
\rowcolor{gray!6}  Automated annotation using MS2 data or databases and retention time. Calculation of spectral and chemical networks. & \href{https://github.com/WMBEdmands/compMS2Miner}{compMS2Miner} & GitHub\\
Screening, annotation, and putative identification of mass spectral features in lipidomics. Default databases contain {\char`\~}25,000 compounds. & \href{https://doi.org/doi:10.18129/B9.bioc.LOBSTAHS}{LOBSTAHS} & BioC\\
\rowcolor{gray!6}  Automated annotation of fragments from MS and MS2 and putative identification against simulated library fragments of {\char`\~}500,000 lipid species across {\char`\~}60 lipid types. & \href{https://github.com/GarrettLab-UF/LipidMatch}{LipidMatch} & GitHub\\
Annotation of lipid type and acyl groups on independent acquisition-mass spectrometry lipidomics based on fragmentation and intensity rules. & \href{https://cran.r-project.org/package=LipidMS}{LipidMS} & CRAN\\
\rowcolor{gray!6}  Accurate mass and/or retention time and/or collisional cross section matching. & \href{https://github.com/michaelwitting/masstrixR}{masstrixR} & GitHub\\
Downloads KEGG compounds orthology data and wraps the KEGGREST package to extractgene data. & \href{https://cran.r-project.org/package=omu}{omu} & CRAN\\
\rowcolor{gray!6}  Paired mass distance analysis to find independent peaks inm/z-retention time profiles based on retention time hierarchical cluster analysis and frequency analysis of paired mass distances within retention time groups. Structure directed analysis to find potential relationship among those independent peaks. Shiny GUI included. & \href{https://cran.r-project.org/package=pmd}{pmd} & CRAN\\
A Bayesian annotation method for LC/MS data integrating biochemical relations, isotope patterns and adduct formation. & \href{https://github.com/francescodc87/IPA}{IPA} & GitHub\\
\rowcolor{gray!6}  Preprocessing (xcms), replicate merging, noise, blank and missingness filtering, feature grouping, annotation of known compounds, isotopic labeling analysis, annotation from KEGG or HMDB, common biotransformations and probabilistic putative metabolite annotation using MetAssign. & \href{https://github.com/andzajan/mzmatch.R}{mzMatch} & GitHub\\
Putative annotation of unknowns in MS1 data. & \href{https://sourceforge.net/projects/xmsannotator/}{xMSAnnotator} & SF\\*
\end{longtable}

\newpage

\hypertarget{metabolite-identification-with-msms-data}{%
\subsection{Metabolite identification with MS/MS data}\label{metabolite-identification-with-msms-data}}

The annotation of features from MS\textsuperscript{1} experiments alone has limited specificity. Additional structural information for metabolite identification is available from tandem MS and higher-order MS\textsuperscript{n} experiments. There are different approaches, ranging from targeted MS/MS experiments and DDA to DIA (e.g.~MS\textsuperscript{E}, all-ion, broad-band CID, SWATH and other vendor terms). Table 3 provides a summarized overview of R packages for these types of experiments.

\hypertarget{msms-data-handling-spectral-matching-and-clustering}{%
\subsubsection{MS/MS data handling, spectral matching and clustering}\label{msms-data-handling-spectral-matching-and-clustering}}

Generation of high-quality MS/MS spectral libraries and MS/MS data can be a tedious task. It involves wet lab steps of preparing solutions of reference standards as well as creating MS machine-specific acquisition methods. Several steps can be automated using different R packages presented here.

In case of targeted MS/MS, the instrument isolates specific (specified via method files) masses and fragments them is one possibility. Manually writing targeted MS/MS methods from metabolomics data can be tedious if several tens to hundreds of ions need to be fragmented. The MetShot package supports creating targeted method files for some Bruker and Waters instruments. For all other vendors optimized lists of non-overlapping peaks (RT-\emph{m/z} pairs) can be generated to optimize acquisition in the lowest possible number of methods.

In Data dependent acquisition (DDA) the instrument is configured to apply a set of rules which determine which precursor ions are fragmented and MS/MS spectra acquired. DDA approaches also produce a lot of spectra for background peaks or contaminants, which are often of limited use for the purpose of metabolomics studies. Using the RMassBank package, MS1 and MS/MS data can be recalibrated and spectra cleaned of artifacts generated. After database lookup of corresponding identifiers, MassBank records are generated.

In data independent acquisition mode (DIA), the isolation windows are broader, or in some cases, all ions are fragmented, e.g.~the Weizmass library {[}\protect\hyperlink{ref-shahaf_2016}{39}{]} is based on MSE. The computational challenge for DIA data is to deconvolute the MS/MS data and assign the correct precursor ion. DIA data analysis support is currently being implemented in several R packages.

MS/MS spectra can be further processed for example by selecting a representative MS/MS spectrum among all spectra associated with a chromatographic peak or by fusing them into a \emph{consensus} spectrum. Subsequently, spectra can be used in downstream analyses such as spectral matching or clustering. Due to the re-use of infrastructure from the MSnbase package, xcms has recently gained native support for MS/MS data handling and hence allows to extract all MS/MS spectra associated with a feature or chromatographic peak for further processing.

While DDA and DIA are convenient methods, users might miss the accuracy and full control on what is fragmented in the targeted approach. The packages rcdk, MetShot and RMassBank can be combined into a workflow (see {[}\protect\hyperlink{ref-witting_website_nd}{40}{]}) for the generation of records to be uploaded to MS/MS spectral databases (e.g.~MassBank {[}\protect\hyperlink{ref-horai_2010}{41}{]}) or to be used off-line. MetShot allows the user to specify an arbitrary number RT-\emph{m/z} pairs and first sorts them into non-overlapping subsets for which in a second step MS/MS methods (Bruker) or target lists (Agilent, Waters) are generated. It is possible to allow multiple collision energies in a single or separate experiment methods. rcdk was used for calculation of exact masses of adducts. MS/MS data were then acquired on a Bruker maXis plus UHR-Q-ToF-MS. After data collection each run was manually checked for data quality and processed with RMassBank.

Spectral matching of measured MS/MS data with spectral libraries is an important step in metabolite identification. Different possibilities for matching of two spectra exist, ranging from simple cosine similarity and the normalized dot product to X-Rank and proprietary algorithms. In MSnbase, different spectra can be compared. Functions for comparison include the number of common peaks, their correlation, their dot product or alternatively a custom comparison function can be supplied. In addition, it will be possible to import spectra from different file formats such as NIST msp, mgf, and Bruker library to MSnbase objects using the MSnio package. MSnbase therefore seems to be the most flexible R package for the computation of spectral similarities. Spectra are binned before comparison. The OrgMassSpecR package contains a simple cosine spectral matching between two spectra. The two spectra are aligned with each other within a defined \emph{m/z} error window using one spectrum as the reference. The feature-rich compMS2Miner can import msp files and uses the dot product to calculate the spectral similarity, the msPurity package can perform spectral matching using different similarity functions, and MatchWeiz implements the probabilistic X-Rank algorithm {[}\protect\hyperlink{ref-mylonas_2009}{42}{]}.

A growing number of packages, e.g.~LOBSTAHS {[}\protect\hyperlink{ref-collins_2016}{43}{]}, LipidMatch {[}\protect\hyperlink{ref-koelmel_2017}{44}{]} and LipidMS {[}\protect\hyperlink{ref-alcorizabalaguer_2018}{45}{]}, support the annotation of lipids, see Table 2. They use a combination of lipid database lookup, spectral or selected fragment mass matching and \emph{in silico} spectra prediction. To improve disambiguation between lipids of the same species that may only differ in their fatty acid chain composition, they usually rely on identifying specific MS/MS feature masses that are indicative of substructure fragments, such as the lipid headgroup, the headgroup with a certain fatty acid attached, or losses of fatty acid(s), and other modifications, such as oxidation. Additionally, they require certain intensity ratios between characteristic fragments of a lipid in order to identify the lipid species or subspecies.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab3}R packages for MS/MS data.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab3}R packages for MS/MS data. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{MS2 and libraries}}\\
Tools for processing raw data to database ready cleaned spectra with metadata. & \href{https://www.bioconductor.org/packages/release/bioc/html/RMassBank.html}{RMassBank} & BioC\\
From RT-m/z pairs (or m/z alone) creates MS2 experiment files with non-overlapping subsets of the targets. Bruker, Agilent and Waters supported. & \href{https://github.com/sneumann/MetShot}{MetShot} & GitHub\\
\rowcolor{gray!6}  Creating MS libraries from LC-MS data using XCMS/CAMERA packages. A multi-modular annotation function including X-Rank spectral scoring matches experimental data against the generated MS library. & \href{https://github.com/AharoniLab/MatchWeiz}{MatchWeiz} & GitHub\\
Assess precursor contribution to fragment spectrum acquired or anticipated isolation windows using "precursor purity" for both LC-MS(/MS) and DI-MS(/MS) data. Spectral matching against a SQLite database of library spectra. & \href{https://doi.org/doi:10.18129/B9.bioc.msPurity}{msPurity} & BioC\\
\rowcolor{gray!6}  Automated quantification of metabolites by targeting mass spectral/retention time libraries into full scan-acquired GC-MS chromatograms. & \href{https://cran.r-project.org/package=baitmet}{baitmet} & CRAN\\
MS2 spectra similarity and unsupervised statistical methods. Workflow from raw data to visualisations and is interfaceable with XCMS. & \href{https://bioconductor.org/packages/CluMSID/}{CluMSID} & BioC\\
\rowcolor{gray!6}  Import of spectra from different file formats such as NIST msp, mgf (mascot generic format), and library (Bruker) to MSnbase objects. & \href{https://github.com/meowcat/MSnio}{MSnio} & GitHub\\
Multi-purpose mass spectrometry package. Contains many different functions .e.g. isotope pattern calculation, spectrum similarity, chromatogram plotting, reading of msp files and peptide related functions. & \href{https://cran.r-project.org/package=OrgMassSpecR}{OrgMassSpecR} & CRAN\\
\rowcolor{gray!6}  Annotation of LC-MS data based on a database of fragments. & \href{https://cran.r-project.org/package=MetaboList}{MetaboList} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{In silico fragmentation}}\\
In silico fragmentation of candidate structures. & \href{https://github.com/c-ruttkies/MetFragR}{MetFragR} & GitHub\\
\rowcolor{gray!6}  SOLUTIONS for High ReSOLUTION Mass Spectrometry including several functions to interact with MetFrag, developed during the SOLUTIONS project (www.solutions-project.eu). & \href{https://github.com/schymane/ReSOLUTION}{ReSOLUTION} & GitHub\\
Uses MetFrag and adds substructure prediction using the isotopic pattern. Can be trained on a custom dataset. & \href{https://github.com/lucanard/CCC}{CCC} & GitHub\\
\rowcolor{gray!6}  Retention time prediction based on compound structuredescriptors. Five different machine learning algorithms are available to build models. Plotting available to explore chemical space and model quality assessment. & \href{https://github.com/PaoloBnn/Retip}{Retip} & GitHub\\*
\end{longtable}

\hypertarget{reading-of-spectral-databases}{%
\subsubsection{Reading of spectral databases}\label{reading-of-spectral-databases}}

NIST msp files and derived msp-like dialects are a commonly used plain text format for the representation of mass spectra. The msp format described by NIST as part of their Library Conversion Tool {[}\protect\hyperlink{ref-thenationalinstituteofstandardsandtechnology_website_2012}{46}{]} documentation, but has many different dialects due to rather loose format definitions. R packages which support the import and export of this file format are able to both use spectral libraries for identification, as well as to create and enrich spectral libraries with new data.

There are various R packages which support the import of NIST msp files (see Table 3), but the support of different dialects varies, e.g., the NIST-like spectral libraries from RIKEN PRIME {[}\protect\hyperlink{ref-tsugawa_2015}{47}{]} cannot be parsed by some readers. In addition, none of these packages currently supports the import of additional attributes such as `InChIKey:' or `Collision\_energy:' as used in the export of MoNA libraries {[}\protect\hyperlink{ref-massbankofnorthamerica_website_nd}{48}{]}. In essence, most of the packages support the format shown in Listing S1 (see Supplementary File 1, `basic NIST' in Table S1). The metaMS package supports NIST msp files as shown in Listing S2 (see Supplementary File 1, termed `canonical NIST') and RIKEN PRIME provides a similar format with different attributes as shown in Listing S3 (see Supplementary File 1). The packages metaMS, OrgMassSpecR, enviGCMS, and TargetSearch support the export of NIST msp files. The remaining packages partially support the export of results to NIST msp files (see Table S1).

One of the most flexible packages for the handling of NIST msp files is metaMS. This package imports and exports the most attributes, although it does not entirely support generic attributes, and the export is very slow (we observed 20 min for an 8 MB file). In addition, a good library reader should also support mgf (mascot generic format) as available for download from GNPS {[}\protect\hyperlink{ref-wang_2016}{49}{]} as well as other common formats such as the MassBank record format and different vendor library formats such as Bruker (.library, another msp flavour) and Agilent (.cef).

\newpage

\hypertarget{nmr-data-handling-and-pre-processing}{%
\subsection{NMR data handling and (pre-)processing}\label{nmr-data-handling-and-pre-processing}}

NMR is another analytical technique commonly used in metabolomics research. The pre-processing steps for NMR data normally include Fourier transformation, apodization, zero filling, phase and baseline correction and finally referencing and alignment of spectra. Other steps commonly used are removing the areas without any metabolites such as the water region (from 4.7 to 4.9 ppm), as they generally contain no useful information. There are several R packages that can carry out the above tasks (see Table 4). The PepsNMR and speaq are two examples of such R-based packages. The 1D NMR spectra can then be segmented into spectral regions (also known as bins or buckets) subjected directly to statistical data analysis after a normalisation step. The size of the bins could be fixed or variable (adopted or intelligent binning) based on NMR peaks or even each data point from each peak (full data point resolution) used for data analysis. The NMRProcFlow {[}\protect\hyperlink{ref-jacob_2017}{50}{]} package provides a graphical and interactive interface for 1D NMR spectral processing and analysis. Additionally, it provides various spectral alignment methods with the ability to use the corresponding experimental-factor levels in a visual and interactive environment, bridging the gap between experimental design and subsequent statistical analyses. Alternatively peak picking (based on the regions of interest, ROI) can be performed and individual compounds can be identified and integrated prior to statistical analysis. Targeted profiling aims to identify and quantify specific compounds in a sample. The packages that use such approach (ROI) are rDolphin, and rNMR. The bucketed/integrated spectra are normalised to minimise the biological and technical variation. The most common methods are normalisation to a constant sum (e.g.~total sum of integral/bin intensities), probabilistic quotient normalisation {[}\protect\hyperlink{ref-dieterle_2006}{51}{]} and dry weight tissue or protein content.

NMR metabolite annotation uses either chemical shifts and multiplicity matching from existing database, such as Human Metabolome Database {[}\protect\hyperlink{ref-wishart_2009}{52}--\protect\hyperlink{ref-wishart_2018}{55}{]} (HMDB), literature experimental search or uses simulated reference library compounds {[}\protect\hyperlink{ref-beckonert_2007}{56}{]} to match or to fit the existing biological spectra. 1D NMR data often is not sufficient for a confident assignment of the metabolite peaks {[}\protect\hyperlink{ref-pudakalakatti_2014}{57}{]} therefore complementary 2D spectral data acquisition are often required to confirm the assignment {[}\protect\hyperlink{ref-ludwig_2010}{58}{]}. The only package that explicitly deals with 2D NMR is rNMR that takes a targeted approach where the user defines regions of interest to be quantified and compared. DOLPHIN, originally written in MATLAB {[}\protect\hyperlink{ref-gmez_2014}{59}{]}, uses both 1D and 2D NMR data for targeted profiling that is also available as an R version called rDolphin. We are not aware of other R packages that handle 2D NMR data processing. Several general multiway statistical tools such as PARAFAC {[}\protect\hyperlink{ref-shinzawa_2012}{60}{]}, Tucker3 {[}\protect\hyperlink{ref-chen_2018}{61}{]} and MCR have been described {[}\protect\hyperlink{ref-pedersen_2006}{62}{]} that are able to analyse 1D and 2D NMR data, see the section on statistical analysis for a list of packages available for these techniques. BATMAN uses a Bayesian model and some template information such as chemical shifts, \emph{J}-couplings, multiplicity and intensity ratios derived from spectral database to automatically quantify metabolites in a targeted manner {[}\protect\hyperlink{ref-hao_2012}{63}{]}.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab4}R packages for NMR data handling, (pre-)processing and analysis.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab4}R packages for NMR data handling, (pre-)processing and analysis. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Data processing and Analysis}}\\
A tool for processing of 1H NMR data including: Apodization, baseline correction, bucketing, Fourier transformation, warping and phase correction. Bruker FID can be directly imported. & \href{https://github.com/ManonMartin/SOAP-NMR}{PepsNMR} & GitHub\\
Spectra alignment, peak picking based processing, Quantitative analysis and visualizations for 1D NMR. & \href{https://cran.r-project.org/package=speaq}{speaq} & CRAN\\
\rowcolor{gray!6}  Interactive environment based on R-Shiny that includes a complete set of tools to process and visualize 1D NMR spectral data. Processing includes baseline correction, ppm calibration, removal of solvents and contaminants and re-alignment of chemical shifts. & \href{https://www.nmrprocflow.org/}{NMRProcFlow} & Bitbucket\\
TheMetaboMateR toolbox covers basic processing andstatisticalanalysis steps including; several spectral quality assessment (such as dealing with baseline distortions, water suppression to quality assessment of shimming and line width) as well as pre-processing (referencing,baselinecorrection, ... ) to multivariate analysis statistics functions. & \href{https://github.com/kimsche/MetaboMate}{MetaboMate} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Data Analysis and Identification}}\\
Analysis of 1D and 2D NMR spectra using a ROIs based approach. Export to MMCD or uploaded to BMRB for identification. & \href{http://rnmr.nmrfam.wisc.edu/}{rNMR} & NA\\
Pre-processing and identification in an R-based GUI for 1D NMR. & \href{https://github.com/danielcanueto/rDolphin}{rDolphin} & GitHub\\
\rowcolor{gray!6}  Bayesian automated metabolite analyser for 1D NMR spectra. Deconvolution of NMR spectra and automate metabolite quantification. Also identification based on chemical shift lists. & \href{http://batman.r-forge.r-project.org/}{BATMAN} & RF\\
“ASICS: an automatic method for identification and quantification of metabolites in complex 1D 1H NMR spectra.” & \href{https://bioconductor.org/packages/release/bioc/html/ASICS.html}{ASICS} & BioC\\
\rowcolor{gray!6}  ASICSdata: 1D NMR spectra for ASICS. & \href{https://www.bioconductor.org/packages/release/data/experiment/vignettes/ASICSdata/inst/doc/ASICSdata.html}{ASICSdata} & BioC\\
shiny-based interactive NMR data import and Statistical TOtal Correlation SpectroscopY (STOCSY) analyses. & \href{https://cran.r-project.org/package=iSTATS}{iSTATS} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{NMR and integration with Genomics}}\\
MWASTools: an integrated pipeline to perform NMR based metabolome-wide association studies (MWAS). Quality control analysis; MWAS using various models (partial correlations, generalized linear models); visualization of statistical outcomes; metabolite assignment using STOCSY; and biological interpretation of MWAS results. & \href{https://bioconductor.org/packages/release/bioc/html/MWASTools.html}{MWASTools} & BioC\\
An Integrated Suite for Genetic Mapping of Quantitative Variations of 1H NMR-Based Metabolic Profiles. mQTL-NMR provides a complete metabotype quantitative trait locus (mQTL) mapping analysis pipeline for metabolomic data. & \href{https://doi.org/doi:10.18129/B9.bioc.mQTL.NMR}{mQTL.NMR} & BioC\\
\rowcolor{gray!6}  Handles hyperspectral data, i.e. spectra plus further information such as spatial information, time, concentrations, etc. Such data are frequently encountered in Raman, IR, NIR, UV/VIS, NMR, MS, etc. & \href{https://cran.r-project.org/package=ChemoSpec}{hyperSpec} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{uv-data-handling-and-pre-processing}{%
\subsection{UV data handling and (pre-)processing}\label{uv-data-handling-and-pre-processing}}

Another, in metabolomics sometimes under-appreciated, analytical approach is UV absorption detection, usually coupled to an HPLC or UHPLC system. In some cases, the photo-diode array detector (DAD or PDA) is part of an LC-MS system, actually an LC-UV-MS setup. There are other detectors (e.g.~fluorescence) with a different principle, but similar characteristics when it comes to the acquired data. Alignment and baseline correction are typically the first steps of preprocessing LC-UV data. Alignment can be achieved for example with the alsace or the ptw package while baseline correction can be achieved using the hyperSpec, ChemoSpec, mdatools (or the baseline packages). The alsace package provides an alternative to using all channels (wavelengths) by first finding unique components (i.e. ``pure'' spectra) and then performing peak-picking in these components. After alignment general multiway statistical methods like PARAFAC, simultaneous component analysis (SCA), and Tucker Factor Analysis can be applied in the same matter as feature tables would be handled. Table 5 provides an overview of the available R packages for UV data.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab5}R Packages for UV data handling and (pre-)processing.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab5}R Packages for UV data handling and (pre-)processing. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{DAD}}\\
Multivariate Curve Resolution (Alternating Least Squares) for DAD data. & \href{https://github.com/rwehrens/alsace}{alsace} & GitHub\\
Parametric Time Warping (RT correction) for both DAD and LC-MS. & \href{https://cran.r-project.org/package=ptw}{ptw} & CRAN\\
\rowcolor{gray!6}  Handles hyperspectral data, i.e. spectra plus further information such as spatial information, time, concentrations, etc. Such data are frequently encountered in Raman, IR, NIR, UV/VIS, NMR, MS, etc. & \href{https://cran.r-project.org/package=ChemoSpec}{hyperSpec} & CRAN\\
Projection based methods for preprocessing, exploring and analysis of multivariate data. & \href{https://cran.r-project.org/package=mdatools}{mdatools} & CRAN\\
\rowcolor{gray!6}  Collection of baseline correction algorithms, along with a GUI for optimising baseline algorithm parameters. & \href{https://cran.r-project.org/package=baseline}{baseline} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{statistical-analysis-of-metabolomics-data}{%
\subsection{Statistical analysis of metabolomics data}\label{statistical-analysis-of-metabolomics-data}}

Following the feature detection and grouping steps outlined in the sections above, different paths to statistical analysis are available in R and Bioconductor. Once the ``sample versus variable'' feature matrix of molecule intensities or abundances has been generated, comprehensive statistical analyzes can be performed by using the vast range of packages provided by the R statistical software and the Bioconductor project (see Table 6), see for instance StatisticalMethod biocViews {[}\protect\hyperlink{ref-bioconductor_website_ndb}{64}{]} and the ExperimentalDesign {[}\protect\hyperlink{ref-groemping_website_2019}{65}{]}, Cluster {[}\protect\hyperlink{ref-leisch_website_2019}{66}{]}, Multivariate {[}\protect\hyperlink{ref-hewson_website_2018}{67}{]}, MachineLearning {[}\protect\hyperlink{ref-hothorn_website_2019}{68}{]} CRAN Task Views {[}\protect\hyperlink{ref-thecomprehensiverarchivenetwork_website_nda}{69}{]}. As mentioned in the introduction we will only cover common statistical approaches used in metabolomics. Areas such as time-series analysis, clustering methods, machine learning and visualisation of high-dimensional data were dealt with in various books and literature reviews {[}\protect\hyperlink{ref-bishop_2006}{70}--\protect\hyperlink{ref-zaslavsky_2016}{78}{]}.

With regard to statistical analyses in untargeted metabolomics, two strategies can be differentiated that necessitate the use of different methods. The first strategy ``metabolite profiling'' is performed by most untargeted metabolomics studies. Here, a bottom-up approach is taken where sets or classes of pre-defined metabolites are studied usually in different phenotypes of the same biological species and differences in metabolites are usually related to more coarse functional or biological levels (e.g.~to phenotype or to control vs.~treatment in biomedical studies) {[}\protect\hyperlink{ref-hall_2011}{79}{]}. Exploratory data analysis, univariate methods, hierarchical clustering (HCA), Principal Component Analysis (PCA) and Multi-Dimensional Scaling (MDS) like methods are very common in metabolite profiling approaches. Feature/variable selection is performed to find only the most significant metabolite candidates that explain the underlying research question, usually using univariate methods to target only specific metabolites that are interesting to the research question of the study {[}\protect\hyperlink{ref-cai_2017}{80}--\protect\hyperlink{ref-wen_2017}{83}{]}.

The second strategy ``metabolite fingerprinting'' is commonly used in biomedicine, environmental metabolomics and eco-metabolomics to find metabolite patterns across metabolite profiles. Here, metabolites are characterised without necessarily identifying them and characterisation usually occurs from spatiotemporally coarser scales to intrinsic scales within biological species {[}\protect\hyperlink{ref-peters_2018}{84}{]}. Multivariate statistical methods are used that require reduction of high-dimensional data and, thus, ordination methods are commonly applied like (Orthogonal) Partial Least Squares regression (sometimes also coupled to Discriminant Analysis) ((O)PLS(-DA)), (Linear) Discriminant Analysis ((L)DA), and (Canonical) Correspondence Analysis ((C)CA) that allow to relate sets of explanatory variables containing species traits or environmental properties (such as soil type, plant height, smoker/non-smoker, gender, etc.) to the metabolite feature matrix {[}\protect\hyperlink{ref-shaw_2003}{77},\protect\hyperlink{ref-gromski_2014}{85},\protect\hyperlink{ref-legendre_2012}{86}{]}. Other machine learning methods like Random Forests (RF), Support Vector Machines (SVM) and Neural Networks (NN or ANN) are also applicable {[}\protect\hyperlink{ref-clarke_2009}{87}{]}. Lately, untargeted metabolomics data is related to other `omics using network analysis or Procrustes analysis to visualise (dis)similarities between two or more `omics data sets {[}\protect\hyperlink{ref-feng_2018}{88}--\protect\hyperlink{ref-wei_2018}{91}{]}.

Extracting a restricted list of features which still provide a high prediction performance (i.e., a molecular signature) is critical for biomarker validation and clinical diagnostic. Several strategies have been described for feature selection {[}\protect\hyperlink{ref-degenhardt_2019}{92},\protect\hyperlink{ref-saeys_2007}{93}{]} (e.g., wrapper approaches such as Recursive Feature Elimination, Genetic Algorithms, or sparse models such as Lasso, Elastic Net, or sparse PLS). Such techniques are implemented in R packages, which also provide detailed comparisons on real datasets in terms of the stability and the size of the selected signature, the prediction performance of the final model, and the computation time {[}\protect\hyperlink{ref-determanjr_2014}{94}--\protect\hyperlink{ref-wehrens_2012}{97}{]}.

A great number of packages is available to perform statistics on metabolomics datasets. Some of them focus on performing a number of specific tasks, such as sample size estimation, batch normalization, exploratory data analysis, univariate hypothesis testing, multivariate modeling and Omics data integration. Others, listed in the section `Multi workflow steps' in Table 9, adopt a more comprehensive approach, providing statistics toolbox that cover different methods and functionalities.

muma is a package designed to be compatible with MS and NMR generated data. The package mainly focuses on performing statistics. It does not contain functions for data extraction and the user has to provide values arranged in a \emph{data.frame} format. The pre-processing is limited to missing value imputation, noise filtering, variable scaling and normalization. The package also provides tools for outlier detection, univariate and multivariate analysis. Notably, the package offers a script for Statistical TOtal Correlation SpetroscopY (STOCSY) on NMR data.

MOFA proposes tools for the integration of data coming from different omics disciplines (multi-omics). Using factor analysis it allows to calculate hidden factors that capture the biological sample variation across multi-omics datasets, thus allowing marker discovery. MOFA also provide various tools for the visualization of results. IntLIM also supports integration of other omics datasets with metabolomics data by leveraging linear modeling to identify gene-metabolite pairs whose relationship differs from one phenotype to another (e.g.~positive correlation in one phenotype, negative or no correlation in another). IntLIM includes a user-friendly web interface to perform data quality control of input data, identification of phenotype-dependent gene-metabolite pairs, and interactive visualization of results. This tool is particularly useful for integrating transcriptomic and metabolomic or other omics data by generating novel hypothesis in a data-driven manner.

MetaboDiff is presented as an entry level, user friendly package for differential metabolomics analysis. The information contained in the input data (metabolomics measurements and metadata) are stored in S4 objects which are used for the downstream processing. The pre-processing consists of missing value imputation, outlier removal and data normalization, while the data analysis part offers a variety of statistical methods including tools to explore how metabolites relate to each other in sub-pathways.

MetaboAnalystR is a toolbox built over several R packages and contains more than 500 functions organised in eleven modules. The package was created to overcome the limitations of the homonymous web application, such as the possibility of creating flexible customized workflows (including xcms interoperability) and the capacity of dealing with large data sets. MetaboAnalystR functionalities cover a wide range of tools: exploratory statistical analysis, biomarker analysis, power analysis, biomarker meta-analysis, functional enrichment analysis, pathway and joint pathway analysis. Through an implementation of the \emph{mummichog} algorithm {[}\protect\hyperlink{ref-li_2013}{98}{]}, MetaboAnalystR also allows to infer pathways for from user-generated \emph{m/z} peak-lists. Using the MetaboAnalyst knowledgebase, MetaboAnalystR provides access to metabolite set libraries, compound libraries and pathway libraries.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab6}R packages for statistical analysis of metabolomics data.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab6}R packages for statistical analysis of metabolomics data. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Sample size}}\\
Estimate sample sizes for metabolomics experiments, (NMR and targeted approaches supported). & \href{https://cran.r-project.org/package=MetSizeR}{MetSizeR} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Normalization}}\\
Cross-contribution robust multiple standard normalization. Normalization using internal standards. & \href{https://cran.r-project.org/web/packages/crmn/index.html}{crmn} & CRAN\\
\rowcolor{gray!6}  Within and between batch correction of LC-MS metabolomics data using either QC samples or all samples. & \href{https://gitlab.com/CarlBrunius/batchCorr}{batchCorr} & GitLab\\
Normalisation for low concentration metabolites. Mixed model with simultaneous estimation of a correlation matrix. & \href{https://sourceforge.net/projects/metabnorm/}{Metabnorm} & SF\\
\rowcolor{gray!6}  A collection of data distribution normalization methods. & \href{http://quantitativeproteomics.org/normalyzer}{Normalizer} & NA\\
Functions for drift removal and data normalisation based on: component correction, median fold change, ComBat or common PCA (CPCA). & \href{http://b2slab.upc.edu/software-and-downloads/intensity-drift-correction/}{intCor} & NA\\
\rowcolor{gray!6}  Normalisation using a singular value decomposition. & \href{https://sourceforge.net/projects/eigenms/}{EigenMS} & SF\\
Normalization based on RUV-random [164]. & \href{https://cran.r-project.org/web/packages/MetNorm/index.html}{MetNorm} & CRAN\\
\rowcolor{gray!6}  SVR-based normalization and integration for large-scale metabolomics data. & \href{http://www.metabolomics-shanghai.org/softwaredetail.php?id=39}{MetNormalizer} & GitHub\\
Drift correction using QC samples or all study samples. & \href{https://github.com/rwehrens/BatchCorrMetabolomics}{BatchCorrMetabolomics} & GitHub\\
\rowcolor{gray!6}  Signal and Batch Correction for Mass Spectrometry & \href{https://github.com/computational-metabolomics/sbcms}{SBCMS} & GitHub\\
Multiple fitting models to correct intra- and inter-batch effects. & \href{https://cran.r-project.org/package=MetaboQC}{MetaboQC} & CRAN\\
\rowcolor{gray!6}  Collection of functions designed to implement, assess, and choose a suitable normalization method for a given metabolomics study. & \href{https://cran.r-project.org/package=NormalizeMets}{NormalizeMets} & CRAN\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Exploratory Data Analysis}}\\
A large number of methods available for PCA. & \href{http://bioconductor.org/packages/release/bioc/html/pcaMethods.html}{pcaMethods} & BioC\\
\rowcolor{gray!6}  Chemometric analysis of NMR, IR or Raman spectroscopy data. It includes functions for spectral visualisation, peak alignment, HCA, PCA and model-based clustering. & \href{https://cran.r-project.org/package=ChemoSpec}{ChemoSpec} & BioC\\
Joint analysis of MS and MS2 data, where hierarchical cluster analysis is applied to MS2 data to annotate metabolite families and principal component analysis is applied to MS data to discover regulated metabolite families. & \href{https://github.com/ipb-halle/MetFamily}{MetFamily} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Univariate hypothesis testing}}\\
Many methods for corrections for multiple testing. & \href{http://bioconductor.org/packages/release/bioc/html/multtest.html}{multtest} & BioC\\
Estimate tail area-based false discovery rates (FDR) as well as local false discovery rates (fdr) for a variety of null models (p-values, z-scores, correlation coefficients, t-scores). & \href{http://strimmerlab.org/software/fdrtool/}{fdrtool} & CRAN\\
\rowcolor{gray!6}  GUI for statistical analysis using linear mixed models to normalize data and ANOVA to test for treatment effects. & \href{http://metabr.r-forge.r-project.org/}{MetabR} & RF\\
Derives stable estimates of the metabolome-wide significance level within a univariate approach based on a permutation procedure which effectively controls the maximum overall type I error rate at the α level. & \href{https://github.com/AlinaPeluso/MWSL}{MWSL} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Multivariate modeling and feature selection}}\\
Various multivariate methods to analyze metabolomics datasets. Main methods include PCA, Partial Least Squares regression (PLS), and extensions to the PLS like PLS Discriminant Analysis PLS-DA and the orthogonal variants OPLS(-DA). & \href{http://bioconductor.org/packages/release/bioc/html/ropls.html}{ropls} & BioC\\
Package for performing Partial Least Squares regression (PLS). & \href{https://cran.r-project.org/package=pls}{pls} & CRAN\\
\rowcolor{gray!6}  PPCA, PPCCA, MPPCA. & \href{https://cran.r-project.org/package=MetabolAnalyze}{MetabolAnalyze} & CRAN\\
General framework for building regression and classification models. & \href{https://cran.r-project.org/package=caret}{caret} & CRAN\\
\rowcolor{gray!6}  ASCA, figure of merit (FoM), PCA, Goeman’s global test for metabolomic pathways (Q-stat), Penalized Jacobian method (for calculating network connections), Time-lagged correlation method and zero slopes method. It also includes centering and scaling functions. & \href{https://cran.r-project.org/package=MetStaT}{MetStaT} & CRAN\\
RF for the construction, optimization and validation of classification models with the aim of identifying biomarkers. Also normalization, scaling, PCA, MDS. & \href{https://cran.r-project.org/package=RFmarkerDetector}{RFmarkerDetector} & CRAN\\
\rowcolor{gray!6}  PLS-DA, RF, SVM, GBM, GLMNET, PAM. & \href{http://bioconductor.org/packages/release/bioc/html/OmicsMarkeR.html}{OmicsMarkeR} & BioC\\
Recursive feature elimination approach that selects features which significantly contribute to the performance of PLS-DA, Random Forest or SVM classifiers. & \href{http://bioconductor.org/packages/release/bioc/html/biosigner.html}{biosigner} & BioC\\
\rowcolor{gray!6}  Find Biomarkers in two class discrimination problems with variable selection methods provided for several classification metods (lasso/elastic net, PC-LDA, PLS-DA, and several t-tests). & \href{https://cran.r-project.org/web/packages/BioMark/index.html}{BioMark} & CRAN\\
Unsupervised feature extraction specifically designed for analysing noisy and high-dimensional datasets. & \href{https://cran.r-project.org/package=KODAMA}{KODAMA} & CRAN\\
\rowcolor{gray!6}  Non-parametric method for identifying differentially expressed features based on the estimated percentage of false predictions. & \href{https://doi.org/doi:10.18129/B9.bioc.RankProd}{RankProd} & CRAN\\
Fits multi-way component models via alternating least squares algorithms with optional constraints: orthogonal, non-negative, unimodal, monotonic, periodic, smooth, or structure. Fit models include InDScal, PARAFAC, PARAFAC2, SCA, Tucker. & \href{https://cran.r-project.org/package=multiway}{multiway} & CRAN\\
\rowcolor{gray!6}  Decompose a tensor of any order, as a generalisation of SVD also supporting non-identity metrics and penalisations. 2-way SVD is also available. Also includes PCAn (Tucker-n) and PARAFAC/CANDECOMP. & \href{https://cran.r-project.org/package=PTAk}{PTAk} & CRAN\\
Fits multi-way component models via alternating least squares algorithms with optional constraints. Fit models include Individual Differences Scaling, Multiway Covariates Regression, PARAFAC (1 and 2), SCA, and Tucker Factor Analysis. & \href{https://cran.r-project.org/package=ThreeWay}{ThreeWay} & CRAN\\
\rowcolor{gray!6}  Performs variable selection in a multivariate linear model by estimating the covariance matrix of the residuals then use it to remove the dependence that may exist among the responses and eventually performs variable selection by using the Lasso criterion. & \href{https://cran.r-project.org/package=MultiVarSel}{MultiVarSel} & CRAN\\
Performs the O2PLS data integration method for two datasets yielding joint and data-specific parts for each dataset. & \href{https://cran.r-project.org/package=OmicsPLS}{OmicsPLS} & CRAN\\
\rowcolor{gray!6}  Contains ordination methods such as ReDundancy Analysis (RDA), (Canonical or Detrended) Correspondence Analysis (CCA, DCA for binary explanatory variables), (Non-metric) Multi-Dimensional Scaling ((N)MDS) and other univariate and multivariate methods. Originally developed for vegetation ecologists, many functions are also applicable to metabolomics. & \href{https://cran.r-project.org/package=vegan}{vegan} & CRAN\\
Linear and non-linear Discriminant Analysis methods (e.g. LDA), stepwise selection and classification methods useful for feature selection. & \href{https://cran.r-project.org/package=klaR}{klaR} & CRAN\\
\rowcolor{gray!6}  Variable selection methods for PLS, including significance multivariate correlation (SMC), selectivity ratio (SR), variable importance in projections (VIP), loading weights (LW), and regression coefficients (RC). It contains also some other modelling methods. & \href{https://cran.r-project.org/package=plsVarSel}{plsVarSel} & CRAN\\
Predictive multivariate modelling using PLS and Random Forest Data. Repeated double cross unbiased validation and variable selection. & \href{https://gitlab.com/CarlBrunius/MUVR}{MUVR} & GitLab\\
\rowcolor{gray!6}  Biomarker validation for predicting survival. Cross validation methods to validate and select biomarkers when the outcome of interest is survival. & \href{https://cran.r-project.org/package=MetabolicSurv}{MetabolicSurv} & CRAN\\
Pre-treatment, classification, feature selection and correlation analyses of metabolomics data. & \href{https://github.com/jasenfinch/metabolyseR}{metabolyseR} & GitHub\\
\rowcolor{gray!6}  Components search, optimal model components number search, optimal model validity test by permutation tests, observed values evaluation of optimal model parameters and predicted categories, bootstrap values evaluation of optimal model parameters and predicted cross-validated categories. & \href{https://cran.r-project.org/package=packMBPLSDA}{packMBPLSDA} & CRAN\\
Robust identification of time intervals are significantly different between groups. & \href{https://doi.org/doi:10.18129/B9.bioc.OmicsLonDA}{OmicsLonDA} & BioC\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Omics Data integration}}\\
Multiple co-inertia analysis of omics datasets (MCIA) is a multivariate approach for visualization and integration of multi-omics datasets. The MCIA method is not dependent on feature annotation therefore can extract important features even when there are not present across all datasets. & \href{http://bioconductor.org/packages/release/bioc/html/omicade4.html}{omicade4} & BioC\\
STATegRa combines information in multiple omics datasets to evaluate the reproducibility among samples and across experimental condition using component analysis (omicsNPC implements the NonParametric Combination) and clustering. & \href{https://doi.org/doi:10.18129/B9.bioc.STATegRa}{STATegRa} & BioC\\
\rowcolor{gray!6}  Statistical framework supporting many different types of multivariate analyses (e.g. PCA, PLS, CCA, PLS-DA, etc.). & \href{https://cran.r-project.org/web/packages/mixOmics/index.html}{mixOmics} & CRAN\\
STatistics in R Using Class Templates - Classes for building statistical workflows using methods, models and validation objects. Provides mechanism for STATO integration. & \href{http://www.github.com/computational-metabolomics/struct}{STRUCT} & GitHub\\
\rowcolor{gray!6}  Multi-omics base clases integrable with commonly used R Bioconductor objects for omics data; container that holds omics results. & \href{https://bioconductor.org/packages/release/bioc/html/MultiDataSet.html}{MultiDataSet} & BioC\\
Identifies analyte-analyte (e.g. gene-metabolite) pairs whose relationship differs by phenotype (e.g. positive correlation in one phenotype, negative or no correlation in another). The software is also accessible as a user-friendly interface at intlim.bmi.osumc.edu. & \href{https://github.com/mathelab/IntLIM}{IntLIM} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Missing value imputation}}\\
Mixture-model for accounting for data missingness'. & \href{https://doi.org/doi:10.18129/B9.bioc.metabomxtr}{metabomxtr} & BioC\\
Kernel-Based Metabolite Differential Analysis provides a kernel-based score test to cluster metabolites between treatment groups, in order to handle missing values. & \href{https://cran.r-project.org/web/packages/KMDA/}{KMDA} & CRAN\\
\rowcolor{gray!6}  Visualization and imputation of missing values. VIM provides methods for the evaluation and visualization of the type and patterns of missing data. The included imputation approaches are kNN, Hot-Deck, iterative robust model-based imputation (IRMI), fast matching/imputation based on categorical variables and regression imputation. & \href{https://cran.r-project.org/web/packages/VIM/index.html}{VIM} & CRAN\\
Graphical user interface for VIM. & \href{https://cran.r-project.org/web/packages/VIMGUI/VIMGUI.pdf}{VIMGUI} & CRAN\\
\rowcolor{gray!6}  kNN based imputation for microarray data. & \href{https://www.bioconductor.org/packages//2.10/bioc/html/impute.html}{impute} & BioC\\
Bootstrap based algorithm and diagnostics for fast and robust multiple imputation for cross sectional, time series or combined cross sectional and time series data. & \href{https://cran.r-project.org/web/packages/Amelia/Amelia.pdf}{Amelia} & CRAN\\
\rowcolor{gray!6}  Algorithms and diagnostics for the univariate imputation of time series data. & \href{https://cran.r-project.org/package=imputeTS}{imputeTS} & CRAN\\
Methods for the Imputation of incomplete continuous or categorical datasets. missMDA allows missing data imputation using in categorial, continuous or mixed-type datasets using PCA, CA, a multiple correspondence analysis (MCA) model, a multiple factor analysis (MFA) model or factorial analysis for mixed data (FAMD). & \href{https://cran.r-project.org/package=missMDA}{missMDA} & CRAN\\
\rowcolor{gray!6}  Random forest based missing data imputation for mixed-type, nonparametric data. An out-of-bag (OOB) error estimate is used for model optimization. & \href{https://cran.r-project.org/web/packages/missForest/missForest.pdf}{missForest} & CRAN\\
Multivariate imputation by chained equations using fully conditional specifications (FCS) for categorical, continuous and binary datasets. It includes various diagnostic plots for the evaluation of the imputation quality. & \href{https://cran.r-project.org/web/packages/mice/mice.pdf}{mice} & CRAN\\
\rowcolor{gray!6}  Missing data imputation using an approximate Bayesian framework. Diagnostic algorithms are included to analyze the models, the assumptions of the imputation algorithm and the multiply imputed datasets. & \href{https://cran.r-project.org/package=mi}{mi} & CRAN\\
Iterative Gibbs sampler based left-censored missing value imputation. & \href{https://github.com/WandeRum/GSimp}{GSimp} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Multiple workflow steps}}\\
Missing value imputation, filtering, normalisation and averaging of technical replications. & \href{https://sourceforge.net/projects/msprep/}{MSPrep} & SF\\
HCA, Fold change analysis, heat maps, linear models (ordinary and empirical Bayes), PCA and volcano plots. Also log transformation, missing value replacement and methods for normalisation. Cross-contribution compensating multiple internal standard normalisation (ccmn) and remove unwanted variation (ruv2). & \href{https://cran.r-project.org/package=metabolomics}{metabolomics} & CRAN\\
\rowcolor{gray!6}  Data processing, normalization, statistical analysis, metabolite set enrichment analysis, metabolic pathway analysis, and biomarker analysis. & \href{https://github.com/xia-lab/MetaboAnalystR}{MetaboAnalystR} & GitHub\\
Pipeline for metabolomics data pre-processing, with particular focus on data representation using univariate and multivariate statistics. Built on already published functions. & \href{https://cran.r-project.org/web/packages/muma/index.html}{muma} & GitHub\\
\rowcolor{gray!6}  Framework for multiomics experiments. Identifies sources of variability in the experiment and performs additional analysis (identification of subgroups, data imputation, outlier detection). & \href{https://www.bioconductor.org/packages/release/bioc/html/MOFA.html}{MOFA} & BioC\\
Performs entry-level differential analysis on metabolomics data. & \href{https://github.com/andreasmock/MetaboDiff}{MetaboDiff} & GitHub\\
\rowcolor{gray!6}  STRUCT wrappers for filtering, normalisation, missing value imputation, glog transform, HCA, PCA, PLSDA, PLSR, t-test, fold-change, ANOVA, Mixed Effects, post-hoc tests & \href{https://github.com/computational-metabolomics/structToolbox}{STRUCTToolbox} & GitHub\\
Data transformation, filtering of feature and/or samples and data normalization. Quality control processing, statistical analysis and visualization of MS data. & \href{https://github.com/pmartR/pmartR}{pmartR} & GitHub\\
\rowcolor{gray!6}  Quality control, signal drift and batch correction, transformation, univariate hypothesis testing. & \href{https://github.com/ethevenot/phenomis}{phenomis} & GitHub\\
Missing value filtering and imputation, zero value filtering, data normalization, data integration, data quality assessment, univariate statistical analysis, multivariate statistical analysis such as PCA and PLS-D and potential marker selection & \href{https://github.com/MetabLAB/MetCleaning}{MetCleaning} & GitHub\\
\rowcolor{gray!6}  Univariate analysis (linear model), PCA, clustered heatmap, and partial correlation network analysis. Based on classes from the Metabase package(Zhu 2019). & \href{https://github.com/zhuchcn/ShinyMetabase}{ShinyMetabase} & GitHub\\
Outlier detection, PCA, drift correction,visualization, missing value imputation,classification. & \href{https://cran.r-project.org/package=MetabolomicsBasics}{MetabolomicsBasics} & CRAN\\
\rowcolor{gray!6}  Pre-processing, differential compound identification and grouping, traditional PK parameters calculation, multivariate statistical analysis, correlations, cluster analyses and resulting visualization. & \href{https://cran.r-project.org/package=polyPK}{polyPK} & CRAN\\
Integration of omics data using multivariate methods such as PLS. Performs community detection and network analysis to allow visualization of positive or negative associations between different datasets generated using samples from the same individuals. Also available as ashinyapp (https://kuppal.shinyapps.io/xmwas). & \href{https://github.com/kuppal2/xMWAS}{xMWAS} & GitHub\\
\rowcolor{gray!6}  Joint metabolic model-based analysis of metabolomics measurements and taxonomic composition from microbial communities. & \href{https://github.com/borenstein-lab/MIMOSA/tree/master/mimosa}{MIMOSA} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{handling-of-molecule-structures-and-chemical-structure-databases}{%
\subsection{Handling of molecule structures and chemical structure databases}\label{handling-of-molecule-structures-and-chemical-structure-databases}}

Several packages that can deal with cheminformatics tasks, property calculations, metabolite lookup in (web) databases or mapping between databases or structure format conversions (see Table 7).

A well-established package is rcdk which provide a comprehensive subset of functions from the Chemistry Development Kit {[}\protect\hyperlink{ref-willighagen_2017}{99}{]}. rcdk provides a computer readable representation of molecular structures and provide a wealth of functions to import structures from different molecule structure description formats, manipulate structures, visualize structures and calculate properties and molecular fingerprints. The package fingerprint can then be used to compare fingerprints. rinchi provides reading and writing of InChI and InChIKeys {[}\protect\hyperlink{ref-heller_2015}{100}{]}. ChemmineR is an alternative to rcdk, providing many similar functions, with more tools for fingerprints, clustering and others through querying the ChemMine Tools web service {[}\protect\hyperlink{ref-backman_2011}{101}{]}. ChemmineR also has significantly faster parsing of SDF files, which can be an advantage when reading large databases. A large number of additional descriptors are available in the package camb which focuses on quantitative predictive models. ChemmineOB provides conversion between a large number of chemical structure formats using OpenBabel {[}\protect\hyperlink{ref-oboyle_2011}{102}{]}. A notable exception is InChI/InChIKey, which is not directly supported by ChemmineOB or ChemmineR and one would thus have to go through rinchi and rcdk for offline import from InChI to ChemmineR or ChemmineOB. RChemMass is a package that combines the functionality of the rcdk with that of RMassBank, and enviPat. The package RRDKit makes (part of) the functionality of the RDKit {[}\protect\hyperlink{ref-landrum_2016}{103}{]} toolkit available from within R.

A number of existing compound databases are useful for metabolomics. These can supply metadata such as common names and synonyms, database identifiers and experimental or predicted properties. The Rpubchem package provides lookup of information available in PubChem {[}\protect\hyperlink{ref-kim_2016}{104},\protect\hyperlink{ref-wang_2009}{105}{]}, while the webchem package provide query of a large number of databases including PubChem, ChemSpider {[}\protect\hyperlink{ref-pence_2010}{106}{]}, Wikidata {[}\protect\hyperlink{ref-erxleben_2014}{107}{]}, Chemical Translation Service {[}\protect\hyperlink{ref-wohlgemuth_2010}{108}{]}, PHYSPROP {[}\protect\hyperlink{ref-srcinc_website_nd}{109}{]}, Chemical Identifier Resolver {[}\protect\hyperlink{ref-ncicaddgroup_website_nd}{110}{]} and others. BridgeDbR can be used to map identifiers (metabolites, but also genes and proteins, and interactions) between databases, e.g.~PubChem to ChemSpider identifiers; RMassBank and RChemMass also provide some useful web-retrieval functions.

The analysis of identified compounds on the level of substance classes can give biochemical insights which are not obvious from the individual structures, or in case the structures are not fully elucidated. The web tool ClassyFire is able to annotate a given structure with compound classes from their ChemOnt taxonomy as well as different substituents {[}\protect\hyperlink{ref-djoumboufeunang_2016}{111}{]}. The classyfireR package supports the retrieval of substance classes using the RESTful API of the ClassyFire tool based on InChIKeys.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab7}R Packages for molecule structures and chemical structure databases.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab7}R Packages for molecule structures and chemical structure databases. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Structure representation and manipulation}}\\
Subset of functions from the Chemistry Development Kit. Provide a computer readable representation of molecular structures and provide functions to import structures from different molecule structure description formats, manipulate structures, visualize structures and calculate properties and molecular fingerprints. & \href{https://cran.r-project.org/package=rcdk}{rcdk} & CRAN\\
Similar torcdkin functionality and provides more fingerprints and clustering methods and provides additional tools through querying the ChemMine Tools web service. & \href{https://doi.org/doi:10.18129/B9.bioc.ChemmineR}{ChemmineR} & BioC\\
\rowcolor{gray!6}  Provides conversion of structure representation through OpenBabel. & \href{https://doi.org/doi:10.18129/B9.bioc.ChemmineOB}{ChemmineOB} & BioC\\
Exposes functionalities of the RDKit library, including reading and writing of SF files and calculating a few physicochemical properties. & \href{https://github.com/pauca/RRDKit}{RRDKit} & GitHub\\
\rowcolor{gray!6}  Read and write InChI and InChIKey from and torcdk. & \href{https://github.com/rajarshi/cdkr}{rinchi} & GitHub\\
Maximum Common Substructure Searching using ChemmineR structures. & \href{https://doi.org/doi:10.18129/B9.bioc.fmcsR}{FmcsR} & BioC\\
\rowcolor{gray!6}  Basic cheminformatics functions tailored for mass spectrometry applications, enhancing functionality available in other packages likercdk, enviPat, RMassBank etc. & \href{https://github.com/schymane/RChemMass}{RChemMass} & GitHub\\
Provides fingerprinting methods forrcdk. & \href{https://cran.r-project.org/package=fingerprint}{fingerprint} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Database queries}}\\
Calculation of molecular properties. & \href{https://github.com/cambDI/camb}{camb} & GitHub\\
Querying information from PubChem. & \href{https://cran.r-project.org/package=rpubchem}{Rpubchem} & CRAN\\
\rowcolor{gray!6}  Querying information from various web services (CACTUS, CTS, PubChem, ChemSpider) as part of compound list generation. & \href{https://doi.org/doi:10.18129/B9.bioc.RMassBank}{RMassBank} & BioC\\
Querying information from a large number of databases. & \href{https://cran.r-project.org/package=webchem}{webchem} & CRAN\\
\rowcolor{gray!6}  R Interface to the ClassyFire REST API. & \href{https://cran.r-project.org/web/packages/classyfireR/index.html}{classyfireR} & CRAN\\
Allows mapping of identifiers from one database to another, for metabolites, genes, proteins, and interactions. & \href{https://doi.org/10.18129/B9.bioc.BridgeDbR}{BridgeDbR} & BioC\\
\rowcolor{gray!6}  Define utilities for exploration of human metabolome database, including functions to retrieve specific metabolite entries and data snapshots with pairwise associations. & \href{https://doi.org/doi:10.18129/B9.bioc.hmdbQuery}{hmdbQuery} & BioC\\
Parsers for many compound databases including HMDB, MetaCyc, ChEBI, FooDB, Wikidata, WikiPathways, RIKEN respect, MaConDa, T3DB, KEGG, Drugbank, LipidMaps, MetaboLights, Phenol-Explorer, MassBank. & \href{https://github.com/UMCUGenetics/MetaDBparse}{MetaDBparse} & GitHub\\
\rowcolor{gray!6}  Functionality to create and use compound databases generated from (mostly publicly) available resources such as HMDB, ChEBI and PubChem. & \href{https://github.com/EuracBiomedicalResearch/CompoundDb}{CompoundDb} & GitHub\\
Standardized and extensible framework to query chemical and biological databases. & \href{https://github.com/pkrog/biodb}{biodb} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{network-analysis-and-biochemical-pathways}{%
\subsection{Network analysis and biochemical pathways}\label{network-analysis-and-biochemical-pathways}}

The R environment offers packages to analyse networks of metabolomics data and metabolic pathways (see Table 8). Within this section, we refer to `pathway' as a linked series of chemical reactions between molecules, conveyed by enzymes that lead to a product or change in a cell. These molecules are also known as metabolites and transformations occur in the same cellular compartment or in close vicinity. The term `network' refers to the entity of metabolites that are connected biologically, chemically or structurally (e.g.~similarity between MS/MS spectra of two metabolites), functionally or by any other measure (e.g.~statistically correlated).

\hypertarget{network-infrastructure-and-analysis}{%
\subsubsection{Network infrastructure and analysis}\label{network-infrastructure-and-analysis}}

The R environment offers general infrastructure for network analysis. Functionality is implemented in a plethora of software packages, amongst others igraph, tidygraph or the statnet suite. These packages offer functions to generate networks from respective data input (e.g.~adjacency matrices), to analyse networks, calculate network properties and to visualize networks. Generally, any kind of metabolomics data that can be converted to an interpretable format for one of these packages can be analyzed by generic network analysis tools. For example, MSnbase offers functionality to calculate similarity scores between MS/MS spectral data that can be readily interpreted as a spectral similarity network (see {[}\protect\hyperlink{ref-watrous_2012}{112}{]} for the pioneering work of mass spectral molecular networking for biological systems). Such networks can be analysed by the functions provided by the above-mentioned packages or by packages tailored more towards the analysis of biological data (e.g.~RedeR). Specifically interesting for metabolomics applications is DiffCorr, an R package to compare correlation networks from two different experimental conditions, that builds on an association measure such as Pearson's correlation coefficient to identify distinctive properties. DiffCorr enables testing of differential correlation of high-dimensional data sets by identifying the first principal component-based `eigen-molecules' in the correlation networks. DiffCorr then tests these differential correlation values based on Fisher's z-transformation to identify discriminating metabolite pairs that show different response to conditions. Another R package, more tailored towards the analysis of metabolomics data, is BioNetStat, which creates correlation-based networks from metabolite concentration data and analyses the networks based on graph spectra (group of eigenvalues in an adjacency matrix), spectral entropy, degree distribution and node centralities. BioNetStat also allows for KEGG pathway visualization of metabolite data.

\hypertarget{metabolite-annotation}{%
\subsubsection{Metabolite annotation}\label{metabolite-annotation}}

As mentioned above in section 2.2, a major challenge in metabolomics is metabolite annotation, spanning the annotation of known compounds (dereplication) or annotation of unknown metabolites and proposing hypotheses of their structures. Network and pathway analysis can be employed to putatively annotate metabolites in metabolomics data sets. The Bioconductor package MetNet aims at facilitating detection and putative annotation of unknown MS1 features in untargeted metabolomic studies. MetNet infers networks by using an ensemble of statistical associations between intensity values across samples and structural information (mass difference matching between features to a list of enzymatic transformation, retention time adjustment) to infer metabolic networks and guide the annotation of especially specialized metabolites of plant, fungi or bacteria samples. Another package to improve annotation is the package xMSAnnotator that incorporates a multi-criteria scoring algorithm to annotate mass features into different confidence levels. xMSAnnotator uses coelution, pathway level correlations, correlation and KEGG {[}\protect\hyperlink{ref-kanehisa_2017}{113}--\protect\hyperlink{ref-kanehisa_2000}{115}{]}, HMDB, Toxin and Toxin Target Database (T3DB) {[}\protect\hyperlink{ref-lim_2010}{116},\protect\hyperlink{ref-wishart_2015}{117}{]}, LipidMaps {[}\protect\hyperlink{ref-fahy_2007}{118}{]} and ChemSpider {[}\protect\hyperlink{ref-pence_2010}{106}{]} for annotation and incorporates several filter steps, e.g.~by defining modules of co-expressing \emph{m/z} features using WGCNA and a topological overlap-based dissimilarity matrix and thereby categorizing related metabolites into the same network modules.

Molecular networking starting from MS/MS data can enhance the annotation of metabolites. MetDNA, implemented in R, JavaScript and Python (available via a web interface on \url{http://metdna.zhulab.cn}), combines MS1 and MS/MS data to putatively annotate features in metabolomics data sets {[}\protect\hyperlink{ref-shen_2019}{119}{]}. MetDNA uses a metabolic reaction network based recursive algorithm for metabolite annotation employing spectral matching of MS/MS spectra in an automatic fashion. The iterated application of similarity matching between reaction pairs, a substrate metabolite with its product metabolite displaying similar chemical structures, allows the expansion of annotation using seed metabolites or previously annotated metabolites.

MetCirc, designed for the annotation of MS/MS features in untargeted metabolomics data, visualizes the spectral similarity matrix (e.g.~the normalized dot product) between MS/MS spectra in a Circos-like interactive shiny application. Within the shiny application, similarity scores can be thresholded, MS/MS spectra can be interactively explored and annotated based on expert knowledge given the similarity score and displayed spectral features. MetCirc relies on the MSnbase framework to store MS/MS spectral data and to calculate similarities between spectra. Similarly, CluMSID employs spectral similarity matching to guide annotation of MS/MS spectra, incorporates functionality to calculate a correlation networks and for hierarchical and density-based clustering. compMS2Miner is another R package for MS/MS feature annotation and offers functionality for noise filtering, MS/MS substructure annotation, calculation of correlation- and spectral similarity-based networks and interactive visualization.

\hypertarget{generation-of-metabolic-networks}{%
\subsubsection{Generation of metabolic networks}\label{generation-of-metabolic-networks}}

Several R packages implement the functionality to generate metabolic networks. These networks can afterwards be analysed by their topological properties, be used to identify motifs that differ between experimental conditions or queried to find associations between metabolic features. MetaMapR generates metabolic networks by integrating enzymatic transformation, structural similarity between metabolites, mass spectral similarity and empirical correlation information. Hereby, MetaMapR queries biochemical reactions in KEGG and molecular fingerprints for structural similarities in PubChem. Furthermore, MetaMapR aims at incorporating metabolites with unknown biochemistry and unknown structures, and integrates other data sources (genomic, proteomic, clinical data). The package Metabox offers a pipeline for metabolomics data analysis, including functionality for data-driven network construction using correlation, estimation of chemical structure similarity networks using substructure fingerprints. Its statistical analysis highlights metabolites that are altered based on the experimental design group, which can be further interrogated by network and pathway analysis tools. Furthermore, the package MetabNet includes functionality to perform targeted metabolome-wide association studies (MWAS) and to guide the association of unknowns to a specific metabolic pathway, followed by mapping a target metabolite to the metabolic network structure.

\hypertarget{pathway-analysis}{%
\subsubsection{Pathway analysis}\label{pathway-analysis}}

Several R packages enable pathway analysis that uses quantitative data of metabolites and maps these to biological pathways. The Bioconductor package pwOmics analyses proteomics, transcriptomics and other -omics data in combination to highlight molecular mechanisms for single-point and time-series experiments. In downstream analyses, pwOmics allows for pathway, transcription factor and target gene identification.

Another important aspect commonly executed is enrichment analysis to identify pathways that are up- or downregulated given an experimental condition. The R environment offers a whole range of enrichment analysis packages (e.g.~tmod for metabolite data). Targeted more towards pathway analysis, FELLA is a Bioconductor package for enrichment analysis. FELLA detects discriminative metabolic features, maps these to known biological pathways of the KEGG database and detects enriched terms by a diffusion algorithm. CePa offers enrichment analysis tools extending conventional gene set enrichment methods by incorporating pathway topologies. CePa takes nodes rather than terms for analysis and uses network centralities as weight of nodes incorporating pathways from the Pathway Interaction Database (PID, {[}\protect\hyperlink{ref-schaefer_2009}{120}{]}), including NCI/Nature Pathway Interaction , BioCarta {[}\protect\hyperlink{ref-nishimura_2001}{121}{]}, Reactome {[}\protect\hyperlink{ref-fabregat_2018}{122}{]} and KEGG {[}\protect\hyperlink{ref-kanehisa_2017}{113}--\protect\hyperlink{ref-kanehisa_2000}{115}{]}.

MetaboDiff offers functionality to pinpoint to metabolome-wide differences using PCA and t-distributed stochastic neighbor embedding (tSNE) building on the MultiAssayExperiment S4 class. Using t-test or ANOVA, MetaboDiff identified metabolites that differ in their abundance between groups and identifies modules/sub-pathways by using WGCNA that indicate changes in biological pathways. SDAMS (Semi-parametric differential abundance analysis method for proteomics and metabolomics data from mass spectrometry) building upon the SummarizedExperiment S4 class, performs differential abundance analysis on metabolomics data by linking (non-normally distributed) metabolite levels to phenotypic data, containing zero and possibly non-normally distributed non-zero intensity values.

Many R packages guide the discovery of biomarkers for specific phenotypes. Among these is lilikoi, that maps features to pathways by using standardized HMDB IDs, transforms metabolomic profiles to pathway-based profiles using pathway deregulation scores, a measure how much a sample deviates from a normal level, followed by feature selection, classification and prediction. INDEED (INtegrated DiffErential Expression and Differential network analysis) aims to detect biomarkers by performing a differential expression analysis, which is combined with a differential network analysis based on partial correlation and followed by a network topology analysis. Subsequently, activity scores are calculated based on differences detected in the differential expression and the topology of the differential network that will guide the selection of biomarkers. Another R package for biomarker and feature selection is MoDentify which finds regulated modules, groups of correlating molecules that can span from few metabolites to entire pathways, to a given phenotype. These groups are possibly functionally coordinated, coregulated or driven by a similar or same biological process. Score maximization using a multivariable linear regression model with the candidate module as dependent and the phenotype and optional covariates as independent variables identifies the modules. Furthermore, MoDentify implements Gaussian graphical models, where depending on the resolution nodes reflect metabolites or entire pathways.

PAPi (Pathway activity profiling) assigns pathway activity scores to samples to represent the potential pathway activity and statistically detects affected pathways by applying t-test or ANOVA. PAPi uses KEGG pathway identifiers. pathwayPCA, with gene selection in mind, offers multi-omics data analysis by estimating sample-specific pathway activities, e.g.~taken from the rWikiPathways interface. pathwayPCA takes continuous, binary or survival outcomes as input and estimates contributions of individual genes towards pathway significance.

R offers packages to analyze metabolic systems and to estimate biochemical reaction rates in metabolic networks using flux balance analysis, e.g.~BiGGR, abcdeFBA, sybil, and fbar. For example, BiGGR interfaces with the BiGG databases that contains reconstructions of metabolic networks. After importing pathways from the database, flux balance and downstream routines can be performed, e.g.~linear optimization routines or likelihood-based ensembles of calculated flux distributions fitting experimental data.

The package MetaboLouise simulates longitudinal metabolomics data. The simulation builds on a mathematical representation that is parameterized according to underlying biological networks, i.e.~by defining metabolites and relation between them by initializing enzyme rates. Optionally, the package implements functionality to vary the rates depending on the network state, to add external fluxes and to analyze results based on different parameters.

\hypertarget{pathway-resources-and-interfaces}{%
\subsubsection{Pathway resources and interfaces}\label{pathway-resources-and-interfaces}}

A plethora of pathway resources exist, aptly aggregated by Pathguide.org. A number of these resources can be accessed by R packages, which were partly reviewed in {[}\protect\hyperlink{ref-kramer_2014}{123}{]}: rBiopaxParser, graphite, NCIgraph, pathview, KEGGgraph, SBMLR, rsbml, gaggle, and PSICQUIC. Of these, graphite stores pathway information for proteins and metabolites of currently fourteen species (version 1.28.0). Available databases are KEGG, Biocarta, Reactome, NCI/Nature Pathway Interaction Database, HumanCyc, Panther, SMPDB and PharmGKB. graphite offers in addition topological and statistical pathway analysis tools for metabolomics data by interfaces with the Bioconductor packages SPIA and clipper and supports functionality to build own pathways. Furthermore, RPathVisio enables creating and editing biological pathways. RPathVisio enables to visualise data on pathways, to perform statistics on pathway data, and provides an interface to WikiPathways. KEGGREST allows to access the KEGG REST API via a client interface. The package provides utility to search keywords, convert identifiers and link across databases. The package also allows to return amino acid sequences as AAStringSet or nucleotide sequences as DNAStringSet objects (from the Biostrings {[}\protect\hyperlink{ref-tenenbaum_website_2019}{124}{]} package).
Another package, paxtoolsr, provides literature-curated pathway using the Biological Pathway Exchange (BioPAX) format by providing an interface to the Pathway Commons database (including data from the NCI Pathway Interaction Database (PID), PantherDB, HumanCyc, Reactome, PhosphoSitePlus and HPRD). rWikiPathways is an interface between R and WikiPathways.org. Pathways can be queried, interrogated and downloaded to the R session. Furthermore, rWikiPathways associates metabolite information to pathways when providing the system code of a chemical database (e.g.~from HMDB, ChEBI, or ChemSpider).
RaMP provides a relational database of Metabolomics Pathways, integrates pathway, gene, and metabolite annotations from KEGG, HMDB, Reactome, and WikiPathways. The database is downloadable as a standalone MySQL dump, for integration with other software, and is also accessible through an R package, and includes a shiny {[}\protect\hyperlink{ref-chang_website_2012}{125}{]} web interface that supports four basic queries: 1) retrieve analytes (genes of metabolites) given a pathway name; 2) retrieve a pathways for one or more analytes; 3) retrieve analytes involved in the same reaction; 4) retrieve ontologies (cellular location, biofluid locations, etc.) from metabolites. The web interface also supports pathway overrepresentation analysis on genes, metabolites, or genes and metabolites combined (query 3) and includes clustering of significantly enriched pathways according to the percent of overlapping analytes between pathways. Further, the web interface provides network visualization of gene-metabolites relationships (query 4).

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab8}R packages for network analysis and Biochemical pathways.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab8}R packages for network analysis and Biochemical pathways. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Network infrastructure and analysis}}\\
Infrastructure for representation of networks, analysis and visualization. & \href{https://cran.r-project.org/package=igraph}{igraph} & CRAN\\
Infrastructure for representation of networks, analysis and visualization. & \href{https://cran.r-project.org/package=tidygraph}{tidygraph} & CRAN\\
\rowcolor{gray!6}  Infrastructure for representation of networks, analysis and visualization. & \href{https://cran.r-project.org/package=statnet}{statnet} & CRAN\\
Interactive visualization and manipulation of networks. & \href{https://bioconductor.org/packages/release/bioc/html/RedeR.html}{RedeR} & BioC\\
\rowcolor{gray!6}  Comparison of correlation networks from two experiments. & \href{https://cran.r-project.org/package=DiffCorr}{DiffCorr} & CRAN\\
Correlation-based networks from metabolomics data and analysis tools. & \href{https://bioconductor.org/packages/release/bioc/html/BioNetStat.html}{BioNetStat} & BioC\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Annotation}}\\
Putative annotation of unknowns in MS1 data. & \href{http://bioconductor.org/packages/release/bioc/html/MetNet.html}{MetNet} & BioC\\
Putative annotation of unknowns in MS1 data. & \href{https://sourceforge.net/projects/xmsannotator/}{xMSAnnotator} & SF\\
\rowcolor{gray!6}  Putative annotation of unknowns using MS1 and MS2 data. & \href{https://github.com/ZhuMSLab/MetDNA}{MetDNA} & GitHub\\
Visualization of spectral similarity networks, putative annotation of unknowns using MS2 data. & \href{https://bioconductor.org/packages/release/bioc/html/MetCirc.html}{MetCirc} & BioC\\
\rowcolor{gray!6}  Putative annotation of unknowns using MS2 data, clustering of MS2 data. & \href{https://bioconductor.org/packages/devel/bioc/html/CluMSID.html}{CluMSID} & BioC\\
Putative annotation of unknowns using MS2 data. & \href{https://github.com/WMBEdmands/compMS2Miner}{compMS2Miner} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Generation of metabolite networks}}\\
Biochemical reaction networks, spectral and structural similarity networks. & \href{http://dgrapov.github.io/MetaMapR/}{MetaMapR} & GitHub\\
Correlation-based networks, structural similarity networks. & \href{http://kwanjeeraw.github.io/metabox/}{Metabox} & GitHub\\
\rowcolor{gray!6}  Targeted metabolome-wide association studies. & \href{https://sourceforge.net/projects/metabnet/}{MetabNet} & SF\\
Generation of scale-free correlation-based networks. & \href{https://cran.r-project.org/package=WGCNA}{WGCNA} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Pathway analysis}}\\
Analysis of -omics data, pathway, transcription factor and target gene identification. & \href{https://doi.org/doi:10.18129/B9.bioc.pwOmics}{pwOmics} & BioC\\
MSEA a metabolite set enrichment analysis with factor loading in principal component analysis. & \href{https://cran.r-project.org/package=mseapca}{mseapca} & CRAN\\
\rowcolor{gray!6}  Enrichment analysis of a list of affected metabolites. & \href{https://cran.r-project.org/package=tmod}{tmod} & CRAN\\
Network-based enrichment analysis of a list of affected metabolites. & \href{https://bioconductor.org/packages/release/bioc/html/FELLA.html}{FELLA} & BioC\\
\rowcolor{gray!6}  Pathway-based enrichment analysis of a list of affected metabolites. & \href{https://cran.r-project.org/package=CePa}{CePa} & CRAN\\
Differential analysis, modules/sub-pathway identification using networks. & \href{https://github.com/andreasmock/MetaboDiff}{MetaboDiff} & GitHub\\
\rowcolor{gray!6}  Integrates metabolic networks and RNA-seq data to construct condition-specific series of metabolic sub-networks and applies to gene set enrichment analysis & \href{https://cran.r-project.org/package=metaboGSE}{metaboGSE} & CRAN\\
Differential analysis. & \href{https://bioconductor.org/packages/release/bioc/html/SDAMS.html}{SDAMS} & BioC\\
\rowcolor{gray!6}  Biomarker identification. & \href{https://cran.r-project.org/package=lilikoi}{lilikoi} & CRAN\\
Biomarker identification. & \href{https://bioconductor.org/packages/release/bioc/html/INDEED.html}{INDEED} & BioC\\
\rowcolor{gray!6}  Biomarker identification. & \href{https://github.com/krumsieklab/MoDentify}{MoDentify} & GitHub\\
Pathway activity profiling. & \href{https://doi.org/doi:10.18129/B9.bioc.PAPi}{PAPi} & BioC\\
\rowcolor{gray!6}  Pathway activity profiling. & \href{https://www.bioconductor.org/packages/devel/bioc/html/pathwayPCA.html}{pathwayPCA} & BioC\\
Flux balance analysis. & \href{https://bioconductor.org/packages/release/bioc/html/BiGGR.html}{BiGGR} & BioC\\
\rowcolor{gray!6}  Flux balance analysis. & \href{https://cran.r-project.org/package=abcdeFBA}{abcdeFBA} & CRAN\\
Flux balance analysis. & \href{https://cran.r-project.org/package=sybil}{sybil} & CRAN\\
\rowcolor{gray!6}  Flux balance analysis. & \href{https://cran.r-project.org/package=fbar}{fbar} & CRAN\\
Identification of affected pathway from phenotype data (interface with graphite). & \href{https://www.bioconductor.org/packages/release/bioc/html/SPIA.html}{SPIA} & BioC\\
\rowcolor{gray!6}  Identification of affected pathway from phenotype data (interface with graphite). & \href{https://www.bioconductor.org/packages/release/bioc/html/clipper.html}{clipper} & BioC\\
Interface to PathVisio and WikiPathways and pathway analysis and enrichment. & \href{https://github.com/PathVisio/RpathVisio}{RPathVisio} & GitHub\\
\rowcolor{gray!6}  Enrichment analysis of a list of genes and metabolites. & \href{https://github.com/Mathelab/RaMP-DB}{RaMP} & GitHub\\
Simulation of longitudinal metabolomics data based on an underlying biological network & \href{https://cran.r-project.org/package=MetaboLouise}{MetaboLouise} & CRAN\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{Pathway resources and interfaces}}\\
BioPax parser and representation in R. & \href{https://www.bioconductor.org/packages/release/bioc/html/rBiopaxParser.html}{rBiopaxParser} & BioC\\
Interface to KEGG, Biocarta, Reactome, NCI/Nature Pathway Interaction Database, HumanCyc, Panther, SMPDB and PharmGKB. & \href{http://bioconductor.org/packages/release/bioc/html/graphite.html}{graphite} & BioC\\
\rowcolor{gray!6}  Interface to NCI Pathways Database. & \href{https://www.bioconductor.org/packages/release/bioc/html/NCIgraph.html}{NCIgraph} & BioC\\
Interface to KEGG. & \href{https://bioconductor.org/packages/release/bioc/html/pathview.html}{pathview} & BioC\\
\rowcolor{gray!6}  Interface to KEGG. & \href{https://www.bioconductor.org/packages/release/bioc/html/KEGGgraph.html}{KEGGgraph} & BioC\\
Interface to systems biology markup language (SBML). & \href{https://www.bioconductor.org/packages/release/bioc/html/SBMLR.html}{SBMLR} & BioC\\
\rowcolor{gray!6}  Interface to systems biology markup language (SBML). & \href{https://bioconductor.org/packages/release/bioc/html/rsbml.html}{rsbml} & BioC\\
Interface to Gaggle-enabled software (Cytoscape, Firegoose, Gaggle Genome browser). & \href{https://bioconductor.org/packages/release/bioc/html/gaggle.html}{gaggle} & BioC\\
\rowcolor{gray!6}  Interface to molecular interaction databases. & \href{https://www.bioconductor.org/packages/release/bioc/html/PSICQUIC.html}{PSICQUIC} & BioC\\
Interface to KEGG REST server. & \href{http://bioconductor.org/packages/release/bioc/html/KEGGREST.html}{KEGGREST} & BioC\\
\rowcolor{gray!6}  Interface to BioPAX OWL files and the Pathway Commons (PW) molecular interaction database. & \href{http://bioconductor.org/packages/release/bioc/html/paxtoolsr.html}{paxtoolsr} & BioC\\
Interface to WikiPathways. & \href{https://bioconductor.org/packages/release/bioc/html/rWikiPathways.html}{rWikiPathways} & BioC\\
\rowcolor{gray!6}  Database that integrates metabolite and gene biological pathways from HMDB, KEGG, Reactome, and WikiPathways. Includes user-friendly R Shiny web application for queries and pathway enrichment analysis. & \href{https://github.com/Mathelab/RaMP-DB/}{RaMP-DB} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{multifunctional-workflows}{%
\subsection{Multifunctional workflows}\label{multifunctional-workflows}}

When dealing with non-targeted metabolomics data sets, data processing represents a key step for obtaining meaningful and consistent results. While the type and number of data processing methods may vary according to the experimental design and aim of the study, some key steps can be identified that are common for most metabolomics experiments. For this reason, a number of multifunctional R based workflows have been developed over the years. A key advantage of using multifunctional workflows is that most of the functions the user needs are available within the same ``environment'', so that the data does not have to be formatted to comply with functions in other packages. In this respect, a quite common backbone of R workflows consists in performing a pre-processing step that generates an R object that can be used as argument for different functions. Another advantage is that, in most cases, workflows allow a certain degree of flexibility so that functionalities can be used as standalone functions (modular workflows) to better comply with the user's needs. The packages covering larger parts of metabolomics workflows available in R are listed in Table 9.

These multifunctional packages include comprehensive workflows that focus on multiple aspects, such as: data pre-processing, data validation, preliminary statistical analysis and data visualisation of large metabolomics datasets. The considered workflows support both MS based data (LC-MS and GC-MS) and data generated by different analytical platforms. MAIT (Metabolite Automatic Identification Toolkit) offers pre-processing, annotation, statistical analysis and data visualization. It relies on xcms for peak picking and on CAMERA for the preliminary annotation. In addition to CAMERA, the peak annotation process is implemented by including a functionality that allows relating in-source mass losses to specific biotransformations. Human biotransformations are already included, additional biotransformation criteria can be added by the end user. MAIT also provides a number of statistical tools and visual representations (e.g.~PCA, boxplot, PLS) as well as a function to perform identifications using accurate mass search in HMDB. MetMSLine shows some similarities with MAIT in terms of processing stages (xcms-based pre-processing, multivariate statistics, metabolite identifications). Functionalities characterizing MetMSLine include: normalisation, signal drift correction using a smoothing method, noise transformation and outlier removal. SimExTargId is a wrapper of different software and R packages for LC-MS data. It includes tools for data conversion (Proteowizard), peak picking and annotation (xcms and CAMERA), outlier detection and data correction (MetMSLine), and basic statistical analysis. A special feature of SimeExTargId is the real time monitoring of the different workflow stages aimed at metabolomics core facilities; users are notified by email in case of processing errors (e.g.~outlier detection, signal drift). mzMatch is slightly different from the above mentioned workflows and is designed to fit in a broader processing pipeline itself. The project also includes a dedicated file format (peakML) and a Java environment. The different modules can still be used independently. mzMatch supports peak picking and grouping using xcms, reproducibility calculation, data normalization. The peakMonitor app identifies peaks using the local database. The identification is performed on the basis of \emph{m/z} and retention time values with user-defined mass accuracy and retention time deviation values.

MetaDB is built by integrating the metaMS R package into a web application written in Grails. It has also been designed to be integrated with the MetaboLights database. MetaDB supports both LC-MS and GC-MS datasets and offers a wide range of functionalities, including: data storage and metadata management (using the ISA-Tab format and ISACreator tool {[}\protect\hyperlink{ref-roccaserra_2010}{126},\protect\hyperlink{ref-sansone_2012}{127}{]}), peak picking and annotation (via metaMS, an xcms and CAMERA add-on) and QC plots.

MStractor is designed for non-expert users to carry out non-targeted data processing on LC-MS experiments. It gathers xcms and CAMERA functions in an user-friendly pipeline, requiring minimal input and providing graphical QC outputs throughout the workflow. It also includes a manual peak curation step and the possibility of calculating descriptive statistics for each sample class.

patRoon is an interface for different MS-based open source software for non-targeted data processing. patRoon covers different aspects of metabolomics workflows, such as: file conversion to open data formats (mzXML and mzML), feature extraction and grouping (using a number of open software and R packages: xcms, OpenMS, enviPick), extraction of MS and MS/MS data (mzR), component generation (RAMClustR, CAMERA, nontarget), formula calculation (GenForm) and compound identification through automatic annotation of MS/MS spectra (MetFrag and SIRIUS with CSI:FingerID). Other functionalities include (interactive) visualization and reporting of workflow data, comparison and combining results from different workflow algorithms and several data reduction and selection strategies.

Specmine provides a general framework that addresses a variety of different analytical platforms, such as LC-MS, GC-MS, NMR, IR and UV-Vis. The package supports many data formats and includes the possibility of adding metadata in a tabular format. It relies on xcms for LC-MS and GC-MS data pre-processing, on hyperSpec for NMR, IR and UV/VIS data processing and on MAIT for metabolite identification. Specmine provides scripts for missing values imputation, univariate and multivariate statistics and machine learning methods. A number of case studies are available for testing purposes.

mQTL.NMR is a package specific for the systematic analysis of 1H NMR metabolomics in quantitative genetics. The package mainly focuses on NMR spectral data pre-processing (normalization, scaling and peak alignment), mQTL mapping in different model organisms, structural assignment of marker metabolites, and result visualization.

enviMass is a comprehensive workflow for the data-mining of LC-MS and GC-MS datasets, which also supports MS/MS experiments. It provides the user with a graphic interface and a flexible workflow structure covering common processing steps such as data conversion, peak picking, noise removal, peak picking, mass re-calibration, data normalisation, and blank subtraction. It also offers a number of more specific and advanced functionalities including: isotopologue and adduct grouping, homologous series detection and visualization, estimation of atom counts for nontarget components, temporal sequences, profile trend detection and processing of both data dependent and data independent acquisition of MS/MS experiments. RMassScreening is a workflow for batch processing of LC-HRMS datasets using a script interface, YAML-based setting configuration and visual interactive data evaluation. It provides wrappers for script-based usage of enviPick and basic enviMass components, and implements suspect screening and combinatorial prediction of possible metabolites (transformation products) from parent compounds. A GUI provides facilities to analyze the results, grouped by sample groups and experimental timepoints, by applying freely adjustable filters.

MetaboNexus is an interactive data analysis platform for metabolomics experiments, which provides a user friendly R shiny-based GUI designed to work without the need of web server connections. It allows pre-processing (using xcms and MZmine), data scaling, univariate and multivariate statistics (t-test, ANOVA, PCA, PLS-DA, Random Forest, Heatmap), putative metabolite identification (library matching of MS and MS/MS adduct with METLIN, HMDB and MassBank databases), and a number of functions for data visualization.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab9}R packages with multifunctional workflows.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab9}R packages with multifunctional workflows. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  Convenience wrapper for pre-processing tools (XCMS, CAMERA) and a number of statistical analyses. & \href{https://www.bioconductor.org/packages/release/bioc/html/MAIT.html}{MAIT} & BioC\\
Preprocessing (xcms), replicate merging, noise, blank and missingness filtering, feature grouping, annotation of known compounds, isotopic labeling analysis, annotation from KEGG or HMDB, common biotransformations and probabilistic putative metabolite annotation using MetAssign. & \href{https://github.com/andzajan/mzmatch.R}{mzMatch} & GitHub\\
\rowcolor{gray!6}  XCMS and CAMERA based workflow for non-targeted processing of LC-MS datasets, It includes pre-processing, peak picking,peak filtering, data normalization and descriptive statistics calculation. & \href{https://github.com/search?q=MStractor}{MStractor} & GitHub\\
Performs simultaneous raw data to mzXML conversion (MSConvert), peak-picking, automatic PCA outlier detection and statistical analysis, visualization and possible MS2 target list determination during an MS1 metabolomic profiling experiment. & \href{https://github.com/WMBEdmands/simExTargId}{simExTargId} & GitHub\\
\rowcolor{gray!6}  Pre-processing of large LC-MS datasets. Performs automatic PCA with iterative automatic outlier removal and, clustering analysis and biomarker discovery. & \href{https://github.com/WMBEdmands/MetMSLine}{MetMSLine} & GitHub\\
\addlinespace
Workflow for the systematic analysis of 1H NMR metabolomics dataset in quantitative genetics. Performs pre-processing, mQTL mapping, metabolites structural assignment and offers data visualisation tools. & \href{https://www.bioconductor.org/packages/3.5/bioc/html/mQTL.NMR.html}{mQTL.NMR} & BioC\\
\rowcolor{gray!6}  Workflow for pre-processing, qc, annotation and statistical data analysis of LC-MS and GC-MS based metabolomics data to be submitted to public repositories. & \href{https://github.com/rmylonas/MetaDB}{MetaDB} & GitHub\\
Specmine is a framework mainly built on a number of already published packages. It supports data processing form different analytical platforms (LC-MS, GC-MS, NMR, IR, UV-Vis). & \href{https://github.com/cran/specmine}{specmine} & GitHub\\
\rowcolor{gray!6}  Common interface for a number of different MS based data processing software. It covers various aspects, such as data preparation and data extraction, formula calculation, compound identification and reporting. & \href{https://github.com/rickhelmus/patRoon}{patRoon} & GitHub\\
Processing of high resolution of LC-MS data for environmental trend analysis. & \href{https://zenodo.org/record/1213098}{enviMass} & Zenodo\\
\addlinespace
\rowcolor{gray!6}  Workflow for preprocessing of LC-HRMS data, suspect screening, screening for transformation products using combinatorial prediction, and interactive filtering based on ratios between sample groups. & \href{https://github.com/meowcat/RMassScreening}{RMassScreening} & GitHub\\
Workflow to perform pre-processing, statistical analysis and metabolite identifications based on database search of detected spectra. & \href{https://github.com/tohweizhong/MetaboNexus}{MetaboNexus} & GitHub\\
\rowcolor{gray!6}  Shiny-based platform to extract differential features from LC-MS data, includes XCMS-based feature detection, statistical analysis, prediction of molecular formulas, annotation of MS2 spectra, MS2 molecular networking and chemical compound database search. & \href{https://github.com/mjhelf/METABOseek}{METABOseek} & GitHub\\
RShiny interface to Metabolomics packages {\&} MetaboAnalyst scripts. & \href{https://github.com/UMCUGenetics/MetaboShiny/}{MetaboShiny} & GitHub\\
\rowcolor{gray!6}  Preprocessing and visualizing for LC-MS data, as well as statistical analyses, mainly based on univariate linear models. & \href{https://github.com/antonvsdata/amp}{amp} & GitHub\\*
\end{longtable}

\newpage

\hypertarget{user-interfaces-and-workflow-management-systems}{%
\subsection{User interfaces and workflow management systems}\label{user-interfaces-and-workflow-management-systems}}

Visualisation is an important part of data analysis. Traditionally graphics in R have been focussed on creating static plots, while typical explorative studies generally require interactive visualisation to fully investigate the data. User interactions could range from simply zooming in chromatographic or spectroscopic data through to temporarily excluding data from a complex plot for clarity. Several packages in R are available for making interactive plots, e.g.~the Plotly library {[}\protect\hyperlink{ref-plotlytechnologiesinc_website_2015}{128}{]} to create interactive graphics from the static plots generated by the popular plotting framework ggplot2 {[}\protect\hyperlink{ref-wickham_2016}{129}{]}. The use of interactive plots in R is growing, and is helped by an increasing number of code examples available.

Another way interactive plots, and even full GUI tools, are being introduced into R is through the shiny framework, which can create web apps using the full power of R packages as the backend. Many such tools related to metabolomics data analysis are also becoming available, which decreases the learning curve considerably for the typical metabolomics scientist without a computational background. A current gap in the shiny metabolomics landscape are powerful and re-usable widget collections for e.g.~spectra viewers, molecular structures or metabolic networks.

There are several approaches to create, share and use data analysis in R for developers and users, with different strengths and weaknesses. Table 10 summarizes several ways to create and run a data analysis with some interpretation and comparative comments. Note that in some cases it is difficult to quantify ``implementation simplicity'', e.g.~in the case of shiny apps, which can range from rather straightforward to highly complex.

\emph{\textbf{Table 10}: Categorization of creating and sharing R code and data analysis functionality. Symbols indicate strengths (+, ++) or weaknesses (-, --) or neutral (o) assessment.}

\begin{longtable}[]{@{}lcccl@{}}
\toprule
\begin{minipage}[b]{0.07\columnwidth}\raggedright
Framework\strut
\end{minipage} & \begin{minipage}[b]{0.12\columnwidth}\centering
Implementation simplicity low to high\strut
\end{minipage} & \begin{minipage}[b]{0.10\columnwidth}\centering
User- friendliness ow to high\strut
\end{minipage} & \begin{minipage}[b]{0.05\columnwidth}\centering
Interactivity\strut
\end{minipage} & \begin{minipage}[b]{0.52\columnwidth}\raggedright
Example URLs\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.07\columnwidth}\raggedright
R script\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\centering
++\strut
\end{minipage} & \begin{minipage}[t]{0.10\columnwidth}\centering
--\strut
\end{minipage} & \begin{minipage}[t]{0.05\columnwidth}\centering
-\strut
\end{minipage} & \begin{minipage}[t]{0.52\columnwidth}\raggedright
\href{https://gist.github.com/sneumann/013004660f1e6c8e34b9}{write.mzTab}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.07\columnwidth}\raggedright
R Markdown vignette\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\centering
o\strut
\end{minipage} & \begin{minipage}[t]{0.10\columnwidth}\centering
o\strut
\end{minipage} & \begin{minipage}[t]{0.05\columnwidth}\centering
--\strut
\end{minipage} & \begin{minipage}[t]{0.52\columnwidth}\raggedright
\href{https://bioconductor.org/packages/release/bioc/vignettes/xcms/inst/doc/xcms.html}{xcms}, \href{https://github.com/rickhelmus/patRoon}{patRoon}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.07\columnwidth}\raggedright
Jupyter Notebook\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\centering
o\strut
\end{minipage} & \begin{minipage}[t]{0.10\columnwidth}\centering
+\strut
\end{minipage} & \begin{minipage}[t]{0.05\columnwidth}\centering
+\strut
\end{minipage} & \begin{minipage}[t]{0.52\columnwidth}\raggedright
\href{https://mybinder.org/v2/gh/afukushima/MSEAp/master}{MSEAp}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.07\columnwidth}\raggedright
LearnR (CRAN)\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\centering
-\strut
\end{minipage} & \begin{minipage}[t]{0.10\columnwidth}\centering
++\strut
\end{minipage} & \begin{minipage}[t]{0.05\columnwidth}\centering
+\strut
\end{minipage} & \begin{minipage}[t]{0.52\columnwidth}\raggedright
\href{https://rstudio.github.io/learnr/examples.html}{LearnR Examples}\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.07\columnwidth}\raggedright
shiny app\strut
\end{minipage} & \begin{minipage}[t]{0.12\columnwidth}\centering
--\strut
\end{minipage} & \begin{minipage}[t]{0.10\columnwidth}\centering
++\strut
\end{minipage} & \begin{minipage}[t]{0.05\columnwidth}\centering
++\strut
\end{minipage} & \begin{minipage}[t]{0.52\columnwidth}\raggedright
\href{https://msbi.ipb-halle.de/MetFamily/}{MetFamily} and apps in e.g. \href{https://github.com/Mathelab/RaMP-DB}{RaMP-DB}, \href{https://github.com/Mathelab/IntLIM}{IntLIM}\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

All of these environments can be run locally, or installed on a (local or cloud-based) server. Recently, several initiatives have started to provide publicly available computing resources. Examples are e.g.~the previously mentioned \href{https://rdrr.io/}{rdrr.io}, which offers to paste R code into an online console for execution. The console can also be embedded into individual websites. The same project also hosts \href{https://rnotebook.io/}{rnotebook.io}, which allows to create and run R notebooks. The \href{https://shinyapps.io/}{shinyapps.io} platform operated by RStudio Inc has free and paid options to host shiny apps. The binder project (involving members from large academic institutions and companies (like UC Berkeley, Cal Poly San Luis Obispo, Wild Tree Tech Switzerland, Netflix or Simula Research Lab) is an infrastructure to create and use shareable, interactive and reproducible data analysis (not only) with R {[}\protect\hyperlink{ref-jupyter_2018}{130}{]} by taking any GitHub repository, turning it into a Docker image and launching it on a cloud service. The package holepunch {[}\protect\hyperlink{ref-ram_website_nd}{131}{]} simplifies preparing an R project for launching on binder. A public instance is the \href{https://mybinder.org/}{mybinder.org} service providing (limited) resources to execute R based scripts in a hosted Rstudio, Jupyter notebook or applications written with e.g.~shiny. The binder infrastructure code is available on GitHub, so that the service can be offered by universities and research groups to its users, lifting the resource limitations of the public instance.

In some cases an R package can provide bindings to existing tools and libraries written in other languages (see Table 11). This is for example in the case for the packages rcdk or MetFragR using the rJava bindings, or mzR which is a wrapper around the Proteowizard C++ library using the Rcpp package. The fairly new reticulate package provides the corresponding infrastructure to execute Python from R code.

Several workflow systems support workflow nodes and tools that can wrap and execute R code, and in turn build on the huge number of R packages (not only) for metabolomics. In this way, systems like KNIME {[}\protect\hyperlink{ref-liggi_2018}{132},\protect\hyperlink{ref-verhoeven_2018}{133}{]} and Galaxy {[}\protect\hyperlink{ref-davidson_2016}{134},\protect\hyperlink{ref-giacomoni_2015}{135}{]} also provide a graphical user interface and visual programming using the wrapped R functionality, and possibly combine with tools developed in other programming frameworks.

Galaxy is a web-based environment for omics data analysis {[}\protect\hyperlink{ref-goecks_2010}{136}{]}. The \href{http://workflow4metabolomics.org}{Workflow4metabolomics.org} online Galaxy infrastructure dedicated to metabolomics {[}\protect\hyperlink{ref-giacomoni_2015}{135}{]} includes wrappers of xcms, CAMERA, metaMS, proFIA, ropls, biosigner and is open to new contributions. W4M is supported by two national infrastructures: the French Institute of Bioinformatics (\href{http://www.france-bioinformatique.fr/en}{www.france-bioinformatique.fr}) and the Infrastructure for Metabolomics and Fluxomics (\href{https://www.metabohub.fr/}{www.metabohub.fr}) {[}\protect\hyperlink{ref-metabohub_website_2019}{137}{]}. Wrapping R code into a Galaxy module is quite straightforward: examples can be found on the toolshed central repository (\href{https://toolshed.g2.bx.psu.edu/}{toolshed.g2.bx.psu.edu}) and in the RGalaxy bioconductor package. An additional benefit is that the workflow developers need to ensure seamless data flow through the workflow steps, and often contribute the glue code to bridge the gap between objects and data structures that are not always directly compatible across different packages and softwares, thus also improving interoperability beyond the use in workflow systems.

Workflows and input/output data can be publicly referenced {[}\protect\hyperlink{ref-guitton_2017}{138},\protect\hyperlink{ref-workflow4metabolomics_website_nd}{139}{]} on the Workflow4metabolomics platform, thus enabling fully reproducible research. By using workflow systems the reuse and reprocessing of data sets is greatly encouraged, as well as the tracking of data provenance {[}\protect\hyperlink{ref-goble_2019}{140}{]}. This way, workflows help to boost the FAIR principles that were shaped for data {[}\protect\hyperlink{ref-wilkinson_2016}{141}{]}.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab11}Packages to interface R with other languages and workflow environments}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab11}Packages to interface R with other languages and workflow environments \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  Given an R function and its manual page, make the documented function available in Galaxy. & \href{http://bioconductor.org/packages/release/bioc/html/RGalaxy.html}{RGalaxy} & BioC\\
Integration of R and C++. Many R data types and objects can be mapped back and forth to C++ equivalents. & \href{https://cran.r-project.org/package=Rcpp}{Rcpp} & CRAN\\
\rowcolor{gray!6}  Low-Level R to Java Interface. & \href{https://cran.r-project.org/package=rJava}{rJava} & CRAN\\
Interface to 'Python' modules, classes, and functions and translation between R and Python objects. & \href{https://cran.r-project.org/package=reticulate}{reticulate} & CRAN\\*
\end{longtable}

\newpage

\hypertarget{metabolomics-data-sets}{%
\subsection{Metabolomics data sets}\label{metabolomics-data-sets}}

Sharing of data has become increasingly common, and metabolomics data are available from MetaboLights {[}\protect\hyperlink{ref-haug_2013}{142}{]} in the EU, GNPS {[}\protect\hyperlink{ref-wang_2016}{49}{]} and Metabolomics Workbench {[}\protect\hyperlink{ref-sud_2016}{143}{]} in the US. In the context of this review, we focus instead on data in R packages, which is important for development, unit testing, documentation and user training (see Table 12). While there is no difference in R between software and data packages per se, they are handled differently in the Bioconductor infrastructure and separate views exist.

There are several data sets with raw data from LC-MS and flow injection analysis, which can be used by the data pre-processing packages in the previous sections. Other packages contain pre-processed data from GC-MS, LC-MS or NMR in the form of peak tables, which are then typically used in statistics packages, network analysis and other downstream analyses.

\begin{longtable}[t]{>{\raggedright\arraybackslash}p{30em}>{\raggedright\arraybackslash}p{10em}>{\raggedright\arraybackslash}p{3em}}
\caption{\label{tab:tab12}Metabolomics data sets packaged as R packages.}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endfirsthead
\caption[]{\label{tab:tab12}Metabolomics data sets packaged as R packages. \textit{(continued)}}\\
\toprule
Functionalities & Package & Repo\\
\midrule
\endhead
\
\endfoot
\bottomrule
\endlastfoot
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{LC-MS}}\\
12 HPLC-MS NetCDF files (Agilent 1100 LC-MSD SL). & \href{http://bioconductor.org/packages/release/data/experiment/html/faahKO.html}{faahKO} & BioC\\
16 UPLC-MS mzData files (Bruker microTOFq). & \href{http://bioconductor.org/packages/release/data/experiment/html/mtbls2.html}{mtbls2} & BioC\\
\rowcolor{gray!6}  12 UPLC-MS mzML files (AB Sciex TripleTOF 5600, SWATH mode). & \href{https://github.com/sneumann/mtbls297/}{mtbls297} & GitHub\\
Different raw MS files (LTQ, TripleQ, FTICR, Orbitrap, QTOF) some in different formats (mzML, mzXML, mzData, mzData.gz, NetCDF, mz5). Also mzid format from proteomics. & \href{http://bioconductor.org/packages/release/data/experiment/html/msdata.html}{msdata} & BioC\\
\rowcolor{gray!6}  Metadata and DDA MS/MS spectra of 15 narcotics standards (LTQ Orbitrap XL). & \href{https://bioconductor.org/packages/release/data/experiment/html/RMassBankData.html}{RMassBankData} & BioC\\
183 x 109 peak table. & \href{http://bioconductor.org/packages/release/bioc/html/ropls.html}{ropls} & BioC\\
\rowcolor{gray!6}  69 x 5,501 peak table. & \href{http://bioconductor.org/packages/release/bioc/html/biosigner.html}{biosigner} & BioC\\
40 x 1,632 peak table. & \href{https://cran.r-project.org/web/packages/BioMark/index.html}{BioMark} & CRAN\\
\rowcolor{gray!6}  Raw MS files from a set of blanks and standards that contain common environmental contaminants (acquired with Bruker maXis 4G). & \href{https://github.com/rickhelmus/patRoonData}{patRoonData} & GitHub\\
Proteomics, metabolomics GC-MS and Lipidomics data from Calu-3 cell culture; 3 mockulum treated and 9 MERS-CoV treated; Time point, 18 hour from MassIVE dataset ids MSV000079152, MSV000079153, MSV000079154. & \href{https://github.com/pmartR/pmartRdata}{pmartRdata} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{FIA-MS}}\\
6 mzML files (human plasma spiked with 40 compounds acquired in positive mode on an orbitrap fusion). & \href{http://bioconductor.org/packages/release/data/experiment/html/plasFIA.html}{plasFIA} & BioC\\
mzML files (Thermo Exactive) from comparison of leaf tissue from 4 B. distachyon ecotypes with Flow-infusion electrospray ionisation-high resolution mass spectrometry (FIE-HRMS). Also includes data sets with 10 technical injections of human urine and another 10 injections from leaf tissue (ecotype ABR1). & \href{https://github.com/aberHRML/metaboData}{metaboData} & GitHub\\
\rowcolor{gray!6}  \addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{GC-MS}}\\
52 x 154 peak table. & \href{http://bioconductor.org/packages/release/bioc/html/pcaMethods.html}{pcaMethods} & BioC\\
\addlinespace[0.3em]
\multicolumn{3}{l}{\textbf{NMR}}\\
18 x 189 peak table. & \href{https://cran.r-project.org/package=MetabolAnalyze}{MetabolAnalyze} & CRAN\\
\rowcolor{gray!6}  33 x 164 peak table. & \href{https://cran.r-project.org/package=MetabolAnalyze}{MetabolAnalyze} & CRAN\\
ASICSdata: 1D NMR spectra for ASICS. & \href{https://www.bioconductor.org/packages/release/data/experiment/vignettes/ASICSdata/inst/doc/ASICSdata.html}{ASICSdata} & BioC\\*
\end{longtable}

\pagebreak

\hypertarget{conclusions}{%
\section{Conclusions}\label{conclusions}}

This review surveyed both the scientific literature and the R landscape for packages relevant to metabolomics research. While it was very easy to find relevant packages in CRAN and even more so in BioC, many packages are scattered across other source code hosting platforms. While GitHub has a concept of topics (see \href{https://github.com/search?q=topic:metabolomics+topic:r}{github.com/search?q=topic:metabolomics+topic:r}), and crawlers like \href{https://rdrr.io/}{rdrr.io} can find R packages across several platforms, the best findability can be achieved through well-integrated umbrella projects like Bioconductor, which provide additional infrastructure and also improve the community interaction through conferences and workshops.

This also shows the need for more detailed metadata of the R packages allowing easier mixing and matching of packages, noting that Bioconductor already does a very good job. R packages already have a long standing history of metadata annotation via their DESCRIPTION and CITATION files. These provide links to other packages (e.g.~dependencies and suggestions) and literature describing the package. Exposing package and vignette metasdata with semantic approaches will support the community in developing further, more advanced multi-functional workflows for metabolomics. Authors have recently adopted Bioschemas {[}\protect\hyperlink{ref-gray_2017}{144}{]} to make metadata easier findable. For example, efforts to start annotation in vignettes allows the ELIXIR Training eSupport System TeSS (\href{https://tess.oerc.ox.ac.uk}{tess.oerc.ox.ac.uk}) to pick up newer versions (see \href{https://github.com/bridgedb/BridgeDbR/commit/40e741aed77765572e77f84f9fea0e0fb511d3f0}{this git commit} {[}@``attempt to add bioschemas.org json-ld to the vignette html · \href{mailto:bridgedb/bridgedbr@40e741a}{\nolinkurl{bridgedb/bridgedbr@40e741a}} · github''\_n.d.{]}), and efforts are underway to expose content from the DESCRIPTION file as Bioschemas annotation on Bioconductor (see \href{https://github.com/Bioconductor/bioconductor.org/pull/25}{this pull request} {[}@``added template for bioschemas tool annotation by egonw · pull request \#25 · bioconductor/bioconductor.org · github''\_n.d.{]}). These actions greatly contribute to community adoption and encourage the reuse of R-based computational workflows in different use cases {[}\protect\hyperlink{ref-goble_2019}{140}{]}.

In some cases, software described in the literature was only available ``on request'', which in practice often turns out to be not available anymore. This review also did not assess whether the R packages (and their dependencies) can be installed on a current R installation. A recent survey {[}\protect\hyperlink{ref-collberg_2016}{145}{]} showed how the repeatability of papers using scientific software drops when software is not available or does not install. Issues/bug reports were filed for packages that were found that were not able to be tested on contemporary operating systems. The way out of the (un-)repeatability trap can be expressed in very few, seemingly trivial, rules {[}\protect\hyperlink{ref-taschuk_2017}{146}{]} and hosting the code in the open repositories, if possible with regular builds or even Continuous Integration. As discussed earlier, the metabolomics packages have tighter connections in an established community such as Bioconductor, rather than in other package repositories. In the last few years, Bioconductor packages for metabolomics and proteomics data analysis started converging towards a common mass spectrometry infrastructure, which simplifies interoperability between these packages. Based on experiences from these efforts, the RforMassSpectrometry (\href{https://rformassspectrometry.org}{RforMassSpectrometry.org}) initiative was recently started aiming at providing efficient, thoroughly documented, tested and flexible R software for MS data import, handling and analysis. Significant improvements can thus be expected in the future, simplifying and unifying MS data handling for the benefit of the end users. RforMassSpectrometry also contains the metaRbolomics-book {[}\protect\hyperlink{ref-stanstrup_website_nd}{147}{]}, which will be a continuously developed resource with additional examples beyond this review.

The authors expect that the metaRbolomics landscape will continue its steady growth rate and keep track of the evolving metabolomics experiments to come.

\newpage

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\leavevmode\hypertarget{ref-emwas_2019}{}%
1. Emwas, A.-H.; Roy, R.; McKay, R.T.; Tenori, L.; Saccenti, E.; Gowda, G.A.N.; Raftery, D.; Alahmari, F.; Jaremko, L.; Jaremko, M. et al. NMR spectroscopy for metabolomics research. \emph{Metabolites} \textbf{2019}, \emph{9}.

\leavevmode\hypertarget{ref-lammerhofer_2013}{}%
2. \emph{Metabolomics in practice: Successful strategies to generate and analyze metabolic data}; Lämmerhofer, M., Weckwerth, W., Eds.; Wiley-VCH Verlag GmbH \& Co. KGaA: Weinheim, Germany, 2013; ISBN \href{https://worldcat.org/isbn/9783527330898}{9783527330898}.

\leavevmode\hypertarget{ref-villasboas_2007}{}%
3. Villas-Boas, S.G.; Nielsen, J.; Smedsgaard, J.; Hansen, M.A.E.; Roessner-Tunali, U. \emph{Metabolome analysis: An introduction}; 1st ed.; Wiley, John \& Sons, Incorporated, 2007; p. 319; ISBN \href{https://worldcat.org/isbn/978-0-471-74344-6}{978-0-471-74344-6}.

\leavevmode\hypertarget{ref-wehrens_2019}{}%
4. \emph{Metabolomics: Practical guide to design and analysis}; Wehrens, R., Salek, R., Eds.; Chapman \& hall/CRC mathematical and computational biology; Chapman; Hall/CRC, 2019; ISBN \href{https://worldcat.org/isbn/1498725260}{1498725260}.

\leavevmode\hypertarget{ref-InternationalMetabolomicsSociety_2019}{}%
5. International Metabolomics Society Free Tools \& Learning Resources - Metabolomics Society Wiki.

\leavevmode\hypertarget{ref-salek_website_nd}{}%
6. Salek, R.; Emery, L.; Beisken, S. Metabolomics: An introduction EMBL-EBI train online.

\leavevmode\hypertarget{ref-rcoredevelopmentteam_website_2018}{}%
7. R Core Development Team R: A language and environment for statistical computing 2018.

\leavevmode\hypertarget{ref-spicer_website_2018}{}%
8. Spicer, R. GitHub - RASpicer/MetabolomicsTools 2018.

\leavevmode\hypertarget{ref-spicer_2017}{}%
9. Spicer, R.; Salek, R.M.; Moreno, P.; Cañueto, D.; Steinbeck, C. Navigating freely-available software tools for metabolomics analysis. \emph{Metabolomics : Official journal of the Metabolomic Society} \textbf{2017}, \emph{13}, 106.

\leavevmode\hypertarget{ref-misra_2016}{}%
10. Misra, B.B.; Hooft, J.J.J. van der Updates in metabolomics tools and resources: 2014-2015. \emph{Electrophoresis} \textbf{2016}, \emph{37}, 86--110.

\leavevmode\hypertarget{ref-misra_2017}{}%
11. Misra, B.B.; Fahrmann, J.F.; Grapov, D. Review of emerging metabolomic tools and resources: 2015-2016. \emph{Electrophoresis} \textbf{2017}, \emph{38}, 2257--2274.

\leavevmode\hypertarget{ref-misra_2018}{}%
12. Misra, B.B. New tools and resources in metabolomics: 2016-2017. \emph{Electrophoresis} \textbf{2018}, \emph{39}, 909--923.

\leavevmode\hypertarget{ref-misra_website_2017}{}%
13. Misra, B. GitHub - biswapriyamisra/metabolomics: Tools databases resources in metabolomics \& integrated omics in 2015-2016 2017.

\leavevmode\hypertarget{ref-kannan_2016}{}%
14. Kannan, L.; Ramos, M.; Re, A.; El-Hachem, N.; Safikhani, Z.; Gendoo, D.M.A.; Davis, S.; Gomez-Cabrero, D.; Castelo, R.; Hansen, K.D. et al. Public data and open source tools for multi-assay genomic investigation of disease. \emph{Briefings in Bioinformatics} \textbf{2016}, \emph{17}, 603--615.

\leavevmode\hypertarget{ref-blaenovi_2018}{}%
15. Blaženović, I.; Kind, T.; Ji, J.; Fiehn, O. Software tools and approaches for compound identification of LC-MS/MS data in metabolomics. \emph{Metabolites} \textbf{2018}, \emph{8}.

\leavevmode\hypertarget{ref-mullen_website_2019}{}%
16. Mullen, K. CRAN task view: Chemometrics and computational physics 2019.

\leavevmode\hypertarget{ref-gentleman_2004}{}%
17. Gentleman, R.C.; Carey, V.J.; Bates, D.M.; Bolstad, B.; Dettling, M.; Dudoit, S.; Ellis, B.; Gautier, L.; Ge, Y.; Gentry, J. et al. Bioconductor: Open software development for computational biology and bioinformatics. \emph{Genome Biology} \textbf{2004}, \emph{5}, R80.

\leavevmode\hypertarget{ref-bioconductor_website_nda}{}%
18. Bioconductor Bioconductor - BiocViews.

\leavevmode\hypertarget{ref-thecomprehensiverarchivenetwork_website_nd}{}%
19. The Comprehensive R Archive Network CRAN repository policy.

\leavevmode\hypertarget{ref-bioconductor_website_nd}{}%
20. Bioconductor Bioconductor - developers.

\leavevmode\hypertarget{ref-theul_2009}{}%
21. Theußl, S.; Zeileis, A. Collaborative software development using r-forge. \emph{The R journal} \textbf{2009}, \emph{1}, 9.

\leavevmode\hypertarget{ref-boettiger_2015}{}%
22. Boettiger, C.; Chamberlain, S.; Hart, E.; Ram, K. Building software, building community: Lessons from the rOpenSci project. \emph{Journal of open research software} \textbf{2015}, \emph{3}.

\leavevmode\hypertarget{ref-devries_website_2015b}{}%
23. Vries, A. de; Rickert, J. The network structure of r packages on CRAN \& BioConductor 2015.

\leavevmode\hypertarget{ref-devries_website_2015a}{}%
24. Vries, A. de Differences in the network structure of CRAN and BioConductor (revolutions) 2015.

\leavevmode\hypertarget{ref-devries_website_2015}{}%
25. Vries, A. de GitHub - andrie/cran-network-structure: Scripts used for my UseR!2015 presentation on the network structure of CRAN 2015.

\leavevmode\hypertarget{ref-neumann_website_2019}{}%
26. Neumann, S. GitHub - sneumann/metaRbolomics: Metabolomics in r and bioconductor 2019.

\leavevmode\hypertarget{ref-chambers_2012}{}%
27. Chambers, M.C.; Maclean, B.; Burke, R.; Amodei, D.; Ruderman, D.L.; Neumann, S.; Gatto, L.; Fischer, B.; Pratt, B.; Egertson, J. et al. A cross-platform toolkit for mass spectrometry and proteomics. \emph{Nature Biotechnology} \textbf{2012}, \emph{30}, 918--920.

\leavevmode\hypertarget{ref-kessner_2008}{}%
28. Kessner, D.; Chambers, M.; Burke, R.; Agus, D.; Mallick, P. ProteoWizard: Open source software for rapid proteomics tools development. \emph{Bioinformatics} \textbf{2008}, \emph{24}, 2534--2536.

\leavevmode\hypertarget{ref-fuhrer_2011}{}%
29. Fuhrer, T.; Heer, D.; Begemann, B.; Zamboni, N. High-throughput, accurate mass metabolome profiling of cellular extracts by flow injection-time-of-flight mass spectrometry. \emph{Analytical Chemistry} \textbf{2011}, \emph{83}, 7074--7080.

\leavevmode\hypertarget{ref-mahieu_2016a}{}%
30. Mahieu, N.G.; Genenbacher, J.L.; Patti, G.J. A roadmap for the XCMS family of software solutions in metabolomics. \emph{Current Opinion in Chemical Biology} \textbf{2016}, \emph{30}, 87--93.

\leavevmode\hypertarget{ref-smith_2006}{}%
31. Smith, C.A.; Want, E.J.; O'Maille, G.; Abagyan, R.; Siuzdak, G. XCMS: Processing mass spectrometry data for metabolite profiling using nonlinear peak alignment, matching, and identification. \emph{Analytical Chemistry} \textbf{2006}, \emph{78}, 779--787.

\leavevmode\hypertarget{ref-tautenhahn_2008}{}%
32. Tautenhahn, R.; Böttcher, C.; Neumann, S. Highly sensitive feature detection for high resolution LC/MS. \emph{BMC Bioinformatics} \textbf{2008}, \emph{9}, 504.

\leavevmode\hypertarget{ref-conley_2014}{}%
33. Conley, C.J.; Smith, R.; Torgrip, R.J.O.; Taylor, R.M.; Tautenhahn, R.; Prince, J.T. Massifquant: Open-source kalman filter-based XC-MS isotope trace feature detection. \emph{Bioinformatics} \textbf{2014}, \emph{30}, 2636--2643.

\leavevmode\hypertarget{ref-huber_2015}{}%
34. Huber, W.; Carey, V.J.; Gentleman, R.; Anders, S.; Carlson, M.; Carvalho, B.S.; Bravo, H.C.; Davis, S.; Gatto, L.; Girke, T. et al. Orchestrating high-throughput genomic analysis with bioconductor. \emph{Nature Methods} \textbf{2015}, \emph{12}, 115--121.

\leavevmode\hypertarget{ref-martinmorgan_2017}{}%
35. Martin Morgan, V.O. SummarizedExperiment. \emph{Bioconductor} \textbf{2017}.

\leavevmode\hypertarget{ref-zhu_website_2019}{}%
36. Zhu, C. Zhuchcn/metabase: A r package to store, manipulate, analyze, and visualize metabolomics data 2019.

\leavevmode\hypertarget{ref-hoffmann_2019}{}%
37. Hoffmann, N.; Rein, J.; Sachsenberg, T.; Hartler, J.; Haug, K.; Mayer, G.; Alka, O.; Dayalan, S.; Pearce, J.T.M.; Rocca-Serra, P. et al. mzTab-m: A data standard for sharing quantitative results in mass spectrometry metabolomics. \emph{Analytical Chemistry} \textbf{2019}, \emph{91}, 3302--3310.

\leavevmode\hypertarget{ref-scheltema_2011}{}%
38. Scheltema, R.A.; Jankevics, A.; Jansen, R.C.; Swertz, M.A.; Breitling, R. PeakML/mzMatch: A file format, java library, r library, and tool-chain for mass spectrometry data analysis. \emph{Analytical Chemistry} \textbf{2011}, \emph{83}, 2786--2793.

\leavevmode\hypertarget{ref-shahaf_2016}{}%
39. Shahaf, N.; Rogachev, I.; Heinig, U.; Meir, S.; Malitsky, S.; Battat, M.; Wyner, H.; Zheng, S.; Wehrens, R.; Aharoni, A. The WEIZMA spectral library for high-confidence metabolite identification. \emph{Nature Communications} \textbf{2016}, \emph{7}, 12423.

\leavevmode\hypertarget{ref-witting_website_nd}{}%
40. Witting, M. GitHub - michaelwitting/ms2dbworkflow.

\leavevmode\hypertarget{ref-horai_2010}{}%
41. Horai, H.; Arita, M.; Kanaya, S.; Nihei, Y.; Ikeda, T.; Suwa, K.; Ojima, Y.; Tanaka, K.; Tanaka, S.; Aoshima, K. et al. MassBank: A public repository for sharing mass spectral data for life sciences. \emph{Journal of Mass Spectrometry} \textbf{2010}, \emph{45}, 703--714.

\leavevmode\hypertarget{ref-mylonas_2009}{}%
42. Mylonas, R.; Mauron, Y.; Masselot, A.; Binz, P.-A.; Budin, N.; Fathi, M.; Viette, V.; Hochstrasser, D.F.; Lisacek, F. X-rank: A robust algorithm for small molecule identification using tandem mass spectrometry. \emph{Analytical Chemistry} \textbf{2009}, \emph{81}, 7604--7610.

\leavevmode\hypertarget{ref-collins_2016}{}%
43. Collins, J.R.; Edwards, B.R.; Fredricks, H.F.; Van Mooy, B.A.S. LOBSTAHS: An adduct-based lipidomics strategy for discovery and identification of oxidative stress biomarkers. \emph{Analytical Chemistry} \textbf{2016}, \emph{88}, 7154--7162.

\leavevmode\hypertarget{ref-koelmel_2017}{}%
44. Koelmel, J.P.; Kroeger, N.M.; Ulmer, C.Z.; Bowden, J.A.; Patterson, R.E.; Cochran, J.A.; Beecher, C.W.W.; Garrett, T.J.; Yost, R.A. LipidMatch: An automated workflow for rule-based lipid identification using untargeted high-resolution tandem mass spectrometry data. \emph{BMC Bioinformatics} \textbf{2017}, \emph{18}, 331.

\leavevmode\hypertarget{ref-alcorizabalaguer_2018}{}%
45. Alcoriza-Balaguer, M.I.; García-Cañaveras, J.C.; Lopez, A.; Conde, I.; Juan, O.; Carretero, J.; Lahoz, A. LipidMS: An r package for lipid annotation in untargeted liquid chromatography-data independent acquisition-mass spectrometry lipidomics. \emph{Analytical Chemistry} \textbf{2018}, \emph{91}, 836--845.

\leavevmode\hypertarget{ref-thenationalinstituteofstandardsandtechnology_website_2012}{}%
46. Standards, T.N.I. of; Technology Library conversion tool 2012.

\leavevmode\hypertarget{ref-tsugawa_2015}{}%
47. Tsugawa, H.; Cajka, T.; Kind, T.; Ma, Y.; Higgins, B.; Ikeda, K.; Kanazawa, M.; VanderGheynst, J.; Fiehn, O.; Arita, M. MS-DIAL: Data-independent MS/MS deconvolution for comprehensive metabolome analysis. \emph{Nature Methods} \textbf{2015}, \emph{12}, 523--526.

\leavevmode\hypertarget{ref-massbankofnorthamerica_website_nd}{}%
48. MassBank of North America MoNA downloads.

\leavevmode\hypertarget{ref-wang_2016}{}%
49. Wang, M.; Carver, J.J.; Phelan, V.V.; Sanchez, L.M.; Garg, N.; Peng, Y.; Nguyen, D.D.; Watrous, J.; Kapono, C.A.; Luzzatto-Knaan, T. et al. Sharing and community curation of mass spectrometry data with global natural products social molecular networking. \emph{Nature Biotechnology} \textbf{2016}, \emph{34}, 828--837.

\leavevmode\hypertarget{ref-jacob_2017}{}%
50. Jacob, D.; Deborde, C.; Lefebvre, M.; Maucourt, M.; Moing, A. NMRProcFlow: A graphical and interactive tool dedicated to 1D spectra processing for NMR-based metabolomics. \emph{Metabolomics : Official journal of the Metabolomic Society} \textbf{2017}, \emph{13}, 36.

\leavevmode\hypertarget{ref-dieterle_2006}{}%
51. Dieterle, F.; Ross, A.; Schlotterbeck, G.; Senn, H. Probabilistic quotient normalization as robust method to account for dilution of complex biological mixtures. Application in 1H NMR metabonomics. \emph{Analytical Chemistry} \textbf{2006}, \emph{78}, 4281--4290.

\leavevmode\hypertarget{ref-wishart_2009}{}%
52. Wishart, D.S.; Knox, C.; Guo, A.C.; Eisner, R.; Young, N.; Gautam, B.; Hau, D.D.; Psychogios, N.; Dong, E.; Bouatra, S. et al. HMDB: A knowledgebase for the human metabolome. \emph{Nucleic Acids Research} \textbf{2009}, \emph{37}, D603--10.

\leavevmode\hypertarget{ref-wishart_2013}{}%
53. Wishart, D.S.; Jewison, T.; Guo, A.C.; Wilson, M.; Knox, C.; Liu, Y.; Djoumbou, Y.; Mandal, R.; Aziat, F.; Dong, E. et al. HMDB 3.0--the human metabolome database in 2013. \emph{Nucleic Acids Research} \textbf{2013}, \emph{41}, D801--7.

\leavevmode\hypertarget{ref-wishart_2007}{}%
54. Wishart, D.S.; Tzur, D.; Knox, C.; Eisner, R.; Guo, A.C.; Young, N.; Cheng, D.; Jewell, K.; Arndt, D.; Sawhney, S. et al. HMDB: The human metabolome database. \emph{Nucleic Acids Research} \textbf{2007}, \emph{35}, D521--6.

\leavevmode\hypertarget{ref-wishart_2018}{}%
55. Wishart, D.S.; Feunang, Y.D.; Marcu, A.; Guo, A.C.; Liang, K.; Vázquez-Fresno, R.; Sajed, T.; Johnson, D.; Li, C.; Karu, N. et al. HMDB 4.0: The human metabolome database for 2018. \emph{Nucleic Acids Research} \textbf{2018}, \emph{46}, D608--D617.

\leavevmode\hypertarget{ref-beckonert_2007}{}%
56. Beckonert, O.; Keun, H.C.; Ebbels, T.M.D.; Bundy, J.; Holmes, E.; Lindon, J.C.; Nicholson, J.K. Metabolic profiling, metabolomic and metabonomic procedures for NMR spectroscopy of urine, plasma, serum and tissue extracts. \emph{Nature Protocols} \textbf{2007}, \emph{2}, 2692--2703.

\leavevmode\hypertarget{ref-pudakalakatti_2014}{}%
57. Pudakalakatti, S.M.; Dubey, A.; Jaipuria, G.; Shubhashree, U.; Adiga, S.K.; Moskau, D.; Atreya, H.S. A fast NMR method for resonance assignments: Application to metabolomics. \emph{Journal of Biomolecular NMR} \textbf{2014}, \emph{58}, 165--173.

\leavevmode\hypertarget{ref-ludwig_2010}{}%
58. Ludwig, C.; Viant, M.R. Two-dimensional j-resolved NMR spectroscopy: Review of a key methodology in the metabolomics toolbox. \emph{Phytochemical Analysis} \textbf{2010}, \emph{21}, 22--32.

\leavevmode\hypertarget{ref-gmez_2014}{}%
59. Gómez, J.; Brezmes, J.; Mallol, R.; Rodríguez, M.A.; Vinaixa, M.; Salek, R.M.; Correig, X.; Cañellas, N. Dolphin: A tool for automatic targeted metabolite profiling using 1D and 2D (1)H-NMR data. \emph{Analytical and Bioanalytical Chemistry} \textbf{2014}, \emph{406}, 7967--7976.

\leavevmode\hypertarget{ref-shinzawa_2012}{}%
60. Shinzawa, H.; Nishida, M.; Kanematsu, W.; Tanaka, T.; Suzuki, K.; Noda, I. Parallel factor (PARAFAC) kernel analysis of temperature- and composition-dependent NMR spectra of poly(lactic acid) nanocomposites. \emph{The Analyst} \textbf{2012}, \emph{137}, 1913--1921.

\leavevmode\hypertarget{ref-chen_2018}{}%
61. Chen, K.; Park, J.; Li, F.; Patil, S.M.; Keire, D.A. Chemometric methods to quantify 1D and 2D NMR spectral differences among similar protein therapeutics. \emph{AAPS PharmSciTech} \textbf{2018}, \emph{19}, 1011--1019.

\leavevmode\hypertarget{ref-pedersen_2006}{}%
62. Pedersen, H.T.; Dyrby, M.; Engelsen, S.; Bro, R. Application of multi-way analysis to 2D NMR data. In; Annual reports on NMR spectroscopy; Elsevier, 2006; Vol. 59, pp. 207--233 ISBN \href{https://worldcat.org/isbn/9780125054591}{9780125054591}.

\leavevmode\hypertarget{ref-hao_2012}{}%
63. Hao, J.; Astle, W.; De Iorio, M.; Ebbels, T.M.D. BATMAN--an r package for the automated quantification of metabolites from nuclear magnetic resonance spectra using a bayesian model. \emph{Bioinformatics} \textbf{2012}, \emph{28}, 2088--2090.

\leavevmode\hypertarget{ref-bioconductor_website_ndb}{}%
64. Bioconductor Bioconductor - BiocViews:~packages found under StatisticalMethod.

\leavevmode\hypertarget{ref-groemping_website_2019}{}%
65. Groemping, U. CRAN task view: Design of experiments (DoE) \& analysis of experimental data 2019.

\leavevmode\hypertarget{ref-leisch_website_2019}{}%
66. Leisch, F.; Gruen, B. CRAN task view: Cluster analysis \& finite mixture models 2019.

\leavevmode\hypertarget{ref-hewson_website_2018}{}%
67. Hewson, P. CRAN task view: Multivariate statistics 2018.

\leavevmode\hypertarget{ref-hothorn_website_2019}{}%
68. Hothorn, T. CRAN task view: Machine learning \& statistical learning 2019.

\leavevmode\hypertarget{ref-thecomprehensiverarchivenetwork_website_nda}{}%
69. The Comprehensive R Archive Network CRAN task views.

\leavevmode\hypertarget{ref-bishop_2006}{}%
70. Bishop, C.M. \emph{Pattern recognition and machine learning (information science and statistics)}; Springer: New York, 2006; p. 738; ISBN \href{https://worldcat.org/isbn/978-0387310732}{978-0387310732}.

\leavevmode\hypertarget{ref-brusco_2017}{}%
71. Brusco, M.J.; Shireman, E.; Steinley, D. A comparison of latent class, k-means, and k-median methods for clustering dichotomous data. \emph{Psychological methods} \textbf{2017}, \emph{22}, 563--580.

\leavevmode\hypertarget{ref-felici_2007}{}%
72. Felici, G. \emph{Mathematical methods for knowledge discovery and data mining}; Idea Group Reference: Hershey, 2007; p. 371; ISBN \href{https://worldcat.org/isbn/978-1599045283}{978-1599045283}.

\leavevmode\hypertarget{ref-Routledge_2018}{}%
73. \emph{Introduction to multivariate analysis}; Routledge, 2018; ISBN \href{https://worldcat.org/isbn/9780203749999}{9780203749999}.

\leavevmode\hypertarget{ref-manly_2017}{}%
74. Manly, B.F. \emph{Multivariate statistical methods}; 4th ed.; Routledge: Boca Raton, 2017; p. 270; ISBN \href{https://worldcat.org/isbn/9781498728966}{9781498728966}.

\leavevmode\hypertarget{ref-mllner_2011}{}%
75. Müllner, D. Modern hierarchical, agglomerative clustering algorithms. \emph{arXiv} \textbf{2011}, \emph{abs/1109.2378}.

\leavevmode\hypertarget{ref-murtagh_2012}{}%
76. Murtagh, F.; Contreras, P. Algorithms for hierarchical clustering: An overview. \emph{Wiley Interdisciplinary Reviews: Data Mining and Knowledge Discovery} \textbf{2012}, \emph{2}, 86--97.

\leavevmode\hypertarget{ref-shaw_2003}{}%
77. Shaw, P.J.A. \emph{Multivariate statistics for the environmental sciences (mathematics)}; 1st ed.; Hodder Education Publishers: London, 2003; p. 233; ISBN \href{https://worldcat.org/isbn/0-340-80763-6}{0-340-80763-6}.

\leavevmode\hypertarget{ref-zaslavsky_2016}{}%
78. Zaslavsky, L.; Ciufo, S.; Fedorov, B.; Tatusova, T. Clustering analysis of proteins from microbial genomes at multiple levels of resolution. \emph{BMC Bioinformatics} \textbf{2016}, \emph{17 Suppl 8}, 276.

\leavevmode\hypertarget{ref-hall_2011}{}%
79. Hall; Robert, D. \emph{Annual plant reviews, biology of plant metabolomics}; 1st ed.; Wiley, John \& Sons, Incorporated, 2011; p. 448; ISBN \href{https://worldcat.org/isbn/978-1-4051-9954-4}{978-1-4051-9954-4}.

\leavevmode\hypertarget{ref-cai_2017}{}%
80. Cai, Q.; Alvarez, J.A.; Kang, J.; Yu, T. Network marker selection for untargeted LC-MS metabolomics data. \emph{Journal of Proteome Research} \textbf{2017}, \emph{16}, 1261--1269.

\leavevmode\hypertarget{ref-christin_2013}{}%
81. Christin, C.; Hoefsloot, H.C.J.; Smilde, A.K.; Hoekman, B.; Suits, F.; Bischoff, R.; Horvatovich, P. A critical assessment of feature selection methods for biomarker discovery in clinical proteomics. \emph{Molecular \& Cellular Proteomics} \textbf{2013}, \emph{12}, 263--276.

\leavevmode\hypertarget{ref-rohart_2017}{}%
82. Rohart, F.; Gautier, B.; Singh, A.; Lê Cao, K.-A. mixOmics: An r package for 'omics feature selection and multiple data integration. \emph{PLoS Computational Biology} \textbf{2017}, \emph{13}, e1005752.

\leavevmode\hypertarget{ref-wen_2017}{}%
83. Wen, B.; Mei, Z.; Zeng, C.; Liu, S. metaX: A flexible and comprehensive software for processing metabolomics data. \emph{BMC Bioinformatics} \textbf{2017}, \emph{18}, 183.

\leavevmode\hypertarget{ref-peters_2018}{}%
84. Peters, K.; Worrich, A.; Weinhold, A.; Alka, O.; Balcke, G.; Birkemeyer, C.; Bruelheide, H.; Calf, O.W.; Dietz, S.; Dührkop, K. et al. Current challenges in plant eco-metabolomics. \emph{International Journal of Molecular Sciences} \textbf{2018}, \emph{19}.

\leavevmode\hypertarget{ref-gromski_2014}{}%
85. Gromski, P.S.; Xu, Y.; Correa, E.; Ellis, D.I.; Turner, M.L.; Goodacre, R. A comparative investigation of modern feature selection and classification approaches for the analysis of mass spectrometry data. \emph{Analytica Chimica Acta} \textbf{2014}, \emph{829}, 1--8.

\leavevmode\hypertarget{ref-legendre_2012}{}%
86. Legendre, P.; Legendre, L.F.J. \emph{Numerical ecology, volume 24 (developments in environmental modelling)}; 3rd ed.; Elsevier, 2012; p. 1006; ISBN \href{https://worldcat.org/isbn/978-0-444-53868-0}{978-0-444-53868-0}.

\leavevmode\hypertarget{ref-clarke_2009}{}%
87. Clarke, B.; Fokoue, E.; Zhang, H.H. \emph{Principles and theory for data mining and machine learning}; Springer series in statistics; Springer New York: New York, NY, 2009; ISBN \href{https://worldcat.org/isbn/978-0-387-98134-5}{978-0-387-98134-5}.

\leavevmode\hypertarget{ref-feng_2018}{}%
88. Feng, J.; Li, B.; Jiang, X.; Yang, Y.; Wells, G.F.; Zhang, T.; Li, X. Antibiotic resistome in a large-scale healthy human gut microbiota deciphered by metagenomic and network analyses. \emph{Environmental Microbiology} \textbf{2018}, \emph{20}, 355--368.

\leavevmode\hypertarget{ref-fukushima_2009}{}%
89. Fukushima, A.; Kusano, M.; Redestig, H.; Arita, M.; Saito, K. Integrated omics approaches in plant systems biology. \emph{Current Opinion in Chemical Biology} \textbf{2009}, \emph{13}, 532--538.

\leavevmode\hypertarget{ref-vaughan_2012}{}%
90. Vaughan, A.A.; Dunn, W.B.; Allwood, J.W.; Wedge, D.C.; Blackhall, F.H.; Whetton, A.D.; Dive, C.; Goodacre, R. Liquid chromatography-mass spectrometry calibration transfer and metabolomics data fusion. \emph{Analytical Chemistry} \textbf{2012}, \emph{84}, 9848--9857.

\leavevmode\hypertarget{ref-wei_2018}{}%
91. Wei, R.; Wang, J.; Su, M.; Jia, E.; Chen, S.; Chen, T.; Ni, Y. Missing value imputation approach for mass spectrometry-based metabolomics data. \emph{Scientific reports} \textbf{2018}, \emph{8}, 663.

\leavevmode\hypertarget{ref-degenhardt_2019}{}%
92. Degenhardt, F.; Seifert, S.; Szymczak, S. Evaluation of variable selection methods for random forests and omics data sets. \emph{Briefings in Bioinformatics} \textbf{2019}, \emph{20}, 492--503.

\leavevmode\hypertarget{ref-saeys_2007}{}%
93. Saeys, Y.; Inza, I.; Larrañaga, P. A review of feature selection techniques in bioinformatics. \emph{Bioinformatics} \textbf{2007}, \emph{23}, 2507--2517.

\leavevmode\hypertarget{ref-determanjr_2014}{}%
94. Determan Jr, C.E. Optimal algorithm for metabolomics classification and feature selection varies by dataset. \emph{International journal of biology} \textbf{2014}, \emph{7}.

\leavevmode\hypertarget{ref-rinaudo_2016}{}%
95. Rinaudo, P.; Boudah, S.; Junot, C.; Thévenot, E.A. Biosigner: A new method for the discovery of significant molecular signatures from omics data. \emph{Frontiers in molecular biosciences} \textbf{2016}, \emph{3}, 26.

\leavevmode\hypertarget{ref-shi_2019}{}%
96. Shi, L.; Westerhuis, J.A.; Rosén, J.; Landberg, R.; Brunius, C. Variable selection and validation in multivariate modelling. \emph{Bioinformatics} \textbf{2019}, \emph{35}, 972--980.

\leavevmode\hypertarget{ref-wehrens_2012}{}%
97. Wehrens, R.; Franceschi, P. Meta-statistics for variable selection: TheR PackageBioMark. \emph{Journal of statistical software} \textbf{2012}, \emph{51}.

\leavevmode\hypertarget{ref-li_2013}{}%
98. Li, S.; Park, Y.; Duraisingham, S.; Strobel, F.H.; Khan, N.; Soltow, Q.A.; Jones, D.P.; Pulendran, B. Predicting network activity from high throughput metabolomics. \emph{PLoS Computational Biology} \textbf{2013}, \emph{9}, e1003123.

\leavevmode\hypertarget{ref-willighagen_2017}{}%
99. Willighagen, E.L.; Mayfield, J.W.; Alvarsson, J.; Berg, A.; Carlsson, L.; Jeliazkova, N.; Kuhn, S.; Pluskal, T.; Rojas-Chertó, M.; Spjuth, O. et al. The chemistry development kit (CDK) v2.0: Atom typing, depiction, molecular formulas, and substructure searching. \emph{Journal of cheminformatics} \textbf{2017}, \emph{9}, 33.

\leavevmode\hypertarget{ref-heller_2015}{}%
100. Heller, S.R.; McNaught, A.; Pletnev, I.; Stein, S.; Tchekhovskoi, D. Inchi, the IUPAC international chemical identifier. \emph{Journal of cheminformatics} \textbf{2015}, \emph{7}, 23.

\leavevmode\hypertarget{ref-backman_2011}{}%
101. Backman, T.W.H.; Cao, Y.; Girke, T. ChemMine tools: An online service for analyzing and clustering small molecules. \emph{Nucleic Acids Research} \textbf{2011}, \emph{39}, W486--91.

\leavevmode\hypertarget{ref-oboyle_2011}{}%
102. O'Boyle, N.M.; Banck, M.; James, C.A.; Morley, C.; Vandermeersch, T.; Hutchison, G.R. Open babel: An open chemical toolbox. \emph{Journal of cheminformatics} \textbf{2011}, \emph{3}, 33.

\leavevmode\hypertarget{ref-landrum_2016}{}%
103. Landrum, G. RDKit: Open-source cheminformatics software. \textbf{2016}.

\leavevmode\hypertarget{ref-kim_2016}{}%
104. Kim, S.; Thiessen, P.A.; Bolton, E.E.; Chen, J.; Fu, G.; Gindulyte, A.; Han, L.; He, J.; He, S.; Shoemaker, B.A. et al. PubChem substance and compound databases. \emph{Nucleic Acids Research} \textbf{2016}, \emph{44}, D1202--13.

\leavevmode\hypertarget{ref-wang_2009}{}%
105. Wang, Y.; Xiao, J.; Suzek, T.O.; Zhang, J.; Wang, J.; Bryant, S.H. PubChem: A public information system for analyzing bioactivities of small molecules. \emph{Nucleic Acids Research} \textbf{2009}, \emph{37}, W623--33.

\leavevmode\hypertarget{ref-pence_2010}{}%
106. Pence, H.E.; Williams, A. Chemspider: An online chemical information resource. \emph{Journal of chemical education} \textbf{2010}, \emph{87}, 1123--1124.

\leavevmode\hypertarget{ref-erxleben_2014}{}%
107. Erxleben, F.; Günther, M.; Krötzsch, M.; Mendez, J.; Vrandečić, D. Introducing wikidata to the linked data web. In \emph{The semantic web -- ISWC 2014}; Mika, P., Tudorache, T., Bernstein, A., Welty, C., Knoblock, C., Vrandečić, D., Groth, P., Noy, N., Janowicz, K., Goble, C., Eds.; Lecture notes in computer science; Springer International Publishing: Cham, 2014; Vol. 8796, pp. 50--65 ISBN \href{https://worldcat.org/isbn/978-3-319-11963-2}{978-3-319-11963-2}.

\leavevmode\hypertarget{ref-wohlgemuth_2010}{}%
108. Wohlgemuth, G.; Haldiya, P.K.; Willighagen, E.; Kind, T.; Fiehn, O. The chemical translation service--a web-based tool to improve standardization of metabolomic reports. \emph{Bioinformatics} \textbf{2010}, \emph{26}, 2647--2648.

\leavevmode\hypertarget{ref-srcinc_website_nd}{}%
109. SRC, Inc. Scientific databases.

\leavevmode\hypertarget{ref-ncicaddgroup_website_nd}{}%
110. Group NCI/CADD chemical identifier resolver.

\leavevmode\hypertarget{ref-djoumboufeunang_2016}{}%
111. Djoumbou Feunang, Y.; Eisner, R.; Knox, C.; Chepelev, L.; Hastings, J.; Owen, G.; Fahy, E.; Steinbeck, C.; Subramanian, S.; Bolton, E. et al. ClassyFire: Automated chemical classification with a comprehensive, computable taxonomy. \emph{Journal of cheminformatics} \textbf{2016}, \emph{8}, 61.

\leavevmode\hypertarget{ref-watrous_2012}{}%
112. Watrous, J.; Roach, P.; Alexandrov, T.; Heath, B.S.; Yang, J.Y.; Kersten, R.D.; Voort, M. van der; Pogliano, K.; Gross, H.; Raaijmakers, J.M. et al. Mass spectral molecular networking of living microbial colonies. \emph{Proceedings of the National Academy of Sciences of the United States of America} \textbf{2012}, \emph{109}, E1743--52.

\leavevmode\hypertarget{ref-kanehisa_2017}{}%
113. Kanehisa, M.; Furumichi, M.; Tanabe, M.; Sato, Y.; Morishima, K. KEGG: New perspectives on genomes, pathways, diseases and drugs. \emph{Nucleic Acids Research} \textbf{2017}, \emph{45}, D353--D361.

\leavevmode\hypertarget{ref-kanehisa_2019}{}%
114. Kanehisa, M.; Sato, Y.; Furumichi, M.; Morishima, K.; Tanabe, M. New approach for understanding genome variations in KEGG. \emph{Nucleic Acids Research} \textbf{2019}, \emph{47}, D590--D595.

\leavevmode\hypertarget{ref-kanehisa_2000}{}%
115. Kanehisa, M.; Goto, S. KEGG: Kyoto encyclopedia of genes and genomes. \emph{Nucleic Acids Research} \textbf{2000}, \emph{28}, 27--30.

\leavevmode\hypertarget{ref-lim_2010}{}%
116. Lim, E.; Pon, A.; Djoumbou, Y.; Knox, C.; Shrivastava, S.; Guo, A.C.; Neveu, V.; Wishart, D.S. T3DB: A comprehensively annotated database of common toxins and their targets. \emph{Nucleic Acids Research} \textbf{2010}, \emph{38}, D781--6.

\leavevmode\hypertarget{ref-wishart_2015}{}%
117. Wishart, D.; Arndt, D.; Pon, A.; Sajed, T.; Guo, A.C.; Djoumbou, Y.; Knox, C.; Wilson, M.; Liang, Y.; Grant, J. et al. T3DB: The toxic exposome database. \emph{Nucleic Acids Research} \textbf{2015}, \emph{43}, D928--34.

\leavevmode\hypertarget{ref-fahy_2007}{}%
118. Fahy, E.; Sud, M.; Cotter, D.; Subramaniam, S. LIPID MAPS online tools for lipid research. \emph{Nucleic Acids Research} \textbf{2007}, \emph{35}, W606--12.

\leavevmode\hypertarget{ref-shen_2019}{}%
119. Shen, X.; Wang, R.; Xiong, X.; Yin, Y.; Cai, Y.; Ma, Z.; Liu, N.; Zhu, Z.-J. Metabolic reaction network-based recursive metabolite annotation for untargeted metabolomics. \emph{Nature Communications} \textbf{2019}, \emph{10}, 1516.

\leavevmode\hypertarget{ref-schaefer_2009}{}%
120. Schaefer, C.F.; Anthony, K.; Krupa, S.; Buchoff, J.; Day, M.; Hannay, T.; Buetow, K.H. PID: The pathway interaction database. \emph{Nucleic Acids Research} \textbf{2009}, \emph{37}, D674--9.

\leavevmode\hypertarget{ref-nishimura_2001}{}%
121. Nishimura, D. BioCarta. \emph{Biotech Software \& Internet Report} \textbf{2001}, \emph{2}, 117--120.

\leavevmode\hypertarget{ref-fabregat_2018}{}%
122. Fabregat, A.; Jupe, S.; Matthews, L.; Sidiropoulos, K.; Gillespie, M.; Garapati, P.; Haw, R.; Jassal, B.; Korninger, F.; May, B. et al. The reactome pathway knowledgebase. \emph{Nucleic Acids Research} \textbf{2018}, \emph{46}, D649--D655.

\leavevmode\hypertarget{ref-kramer_2014}{}%
123. Kramer, F.; Bayerlová, M.; Beißbarth, T. R-based software for the integration of pathway data into bioinformatic algorithms. \emph{Biology} \textbf{2014}, \emph{3}, 85--100.

\leavevmode\hypertarget{ref-tenenbaum_website_2019}{}%
124. Tenenbaum, D. Bioconductor - KEGGREST 2019.

\leavevmode\hypertarget{ref-chang_website_2012}{}%
125. Chang, W.; Cheng, J.; Allaire, J.; Xie, Y.; McPherson, J. Shiny: Web application framework for r 2012.

\leavevmode\hypertarget{ref-roccaserra_2010}{}%
126. Rocca-Serra, P.; Brandizi, M.; Maguire, E.; Sklyar, N.; Taylor, C.; Begley, K.; Field, D.; Harris, S.; Hide, W.; Hofmann, O. et al. ISA software suite: Supporting standards-compliant experimental annotation and enabling curation at the community level. \emph{Bioinformatics} \textbf{2010}, \emph{26}, 2354--2356.

\leavevmode\hypertarget{ref-sansone_2012}{}%
127. Sansone, S.-A.; Rocca-Serra, P.; Field, D.; Maguire, E.; Taylor, C.; Hofmann, O.; Fang, H.; Neumann, S.; Tong, W.; Amaral-Zettler, L. et al. Toward interoperable bioscience data. \emph{Nature Genetics} \textbf{2012}, \emph{44}, 121--126.

\leavevmode\hypertarget{ref-plotlytechnologiesinc_website_2015}{}%
128. Plotly Technologies Inc Collaborative data science 2015.

\leavevmode\hypertarget{ref-wickham_2016}{}%
129. Wickham, H. \emph{Ggplot2: Elegant graphics for data analysis}; Springer International Publishing : 2016; ISBN \href{https://worldcat.org/isbn/978-3-319-24277-4}{978-3-319-24277-4}.

\leavevmode\hypertarget{ref-jupyter_2018}{}%
130. Jupyter, P.; Bussonnier, M.; Forde, J.; Freeman, J.; Granger, B.; Head, T.; Holdgraf, C.; Kelley, K.; Nalvarte, G.; Osheroff, A. et al. Binder 2.0 - reproducible, interactive, sharable environments for science at scale. In Proceedings of the Proceedings of the 17th python in science conference; SciPy, 2018; pp. 113--120.

\leavevmode\hypertarget{ref-ram_website_nd}{}%
131. Ram, K. Configure your r project for binderhub • hole punch.

\leavevmode\hypertarget{ref-liggi_2018}{}%
132. Liggi, S.; Hinz, C.; Hall, Z.; Santoru, M.L.; Poddighe, S.; Fjeldsted, J.; Atzori, L.; Griffin, J.L. KniMet: A pipeline for the processing of chromatography-mass spectrometry metabolomics data. \emph{Metabolomics : Official journal of the Metabolomic Society} \textbf{2018}, \emph{14}, 52.

\leavevmode\hypertarget{ref-verhoeven_2018}{}%
133. Verhoeven, A.; Giera, M.; Mayboroda, O.A. KIMBLE: A versatile visual NMR metabolomics workbench in KNIME. \emph{Analytica Chimica Acta} \textbf{2018}, \emph{1044}, 66--76.

\leavevmode\hypertarget{ref-davidson_2016}{}%
134. Davidson, R.L.; Weber, R.J.M.; Liu, H.; Sharma-Oates, A.; Viant, M.R. Galaxy-m: A galaxy workflow for processing and analyzing direct infusion and liquid chromatography mass spectrometry-based metabolomics data. \emph{GigaScience} \textbf{2016}, \emph{5}, 10.

\leavevmode\hypertarget{ref-giacomoni_2015}{}%
135. Giacomoni, F.; Le Corguillé, G.; Monsoor, M.; Landi, M.; Pericard, P.; Pétéra, M.; Duperier, C.; Tremblay-Franco, M.; Martin, J.-F.; Jacob, D. et al. Workflow4Metabolomics: A collaborative research infrastructure for computational metabolomics. \emph{Bioinformatics} \textbf{2015}, \emph{31}, 1493--1495.

\leavevmode\hypertarget{ref-goecks_2010}{}%
136. Goecks, J.; Nekrutenko, A.; Taylor, J.; Team, G. Galaxy: A comprehensive approach for supporting accessible, reproducible, and transparent computational research in the life sciences. \emph{Genome Biology} \textbf{2010}, \emph{11}, R86.

\leavevmode\hypertarget{ref-metabohub_website_2019}{}%
137. Metabohub National infrastructure in metabolomics and fluxomics 2019.

\leavevmode\hypertarget{ref-guitton_2017}{}%
138. Guitton, Y.; Tremblay-Franco, M.; Le Corguillé, G.; Martin, J.-F.; Pétéra, M.; Roger-Mele, P.; Delabrière, A.; Goulitquer, S.; Monsoor, M.; Duperier, C. et al. Create, run, share, publish, and reference your LC-MS, FIA-MS, GC-MS, and NMR data analysis workflows with the Workflow4Metabolomics 3.0 galaxy online infrastructure for metabolomics. \emph{The International Journal of Biochemistry \& Cell Biology} \textbf{2017}, \emph{93}, 89--101.

\leavevmode\hypertarget{ref-workflow4metabolomics_website_nd}{}%
139. Workflow4metabolomics Referenced W4M histories workflow4metabolomics.org.

\leavevmode\hypertarget{ref-goble_2019}{}%
140. Goble, C.; Cohen-Boulakia, S.; Soiland-Reyes, S.; Garijo, D.; Gil, Y.; Crusoe, M.R.; Peters, K.; Schober, D. FAIR computational workflows. \emph{Data Intelligence} \textbf{2020}, \emph{2}, 108--121.

\leavevmode\hypertarget{ref-wilkinson_2016}{}%
141. Wilkinson, M.D.; Dumontier, M.; Aalbersberg, I.J.J.; Appleton, G.; Axton, M.; Baak, A.; Blomberg, N.; Boiten, J.-W.; Silva Santos, L.B. da; Bourne, P.E. et al. The FAIR guiding principles for scientific data management and stewardship. \emph{Scientific data} \textbf{2016}, \emph{3}, 160018.

\leavevmode\hypertarget{ref-haug_2013}{}%
142. Haug, K.; Salek, R.M.; Conesa, P.; Hastings, J.; Matos, P. de; Rijnbeek, M.; Mahendraker, T.; Williams, M.; Neumann, S.; Rocca-Serra, P. et al. MetaboLights--an open-access general-purpose repository for metabolomics studies and associated meta-data. \emph{Nucleic Acids Research} \textbf{2013}, \emph{41}, D781--6.

\leavevmode\hypertarget{ref-sud_2016}{}%
143. Sud, M.; Fahy, E.; Cotter, D.; Azam, K.; Vadivelu, I.; Burant, C.; Edison, A.; Fiehn, O.; Higashi, R.; Nair, K.S. et al. Metabolomics workbench: An international repository for metabolomics data and metadata, metabolite standards, protocols, tutorials and training, and analysis tools. \emph{Nucleic Acids Research} \textbf{2016}, \emph{44}, D463--70.

\leavevmode\hypertarget{ref-gray_2017}{}%
144. Gray, A.J.G.; Goble, C.A.; Jimenez, R. Bioschemas: From potato salad to protein annotation. In Proceedings of the Proceedings of the ISWC 2017 posters \& demonstrations and industry tracks co-located with 16th international semantic web conference (ISWC 2017), vienna, austria, october 23rd - to - 25th, 2017.; Nikitina, N., Song, D., Fokoue, A., Haase, P., Eds.; CEUR-WS.org, 2017; Vol. 1963.

\leavevmode\hypertarget{ref-collberg_2016}{}%
145. Collberg, C.; Proebsting, T.A. Repeatability in computer systems research. \emph{Communications of the ACM} \textbf{2016}, \emph{59}, 62--69.

\leavevmode\hypertarget{ref-taschuk_2017}{}%
146. Taschuk, M.; Wilson, G. Ten simple rules for making research software more robust. \emph{PLoS Computational Biology} \textbf{2017}, \emph{13}, e1005412.

\leavevmode\hypertarget{ref-stanstrup_website_nd}{}%
147. Stanstrup, J.; Broeckling, C.D.; Helmus, R.; Hoffmann, N.; Mathé, E.; Naake, T.; Nicolotti, L.; Peters, K.; Rainer, J.; Salek, R. et al. The MetaRbolomics book.

\newpage

\hypertarget{appendices}{%
\section*{Appendices}\label{appendices}}
\addcontentsline{toc}{section}{Appendices}

\newpage

\hypertarget{appendix-1-the-msp-file-format-and-package-support}{%
\subsection*{Appendix 1: The MSP File Format and package support}\label{appendix-1-the-msp-file-format-and-package-support}}
\addcontentsline{toc}{subsection}{Appendix 1: The MSP File Format and package support}

\begin{longtable}[]{@{}l@{}}
\toprule
\endhead
\begin{minipage}[t]{0.38\columnwidth}\raggedright
Name: unknown\\
Num Peaks: 2\\
85.345 100; 76.321 50;\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\textbf{Listing S1:} Example for the basic NIST format.

\leavevmode\newline\leavevmode\newline\leavevmode\newline\leavevmode\newline

\begin{tabular}{>{\raggedright\arraybackslash}p{22em}|>{\raggedright\arraybackslash}p{22em}}
\toprule
Name: 1-Methylhistidine\newline Synon: (2S)-2-amino-3-(1-methyl-1H-imidazol-4-yl)propanoic acid\newline SYNON: {\$}:00in-source\newline DB{\#}: HMDB0000001{\_{}}c{\_{}}ms{\_{}}1469\newline InChIKey: BRMWTNUJHUMWMS-LURJTMIESA-N\newline Instrument{\_{}}type: GC-MS\newline Retention{\_{}}index: 1807.71\newline Formula: C7H11N3O2\newline MW: 169\newline ExactMass: 169.0851\newline Comments: "column=5{\%}-phenyl-95{\%}-dimethylpolysiloxane capillary column" "derivatization type=2 TMS" "derivatization formula=C13H27N3O2Si2" "derivative mw=313.544" "retention index=1807.71" "retention index type=based on 9 n-alkanes (C10–C36)" "instrument type=GC-MS" "chromatography type=GC" "cas number=332-80-9" "molecular formula=C7H11N3O2" "total exact mass=169.085126592" "InChIKey=BRMWTNUJHUMWMS-LURJTMIESA-N"\newline Num Peaks: 10\newline  70  0.014;  71  0.007;  72  0.02;  76  0.008;  77  0.008;\newline  78  0.002;  79  0.003;  80  0.005;  81  0.108;  82  0.017; & NAME: Aspartame; LC-ESI-ITFT; MS2; CE\newline PRECURSORMZ: 295.128848\newline PRECURSORTYPE: [M+H]+\newline INSTRUMENTTYPE: LC-ESI-ITFT\newline SMILES: COC(=O)C(CC1=CC=CC=C1)N=C(O)C(N)CC(O)=O\newline INCHIKEY: IAOZJIPTCAWIRG-UHFFFAOYNA-N\newline Ontology: Peptides\newline COLLISIONENERGY: 35\newline FORMULA: C14H18N2O5\newline RETENTIONTIME:  \newline IONMODE: Positive\newline Comment: registered in MassBank\newline Num Peaks: 9\newline 120.0804    13\newline 180.10201    138\newline 217.0968    14\newline 235.10789    390\newline 245.0921    274\newline 260.09171    132\newline 263.1026    286\newline 277.11859    1000\newline 278.1022    28\\
\hline
\rowcolor{white}  {\bf Listing S2:} Example for the canonical NIST format. & {\bf Listing S3:} RIKEN PRIME msp format example.\\
\bottomrule
\end{tabular}

\leavevmode\newline\leavevmode\newline\leavevmode\newline\leavevmode\newline
\newpage

\textbf{Table S1:} Overview of MS/MS handling in different R packages. `-' means not available, for the remaining entries see the text above.

\begin{longtable}[]{@{}llll@{}}
\toprule
\begin{minipage}[b]{0.09\columnwidth}\raggedright
package\strut
\end{minipage} & \begin{minipage}[b]{0.17\columnwidth}\raggedright
read msp\strut
\end{minipage} & \begin{minipage}[b]{0.17\columnwidth}\raggedright
write msp\strut
\end{minipage} & \begin{minipage}[b]{0.46\columnwidth}\raggedright
spectral matching and additional information\strut
\end{minipage}\tabularnewline
\midrule
\endhead
\begin{minipage}[t]{0.09\columnwidth}\raggedright
baitmet\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; cosine, Stein \& Scott composite similarity product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
compMS2Miner\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST, RIKEN PRIME msp\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
RIKEN PRIME msp\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; dot product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
enviGCMS\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
basic NIST\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
erah\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
only result export\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; cosine\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
flagme\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
only result export\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
metaMS\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST; slow\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
1 vs DB, N vs DB; proprietary\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MatchWeiz\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; X-Rank\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MetCirc\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs N; normalized dot product;\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
will switch to MSnbase functions soon\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MSeasy\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
only result export\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; Queries the NIST mass spectral search tool\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
MSnbase\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
**\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
**\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
1 vs 1, N vs N; dot product and more, user def.\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
msPurity\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; dot product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
OrgMassSpecR\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
basic NIST\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
basic NIST\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
1 vs 1; normalized dot product\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
RAMClustR\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
RAMClustR can import and utilize spectrum similarities from MS-FINDER;\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
rTANDEM\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; dot product; R wrapper for X!Tandem software\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
SwathXtend\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
- (PeakView / OpenSWATH)\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
- (PeakView / OpenSWATH)\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
\strut
\end{minipage}\tabularnewline
\begin{minipage}[t]{0.09\columnwidth}\raggedright
TargetSearch\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST (with error)\strut
\end{minipage} & \begin{minipage}[t]{0.17\columnwidth}\raggedright
NIST\strut
\end{minipage} & \begin{minipage}[t]{0.46\columnwidth}\raggedright
N vs DB; RI-based\strut
\end{minipage}\tabularnewline
\bottomrule
\end{longtable}

\newpage

\hypertarget{appendix-2-metarbolomics-dependencies-network}{%
\subsection*{Appendix 2: metaRbolomics dependencies network}\label{appendix-2-metarbolomics-dependencies-network}}
\addcontentsline{toc}{subsection}{Appendix 2: metaRbolomics dependencies network}

\hypertarget{libraries-and-settings}{%
\subsubsection*{Libraries and settings}\label{libraries-and-settings}}
\addcontentsline{toc}{subsubsection}{Libraries and settings}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{options}\NormalTok{(}\StringTok{"repos"}\NormalTok{ =}\StringTok{ }\KeywordTok{list}\NormalTok{(}\DataTypeTok{CRAN=}\StringTok{"http://cran.rstudio.com/"}\NormalTok{))}

\KeywordTok{library}\NormalTok{(devtools)   }\CommentTok{# for revdp()}
\KeywordTok{library}\NormalTok{(igraph)     }\CommentTok{# for graph_from_edgelist/( and simplify() )}
\KeywordTok{library}\NormalTok{(visNetwork) }\CommentTok{# for visNetwork() and friends}
\KeywordTok{library}\NormalTok{(networkD3)  }\CommentTok{# for saveNetwork()}
\KeywordTok{library}\NormalTok{(chromote)   }\CommentTok{# for default_chromote_object()}
\KeywordTok{library}\NormalTok{(webshot2)   }\CommentTok{# for webshot()}
\KeywordTok{library}\NormalTok{(png)        }\CommentTok{# For displaying an image}
\KeywordTok{library}\NormalTok{(dplyr)}
\KeywordTok{library}\NormalTok{(purrr)}

\KeywordTok{source}\NormalTok{(}\StringTok{"scripts/revDepNetHelper.R"}\NormalTok{)}

\KeywordTok{set_default_chromote_object}\NormalTok{(Chromote}\OperatorTok{$}\KeywordTok{new}\NormalTok{(}\DataTypeTok{browser =}\NormalTok{ Chrome}\OperatorTok{$}\KeywordTok{new}\NormalTok{(}\DataTypeTok{args =} \StringTok{"--no-sandbox"}\NormalTok{)))}
\end{Highlighting}
\end{Shaded}

\hypertarget{read-package-names-from-our-table}{%
\subsubsection*{Read package names from our table}\label{read-package-names-from-our-table}}
\addcontentsline{toc}{subsubsection}{Read package names from our table}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{reviewTables <-}\StringTok{ }\KeywordTok{read.delim}\NormalTok{(}\StringTok{"public/data/AllMetaRbolomicsTables.csv"}\NormalTok{, }\DataTypeTok{stringsAsFactors =} \OtherTok{FALSE}\NormalTok{)}
\NormalTok{reviewPkgs <-}\StringTok{ }\NormalTok{reviewTables[,}\StringTok{"Package"}\NormalTok{]}

\NormalTok{pkgs <-}\StringTok{ }\NormalTok{reviewPkgs}
\end{Highlighting}
\end{Shaded}

\hypertarget{get-reverse-dependencies}{%
\subsubsection*{Get reverse dependencies}\label{get-reverse-dependencies}}
\addcontentsline{toc}{subsubsection}{Get reverse dependencies}

\hypertarget{for-cran-and-bioc-packages}{%
\paragraph{For CRAN and BioC packages}\label{for-cran-and-bioc-packages}}
\addcontentsline{toc}{paragraph}{For CRAN and BioC packages}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{el <-}\StringTok{ }\KeywordTok{sapply}\NormalTok{(pkgs, }\ControlFlowTok{function}\NormalTok{(pkg) \{}
\NormalTok{  rd <-}\StringTok{ }\KeywordTok{revdep}\NormalTok{(pkg, }\DataTypeTok{dependencies =} \KeywordTok{c}\NormalTok{(}\StringTok{"Depends"}\NormalTok{, }\StringTok{"Imports"}\NormalTok{, }\StringTok{"LinkingTo"}\NormalTok{), }
               \DataTypeTok{recursive =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{ignore =} \OtherTok{NULL}\NormalTok{, }\DataTypeTok{bioconductor =} \OtherTok{TRUE}\NormalTok{)}
  \KeywordTok{as.matrix}\NormalTok{(}\KeywordTok{cbind}\NormalTok{(}\DataTypeTok{Package=}\KeywordTok{rep}\NormalTok{(pkg, }\DataTypeTok{length.out=}\KeywordTok{length}\NormalTok{(rd)), }\DataTypeTok{ReverseDep=}\NormalTok{rd))}
\NormalTok{\})}
\NormalTok{el <-}\StringTok{ }\KeywordTok{do.call}\NormalTok{(rbind, el)}
\end{Highlighting}
\end{Shaded}

\hypertarget{for-github-and-gitlab}{%
\paragraph{For GitHub and GitLab}\label{for-github-and-gitlab}}
\addcontentsline{toc}{paragraph}{For GitHub and GitLab}

The above \texttt{devtools::revdep} cannot read from GitHub/GitLab repositories.
We have a helper function that downloads and parses the DESCRIPTION file from GitHub/GitLab.
Since we cannot get reverse dependencies directly for GitHub/GitLab packages, those packages they are only used as additional reverse depedencies for the CRAN/BioC packages.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gitdeps_reverse <-}\StringTok{ }\NormalTok{reviewTables }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{mutate}\NormalTok{(}\DataTypeTok{dep_tree =} \KeywordTok{map}\NormalTok{(Code_link, get_git_deps)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{pull}\NormalTok{(dep_tree) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{bind_rows}\NormalTok{() }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{filter}\NormalTok{(Dep }\OperatorTok{%in%}\StringTok{ }\NormalTok{el[,}\StringTok{"Package"}\NormalTok{]) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{rename}\NormalTok{(}\DataTypeTok{Package =}\NormalTok{ Dep, }\DataTypeTok{ReverseDep =}\NormalTok{ Package) }\OperatorTok{%>%}\StringTok{ }
\StringTok{                      }\KeywordTok{as.matrix}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in readLines(file): incomplete final line found on '/tmp/RtmpkgrW4F/
## file76e30297c'
\end{verbatim}

\begin{verbatim}
## Warning in readLines(file): incomplete final line found on '/tmp/RtmpkgrW4F/
## file76566122b'
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{el <-}\StringTok{ }\KeywordTok{rbind}\NormalTok{(el, gitdeps_reverse)}
\end{Highlighting}
\end{Shaded}

\hypertarget{building-dependency-network}{%
\subsubsection*{Building dependency network}\label{building-dependency-network}}
\addcontentsline{toc}{subsubsection}{Building dependency network}

In total, we were analysing 297 packages.
For each package, this returns the set of packages in CRAN or BioC that depend on, import from or link to the package (i.e., its direct reverse dependencies) using the \texttt{devtools::revdep()} function. A few packages with the highest number of reverse dependencies have been excluded, as they would dominate the visualisation.
It was not possible to detect reverse dependencies from other hosting places such as GitHub or GitLab.

From the total, 70 packages had at least one
such reverse dependency.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{## Remove packages with most reverse dependencies }
\CommentTok{## which would dominate the network}

\NormalTok{el <-}\StringTok{ }\NormalTok{el[}\OperatorTok{!}\StringTok{ }\NormalTok{el[,}\StringTok{"Package"}\NormalTok{] }\OperatorTok{%in%}\StringTok{ }\KeywordTok{c}\NormalTok{(}\StringTok{"Rcpp"}\NormalTok{, }\StringTok{"igraph"}\NormalTok{, }\StringTok{"vegan"}\NormalTok{, }\StringTok{"caret"}\NormalTok{, }\StringTok{"rJava"}\NormalTok{, }\StringTok{"reticulate"}\NormalTok{), ]}

\CommentTok{## Create graph, and simplify redundancy}
\NormalTok{g <-}\StringTok{ }\KeywordTok{graph_from_edgelist}\NormalTok{(el, }\DataTypeTok{directed =} \OtherTok{TRUE}\NormalTok{)}
\NormalTok{g <-}\StringTok{ }\NormalTok{igraph}\OperatorTok{::}\KeywordTok{simplify}\NormalTok{(g, }\DataTypeTok{remove.multiple =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{remove.loops =} \OtherTok{TRUE}\NormalTok{)}

\CommentTok{# get data and plot :}
\NormalTok{data <-}\StringTok{ }\KeywordTok{toVisNetworkData}\NormalTok{(g)}

\NormalTok{data}\OperatorTok{$}\NormalTok{nodes <-}\StringTok{ }\KeywordTok{cbind}\NormalTok{(data}\OperatorTok{$}\NormalTok{nodes, }
                    \DataTypeTok{font.size=}\DecValTok{30}\NormalTok{, }
                    \DataTypeTok{color.background =} \KeywordTok{ifelse}\NormalTok{(data}\OperatorTok{$}\NormalTok{nodes[,}\StringTok{"id"}\NormalTok{] }\OperatorTok{%in%}\StringTok{ }\NormalTok{pkgs , }
                           \KeywordTok{rgb}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{0}\NormalTok{, }\DecValTok{200}\NormalTok{, }\DecValTok{128}\NormalTok{, }\DataTypeTok{max =} \DecValTok{255}\NormalTok{), }
                           \KeywordTok{rgb}\NormalTok{(}\DecValTok{0}\NormalTok{, }\DecValTok{200}\NormalTok{, }\DecValTok{0}\NormalTok{, }\DecValTok{128}\NormalTok{, }\DataTypeTok{max =} \DecValTok{255}\NormalTok{)))}

\NormalTok{vn <-}\StringTok{ }\KeywordTok{visNetwork}\NormalTok{(}\DataTypeTok{nodes =}\NormalTok{ data}\OperatorTok{$}\NormalTok{nodes, }
                 \DataTypeTok{edges =}\NormalTok{ data}\OperatorTok{$}\NormalTok{edges,}
                 \DataTypeTok{width=}\DecValTok{1000}\NormalTok{, }\DataTypeTok{height=}\DecValTok{1000}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{visPhysics}\NormalTok{(}\DataTypeTok{timestep =} \FloatTok{0.3}\NormalTok{,}
             \DataTypeTok{barnesHut =} \KeywordTok{list}\NormalTok{(}\DataTypeTok{centralGravity=}\FloatTok{0.35}\NormalTok{,}
                              \DataTypeTok{springLength =} \DecValTok{95}\NormalTok{)) }\OperatorTok{%>%}\StringTok{ }
\StringTok{  }\KeywordTok{visOptions}\NormalTok{(}\DataTypeTok{highlightNearest =} \OtherTok{TRUE}\NormalTok{)}

\NormalTok{vn}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## PhantomJS not found. You can install it with webshot::install_phantomjs(). If it is installed, please make sure the phantomjs executable can be found via the PATH variable.
\end{verbatim}

\hypertarget{htmlwidget-35a8d67cef502b7d815a}{}

Figure S1: Dependency network of R packages. Shown in blue are packages mentioned in the review. Edges connect to packages that depend on another package, as long as that is in CRAN or BioC. Green nodes correspond to packages in CRAN or BioC not covered in the review. Not shown are 1) infrastructure packages e.g.~rJava, Rcpp 2) packages from the review without reverse dependencies and 3) data packages. Some packages from the review are not in current versions of CRAN or BioC. An interactive version of this figure is available from \href{appendix-2-metarbolomics-dependencies-network.html}{https://stanstrup.gitlab.io/metaRbolomics-book/appendix-2-metarbolomics-dependencies-network.html}.

\hypertarget{save-network-plot}{%
\subsubsection*{Save network plot}\label{save-network-plot}}
\addcontentsline{toc}{subsubsection}{Save network plot}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{saveNetwork}\NormalTok{(vn, }\StringTok{"vn.html"}\NormalTok{)}
\KeywordTok{webshot}\NormalTok{(}\StringTok{"vn.html"}\NormalTok{, }\StringTok{"revDepNet-60.png"}\NormalTok{, }\DataTypeTok{delay =} \DecValTok{60}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{The_MetaRbolomics_book_files/figure-latex/fileVis-1.png}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{vnZoom <-}\StringTok{ }\KeywordTok{visNetwork}\NormalTok{(}\DataTypeTok{nodes =}\NormalTok{ data}\OperatorTok{$}\NormalTok{nodes, }
                 \DataTypeTok{edges =}\NormalTok{ data}\OperatorTok{$}\NormalTok{edges,}
                 \DataTypeTok{width=}\DecValTok{1000}\NormalTok{, }\DataTypeTok{height=}\DecValTok{1000}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{visIgraphLayout}\NormalTok{()}\OperatorTok{%>%}
\StringTok{    }\KeywordTok{visEvents}\NormalTok{(}\DataTypeTok{type=}\StringTok{"once"}\NormalTok{, }\DataTypeTok{startStabilizing =} \StringTok{'function() \{}
\StringTok{              this.fit(\{nodes:["ptw", "Rnmr1D", "RpeakChrom", "alsace", }
\StringTok{                               "PepsNMR", "ASICS", "MODIS", "RGCxGC"]\})}
\StringTok{            \}'}\NormalTok{)}
\KeywordTok{saveNetwork}\NormalTok{(vnZoom, }\StringTok{"vnZoom.html"}\NormalTok{)}
\KeywordTok{webshot}\NormalTok{(}\StringTok{"vnZoom.html"}\NormalTok{, }\StringTok{"revDepNet-zoom.png"}\NormalTok{, }\DataTypeTok{delay =} \DecValTok{5}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{The_MetaRbolomics_book_files/figure-latex/fileVisZoom-1.png}

You ca access the files at:

\begin{itemize}
\tightlist
\item
  \url{vn.html}
\item
  \url{revDepNet-60.png}
\item
  \url{vnZoom.html}
\item
  \url{revDepNet-zoom.png}
\end{itemize}

\hypertarget{notes}{%
\subsubsection*{Notes}\label{notes}}
\addcontentsline{toc}{subsubsection}{Notes}

The source code for this page is on GitHub at \href{https://gitlab.com/stanstrup/metaRbolomics-book}{gitlab.com/stanstrup/metaRbolomics-book}

The HTML output is shown at
\href{appendix-2-metarbolomics-dependencies-network.html}{https://stanstrup.gitlab.io/metaRbolomics-book/appendix-2-metarbolomics-dependencies-network.html}

and \href{vn.html}{https://stanstrup.gitlab.io/metaRbolomics-book/vn.html} (Caveat: long rendering time, blank page without any visible progress)

This page was created with the following packages:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sessionInfo}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## R version 3.6.2 (2019-12-12)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Debian GNU/Linux 10 (buster)
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.3.5.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] desc_1.2.0          png_0.1-7           webshot2_0.0.0.9000
##  [4] chromote_0.0.0.9001 networkD3_0.4       visNetwork_2.0.9   
##  [7] igraph_1.2.4.2      devtools_2.2.1      usethis_1.5.1      
## [10] tikzDevice_0.12.3   purrr_0.3.3         kableExtra_1.1.0   
## [13] DT_0.11             dplyr_0.8.3         googlesheets_0.3.0 
## [16] readr_1.3.1         knitr_1.26         
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.1          pkgload_1.0.2       jsonlite_1.6       
##  [4] viridisLite_0.3.0   shiny_1.4.0         assertthat_0.2.1   
##  [7] BiocManager_1.30.10 cellranger_1.1.0    yaml_2.2.0         
## [10] remotes_2.1.0       sessioninfo_1.1.1   pillar_1.4.3       
## [13] backports_1.1.5     glue_1.3.1          digest_0.6.23      
## [16] promises_1.1.0.9000 rvest_0.3.5         colorspace_1.4-1   
## [19] websocket_1.1.0     htmltools_0.4.0     httpuv_1.5.2       
## [22] pkgconfig_2.0.3     bookdown_0.16       xtable_1.8-4       
## [25] scales_1.1.0        webshot_0.5.2       processx_3.4.1     
## [28] later_1.0.0.9001    tibble_2.1.3        ellipsis_0.3.0     
## [31] withr_2.1.2         cli_2.0.0           magrittr_1.5       
## [34] crayon_1.3.4        mime_0.8            memoise_1.1.0      
## [37] evaluate_0.14       ps_1.3.0            fs_1.3.1           
## [40] fansi_0.4.0         xml2_1.2.2          pkgbuild_1.0.6     
## [43] tools_3.6.2         prettyunits_1.0.2   hms_0.5.2          
## [46] lifecycle_0.1.0     stringr_1.4.0       munsell_0.5.0      
## [49] callr_3.4.0         compiler_3.6.2      rlang_0.4.2        
## [52] grid_3.6.2          rstudioapi_0.10     htmlwidgets_1.5.1  
## [55] filehash_2.4-2      crosstalk_1.0.0     rmarkdown_2.0      
## [58] testthat_2.3.1      codetools_0.2-16    curl_4.3           
## [61] R6_2.4.1            fastmap_1.0.1       zeallot_0.1.0      
## [64] rprojroot_1.3-2     stringi_1.4.3       Rcpp_1.0.3         
## [67] vctrs_0.2.1         tidyselect_0.2.5    xfun_0.11
\end{verbatim}

\end{document}
